---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

# Liver disease atlas

<!-- badges: start -->
<!-- **Publication:** [![DOI](https://img.shields.io/badge/DOI-10.1186%2Fs13073--019--0615--0-blue.svg)][pub] -->
**Data:** [![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.4407208.svg)](https://doi.org/10.5281/zenodo.4407208)
<!-- badges: end -->

This repository contains the code for the analyses presented in _"Transcriptomic cross-species analysis of chronic liver disease reveals a consistent regulation pattern between humans and mice"_.

Rendered Rmarkdown files of all analyses can be found [here](https://saezlab.github.io/liver-disease-atlas/index.html).

## Transcriptomic cross-species analysis of chronic liver disease reveals a consistent regulation pattern between humans and mice

### Abstract

#### Background and aims 
Mouse models are frequently used to study chronic liver diseases (CLD). To allow a better assessment of translational relevance, we quantified the similarity of prevalent mouse models to human CLD based on transcriptome data.

#### Methods
RNA-sequencing and gene array data from 372 patients with CLD (NAFLD, NASH, HCV, PBC, and PSC) were compared to acute and chronic mouse models with 227 mice and additionally 9 published gene sets of chronic mouse models. Genes consistently altered in humans and mice were mapped to specific liver cell types based on single-cell RNA-sequencing and validated by immunostaining. 

#### Results 
Similarity of the top differentially expressed genes between humans and mice varied among the individual mouse models and depended on the period of damage induction. Some models, e.g. 12 months induction of damage by CCl4, reached high similarity to humans with 0.4 recall and 0.33 precision, respectively. Consistently upregulated genes between the chronic CCl4 mouse model and human data were enriched in inflammatory and developmental processes, and mapped to cholangiocytes, macrophages, endothelial and mesenchymal cells, while downregulated genes were enriched in metabolic functions and mapped to hepatocytes. Immunostaining confirmed selected consistent genes and their cell-type specificity. Upregulated genes in both acute and chronic mouse models showed a higher recall and precision with respect to human CLDNAFLD and NASH than exclusively acute or chronic genes. 

#### Conclusion
Our analyses led to the identification of similarly regulated genes in human and mouse liver disease. Although major species differences exist, mouse models may recall at least 30% of the genes significantly altered in human CLDliver diseases. The relevance of individual genes in CLD can be assessed at https://saezlab.shinyapps.io/liverdiseaseatlas/.

-------

### How to cite?
>Holland CH, Ramirez Flores RO, Myllys M, Hassan R, Edlund K, Hofmann U, Marchan R, Cadenas C, Reinders J, Hoehme S, Seddek A, Dooley S, Keitel V, Godoy P, Begher-Tibbe B, Trautwein C, Rupp C, Mueller S, Longerich T, Hengstler JG^#^, Saez-Rodriguez J^#^, Ghallab A^#^. "Transcriptomic cross-species analysis of chronic liver disease reveals a consistent regulation pattern between humans and mice." _In preparation_. 2021.

^#^_Shared senior authorship_
