<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Christian H. Holland" />

<meta name="date" content="2020-12-20" />

<title>Fibrosis patient cohort by Ramnath et al.</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">meta-liver</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fas fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Mouse models
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="mouse-chronic-ccl4.html">Chronic CCl4 mouse model</a>
    </li>
    <li>
      <a href="mouse-acute-apap.html">Acute APAP mouse model</a>
    </li>
    <li>
      <a href="mouse-acute-ccl4.html">Acute CCl4 mouse model</a>
    </li>
    <li>
      <a href="mouse-acute-bdl.html">Acute BDL mouse model</a>
    </li>
    <li>
      <a href="mouse-acute-ph.html">Acute PH mouse model</a>
    </li>
    <li>
      <a href="mouse-acute-lps.html">Acute LPS mouse model</a>
    </li>
    <li>
      <a href="mouse-acute-tunicamycin.html">Acute Tunicamycin mouse model</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Patient cohorts
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="human-hoang-nafld.html">NAFLD - Hoang et al.</a>
    </li>
    <li>
      <a href="human-diehl-nafld.html">NAFLD - Diehl et al.</a>
    </li>
    <li>
      <a href="human-ramnath-fibrosis.html">Fibrosis - Ramnath et al.</a>
    </li>
    <li>
      <a href="human-hampe13-nash.html">NASH - Hampe et al. '13</a>
    </li>
    <li>
      <a href="human-hampe13-misc.html">Misc - Hampe et al. '14</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Meta analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="meta-analysis-chronic-vs-acute.html">Chronic vs. acute</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://zenodo.org/record/xxx">
    <span class="fas fa-database"></span>
     
    Get required data
  </a>
</li>
<li>
  <a href="https://github.com/saezlab/meta-liver">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Fibrosis patient cohort by Ramnath et al.</h1>
<h4 class="author">Christian H. Holland</h4>
<h4 class="date">2020-12-20</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-12-20
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>meta-liver/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20201218code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20201218)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20201218code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20201218)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsaezlabmetalivertreebc5a3e60331a6eaaac5938ee7c4ce37b9430ec7dtargetblankbc5a3e6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/saezlab/meta-liver/tree/bc5a3e60331a6eaaac5938ee7c4ce37b9430ec7d" target="_blank">bc5a3e6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsaezlabmetalivertreebc5a3e60331a6eaaac5938ee7c4ce37b9430ec7dtargetblankbc5a3e6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/saezlab/meta-liver/tree/bc5a3e60331a6eaaac5938ee7c4ce37b9430ec7d" target="_blank">bc5a3e6</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/human-diehl-nafld_cache/
    Ignored:    analysis/human-hampe13-nash_cache/
    Ignored:    analysis/human-hampe14-misc_cache/
    Ignored:    analysis/human-hoang-nafld_cache/
    Ignored:    analysis/meta-analysis-chronic-vs-acute_cache/
    Ignored:    analysis/mouse-acute-apap_cache/
    Ignored:    analysis/mouse-acute-bdl_cache/
    Ignored:    analysis/mouse-acute-ccl4_cache/
    Ignored:    analysis/mouse-acute-lps_cache/
    Ignored:    analysis/mouse-acute-ph_cache/
    Ignored:    analysis/mouse-acute-tunicamycin_cache/
    Ignored:    analysis/mouse-chronic-ccl4_cache/
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/annotation/
    Ignored:    data/human-diehl-nafld/
    Ignored:    data/human-hampe13-nash/
    Ignored:    data/human-hampe14-misc/
    Ignored:    data/human-hoang-nafld/
    Ignored:    data/human-ramnath-fibrosis/
    Ignored:    data/mouse-acute-apap/
    Ignored:    data/mouse-acute-bdl/
    Ignored:    data/mouse-acute-lps/
    Ignored:    data/mouse-acute-ph/
    Ignored:    data/mouse-acute-tunicamycin/
    Ignored:    data/mouse-chronic-ccl4/
    Ignored:    external_software/.DS_Store
    Ignored:    external_software/README.html
    Ignored:    external_software/stem/.DS_Store
    Ignored:    output/.DS_Store
    Ignored:    output/human-diehl-nafld/
    Ignored:    output/human-hampe13-nash/
    Ignored:    output/human-hampe14-misc/
    Ignored:    output/human-hoang-nafld/
    Ignored:    output/human-ramnath-fibrosis/
    Ignored:    output/meta-chronic-vs-acute/
    Ignored:    output/mouse-acute-apap/
    Ignored:    output/mouse-acute-bdl/
    Ignored:    output/mouse-acute-ccl4/
    Ignored:    output/mouse-acute-lps/
    Ignored:    output/mouse-acute-ph/
    Ignored:    output/mouse-acute-tunicamycin/
    Ignored:    output/mouse-chronic-ccl4/
    Ignored:    renv/library/
    Ignored:    renv/staging/

Untracked files:
    Untracked:  code/human-ramnath-fibrosis/
    Untracked:  data/mouse-acute-ccl4/

Unstaged changes:
    Modified:   analysis/_site.yml
    Modified:   analysis/meta-analysis-chronic-vs-acute.Rmd
    Modified:   code/human-hoang-nafld/preprocessing.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/human-ramnath-fibrosis.Rmd</code>) and HTML (<code>docs/human-ramnath-fibrosis.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/saezlab/meta-liver/blob/bc5a3e60331a6eaaac5938ee7c4ce37b9430ec7d/analysis/human-ramnath-fibrosis.Rmd" target="_blank">bc5a3e6</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-20
</td>
<td>
wflow_publish(c("analysis/human*.Rmd“,”analysis/index.Rmd"),
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Here we analysis a patient cohort covering patients suffering from HCV and NAFLD generated by <a href="https://doi.org/10.1172/jci.insight.120274">Ramnath et al.</a>.</p>
</div>
<div id="libraries-and-sources" class="section level2">
<h2>Libraries and sources</h2>
<p>These libraries and sources are used for this analysis.</p>
<pre class="r"><code>library(tidyverse)
library(tidylog)
library(here)

library(edgeR)
library(biobroom)

library(AachenColorPalette)
library(cowplot)
library(lemon)
library(patchwork)

options(&quot;tidylog.display&quot; = list(print))
source(here(&quot;code/utils-rnaseq.R&quot;))
source(here(&quot;code/utils-utils.R&quot;))
source(here(&quot;code/utils-plots.R&quot;))</code></pre>
<p>Definition of global variables that are used throughout this analysis.</p>
<pre class="r"><code># i/o
data_path &lt;- &quot;data/human-ramnath-fibrosis&quot;
output_path &lt;- &quot;output/human-ramnath-fibrosis&quot;

# graphical parameters
# fontsize
fz &lt;- 9</code></pre>
</div>
<div id="preliminary-exploratory-analysis" class="section level2">
<h2>Preliminary exploratory analysis</h2>
<div id="library-size" class="section level3">
<h3>Library size</h3>
<p>Barplot of the library size (total counts) for each of the samples.</p>
<pre class="r"><code>count_matrix &lt;- readRDS(here(data_path, &quot;count_matrix.rds&quot;))

plot_libsize(count_matrix) +
  my_theme(fsize = fz)</code></pre>
<p><img src="figure/human-ramnath-fibrosis.Rmd/lib-size-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="count-distribution" class="section level3">
<h3>Count distribution</h3>
<p>Violin plots of the raw read counts for each of the samples.</p>
<pre class="r"><code>count_matrix &lt;- readRDS(here(data_path, &quot;count_matrix.rds&quot;))
meta &lt;- readRDS(here(data_path, &quot;meta_data.rds&quot;))

count_matrix %&gt;%
  tdy(&quot;gene&quot;, &quot;sample&quot;, &quot;count&quot;, meta) %&gt;%
  arrange(disease) %&gt;%
  ggplot(aes(
    x = fct_reorder(sample, as.numeric(disease)), y = log10(count + 1),
    group = sample, fill = disease
  )) +
  geom_violin() +
  theme(
    axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1),
    legend.position = &quot;top&quot;
  ) +
  labs(x = NULL) +
  my_theme(grid = &quot;no&quot;, fsize = fz)
#&gt; gather: reorganized (sample01, sample02, sample03, sample04, sample05, …) into (sample, count) [was 25369x68, now 1699723x3]
#&gt; left_join: added 9 columns (age, gender, disease, stage, infect, …)
#&gt;            &gt; rows only in x           0
#&gt;            &gt; rows only in y  (        0)
#&gt;            &gt; matched rows     1,699,723
#&gt;            &gt;                 ===========
#&gt;            &gt; rows total       1,699,723</code></pre>
<p><img src="figure/human-ramnath-fibrosis.Rmd/count-distribution-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="pca-of-raw-data" class="section level3">
<h3>PCA of raw data</h3>
<p>PCA plot of raw read counts contextualized based on etiology stages. Before gene with a constant expression across all samples are removed and count values are transformed to log2 scale. Only the top 1000 most variable genes are used as features.</p>
<pre class="r"><code>count_matrix &lt;- readRDS(here(data_path, &quot;count_matrix.rds&quot;))
meta &lt;- readRDS(here(data_path, &quot;meta_data.rds&quot;))

stopifnot(colnames(count_matrix) == meta$sample)

# remove constant expressed genes and transform to log2 scale
preprocessed_count_matrix &lt;- preprocess_count_matrix(count_matrix)
#&gt; Discarding 1920 genes 
#&gt; Keeping 23449 genes


pca_result &lt;- do_pca(preprocessed_count_matrix, meta, top_n_var_genes = 1000)
#&gt; left_join: added 9 columns (age, gender, disease, stage, infect, …)
#&gt;            &gt; rows only in x    0
#&gt;            &gt; rows only in y  ( 0)
#&gt;            &gt; matched rows     67
#&gt;            &gt;                 ====
#&gt;            &gt; rows total       67

plot_pca(pca_result, feature = &quot;disease&quot;) +
  plot_pca(pca_result, feature = &quot;stage&quot;) &amp;
  my_theme(fsize = fz)</code></pre>
<p><img src="figure/human-ramnath-fibrosis.Rmd/pca-raw-data-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="data-processing" class="section level2">
<h2>Data processing</h2>
<div id="normalization" class="section level3">
<h3>Normalization</h3>
<p>Raw read counts are normalized by first filtering out lowly expressed genes, TMM normalization and finally logCPM transformation.</p>
<pre class="r"><code>count_matrix &lt;- readRDS(here(data_path, &quot;count_matrix.rds&quot;))
meta &lt;- readRDS(here(data_path, &quot;meta_data.rds&quot;))

stopifnot(meta$sample == colnames(count_matrix))

dge_obj &lt;- DGEList(count_matrix, group = meta$group)

# filter low read counts, TMM normalization and logCPM transformation
norm &lt;- voom_normalization(dge_obj)
#&gt; Discarding 8654 genes 
#&gt; Keeping 16715 genes

saveRDS(norm, here(output_path, &quot;normalized_expression.rds&quot;))</code></pre>
</div>
<div id="pca-of-normalized-data" class="section level3">
<h3>PCA of normalized data</h3>
<p>PCA plot of normalized expression data contextualized based on etiology stages. Only the top 1000 most variable genes are used as features.</p>
<pre class="r"><code>expr &lt;- readRDS(here(output_path, &quot;normalized_expression.rds&quot;))
meta &lt;- readRDS(here(data_path, &quot;meta_data.rds&quot;))

pca_result &lt;- do_pca(expr, meta, top_n_var_genes = 1000)
#&gt; left_join: added 9 columns (age, gender, disease, stage, infect, …)
#&gt;            &gt; rows only in x    0
#&gt;            &gt; rows only in y  ( 0)
#&gt;            &gt; matched rows     67
#&gt;            &gt;                 ====
#&gt;            &gt; rows total       67

saveRDS(pca_result, here(output_path, &quot;pca_result.rds&quot;))

plot_pca(pca_result, feature = &quot;disease&quot;) +
  plot_pca(pca_result, feature = &quot;stage&quot;) &amp;
  my_theme(fsize = fz)</code></pre>
<p><img src="figure/human-ramnath-fibrosis.Rmd/pca-norm-data-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="differential-gene-expression-analysis" class="section level2">
<h2>Differential gene expression analysis</h2>
<div id="running-limma" class="section level3">
<h3>Running limma</h3>
<p>Differential gene expression analysis via limma with the aim to identify the transcriptomic signatures of HCV and NAFLD.</p>
<pre class="r"><code># load expression and meta data
expr &lt;- readRDS(here(output_path, &quot;normalized_expression.rds&quot;))
meta &lt;- readRDS(here(data_path, &quot;meta_data.rds&quot;))

stopifnot(colnames(expr) == meta$sample)

# build design matrix
design &lt;- model.matrix(~ 0 + group, data = meta)
rownames(design) &lt;- meta$sample
colnames(design) &lt;- levels(meta$group)


# define contrasts
contrasts &lt;- makeContrasts(
  hcv_adv_vs_early = hcv_advanced - hcv_early,
  nafld_adv_vs_early = nafld_advanced - nafld_early,
  levels = design
)

limma_result &lt;- run_limma(expr, design, contrasts) %&gt;%
  assign_deg()
#&gt; Warning: `tbl_df()` is deprecated as of dplyr 1.0.0.
#&gt; Please use `tibble::as_tibble()` instead.
#&gt; This warning is displayed once every 8 hours.
#&gt; Call `lifecycle::last_warnings()` to see where this warning was generated.
#&gt; select: renamed 3 variables (contrast, logFC, pval) and dropped one variable
#&gt; group_by: one grouping variable (contrast)
#&gt; mutate (grouped): new variable &#39;fdr&#39; (double) with 13,118 unique values and 0% NA
#&gt; ungroup: no grouping variables
#&gt; mutate: new variable &#39;regulation&#39; (character) with 3 unique values and 0% NA
#&gt; mutate: converted &#39;regulation&#39; from character to factor (0 new NA)

deg_df &lt;- limma_result %&gt;%
  mutate(contrast = factor(contrast)) %&gt;%
  mutate(contrast_reference = contrast)
#&gt; mutate: no changes
#&gt; mutate: new variable &#39;contrast_reference&#39; (factor) with 2 unique values and 0% NA

saveRDS(deg_df, here(output_path, &quot;limma_result.rds&quot;))</code></pre>
</div>
<div id="volcano-plots" class="section level3">
<h3>Volcano plots</h3>
<p>Volcano plots visualizing the transcriptomic signatures of HCV and NAFLD.</p>
<pre class="r"><code>df &lt;- readRDS(here(output_path, &quot;limma_result.rds&quot;))

df %&gt;%
  plot_volcano() +
  my_theme(grid = &quot;y&quot;, fsize = fz)
#&gt; rename: renamed one variable (p)</code></pre>
<p><img src="figure/human-ramnath-fibrosis.Rmd/volcano-plots-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
#&gt; R version 4.0.2 (2020-06-22)
#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)
#&gt; Running under: macOS Mojave 10.14.5
#&gt; 
#&gt; Matrix products: default
#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
#&gt; 
#&gt; locale:
#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
#&gt; 
#&gt; attached base packages:
#&gt; [1] stats     graphics  grDevices datasets  utils     methods   base     
#&gt; 
#&gt; other attached packages:
#&gt;  [1] patchwork_1.1.1          lemon_0.4.5              cowplot_1.1.0           
#&gt;  [4] AachenColorPalette_1.1.2 biobroom_1.20.0          broom_0.7.3             
#&gt;  [7] edgeR_3.30.3             limma_3.44.3             here_1.0.1              
#&gt; [10] tidylog_1.0.2            forcats_0.5.0            stringr_1.4.0           
#&gt; [13] dplyr_1.0.2              purrr_0.3.4              readr_1.4.0             
#&gt; [16] tidyr_1.1.2              tibble_3.0.4             ggplot2_3.3.2           
#&gt; [19] tidyverse_1.3.0          workflowr_1.6.2         
#&gt; 
#&gt; loaded via a namespace (and not attached):
#&gt;  [1] Biobase_2.48.0      httr_1.4.2          jsonlite_1.7.2     
#&gt;  [4] modelr_0.1.8        assertthat_0.2.1    renv_0.12.3        
#&gt;  [7] cellranger_1.1.0    yaml_2.2.1          pillar_1.4.7       
#&gt; [10] backports_1.2.1     lattice_0.20-41     glue_1.4.2         
#&gt; [13] digest_0.6.27       promises_1.1.1      rvest_0.3.6        
#&gt; [16] colorspace_2.0-0    htmltools_0.5.0     httpuv_1.5.4       
#&gt; [19] plyr_1.8.6          clisymbols_1.2.0    pkgconfig_2.0.3    
#&gt; [22] haven_2.3.1         scales_1.1.1        whisker_0.4        
#&gt; [25] later_1.1.0.1       git2r_0.27.1        farver_2.0.3       
#&gt; [28] generics_0.1.0      ellipsis_0.3.1      withr_2.3.0        
#&gt; [31] BiocGenerics_0.34.0 cli_2.2.0           magrittr_2.0.1     
#&gt; [34] crayon_1.3.4        readxl_1.3.1        evaluate_0.14      
#&gt; [37] fs_1.5.0            fansi_0.4.1         xml2_1.3.2         
#&gt; [40] tools_4.0.2         hms_0.5.3           lifecycle_0.2.0    
#&gt; [43] munsell_0.5.0       reprex_0.3.0        locfit_1.5-9.4     
#&gt; [46] compiler_4.0.2      rlang_0.4.9         grid_4.0.2         
#&gt; [49] rstudioapi_0.13     labeling_0.4.2      rmarkdown_2.6      
#&gt; [52] codetools_0.2-16    gtable_0.3.0        DBI_1.1.0          
#&gt; [55] R6_2.5.0            gridExtra_2.3       lubridate_1.7.9.2  
#&gt; [58] knitr_1.30          rprojroot_2.0.2     stringi_1.5.3      
#&gt; [61] parallel_4.0.2      Rcpp_1.0.5          vctrs_0.3.6        
#&gt; [64] dbplyr_2.0.0        tidyselect_1.1.0    xfun_0.19</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
