<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Meta anlaysis of patients cohorts and chronic (CCl4) mouse models</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Liver disease atlas</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fas fa-home"></span>
     
    Home
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Mouse models
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="mouse-chronic-ccl4.html">CCl4 (chronic)</a>
    </li>
    <li>
      <a href="mouse-acute-apap.html">APAP (acute)</a>
    </li>
    <li>
      <a href="mouse-acute-ccl4.html">CCl4 (acute)</a>
    </li>
    <li>
      <a href="mouse-acute-bdl.html">BDL (acute)</a>
    </li>
    <li>
      <a href="mouse-acute-ph.html">PH (acute)</a>
    </li>
    <li>
      <a href="mouse-acute-lps.html">LPS (acute)</a>
    </li>
    <li>
      <a href="mouse-acute-tunicamycin.html">Tunicamycin (acute)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Patient cohorts
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="human-hoang-nafld.html">NAFLD - Hoang et al.</a>
    </li>
    <li>
      <a href="human-diehl-nafld.html">NAFLD - Diehl et al.</a>
    </li>
    <li>
      <a href="human-ramnath-fibrosis.html">Fibrosis - Ramnath et al.</a>
    </li>
    <li>
      <a href="human-hampe13-nash.html">NASH - Hampe et al. '13</a>
    </li>
    <li>
      <a href="human-hampe14-misc.html">Misc - Hampe et al. '14</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Meta analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="meta-chronic-vs-acute.html">Chronic vs. acute</a>
    </li>
    <li>
      <a href="meta-mouse-vs-human.html">Mouse vs. human</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Figures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="plot-study-overview.html">Study overview (SFig. 0.1)</a>
    </li>
    <li>
      <a href="plot-chronic-ccl4.html">Chronic CCl4 (Fig. 1)</a>
    </li>
    <li>
      <a href="plot-acute-apap.html">APAP (Fig. 2)</a>
    </li>
    <li>
      <a href="plot-acute-ccl4.html">Acute CCl4 (SFig. 2.2)</a>
    </li>
    <li>
      <a href="plot-acute-ph.html">PH (SFig. 2.4)</a>
    </li>
    <li>
      <a href="plot-acute-bdl.html">BDL (SFig. 2.6)</a>
    </li>
    <li>
      <a href="plot-chronic-vs-acute.html">Chronic vs. acute (Fig. 3)</a>
    </li>
    <li>
      <a href="plot-mouse-vs-human.html">Mouse vs. human (Fig. 4)</a>
    </li>
    <li>
      <a href="plot-precision-recall.html">Precision and recall analysis (Fig. 6)</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://doi.org/10.5281/zenodo.4407208">
    <span class="fas fa-database"></span>
     
    Get data
  </a>
</li>
<li>
  <a href="https://github.com/saezlab/liver-disease-atlas">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Meta anlaysis of patients cohorts and chronic (CCl4) mouse models</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-02-27
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>liver-disease-atlas/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20201218code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20201218)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20201218code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20201218)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsaezlabliverdiseaseatlastree40203d5fcfd0af9fa5b006d6711874376826d215targetblank40203d5a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/saezlab/liver-disease-atlas/tree/40203d5fcfd0af9fa5b006d6711874376826d215" target="_blank">40203d5</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsaezlabliverdiseaseatlastree40203d5fcfd0af9fa5b006d6711874376826d215targetblank40203d5a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/saezlab/liver-disease-atlas/tree/40203d5fcfd0af9fa5b006d6711874376826d215" target="_blank">40203d5</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/human-diehl-nafld_cache/
    Ignored:    analysis/human-hampe13-nash_cache/
    Ignored:    analysis/human-hampe14-misc_cache/
    Ignored:    analysis/human-hoang-nafld_cache/
    Ignored:    analysis/human-ramnath-fibrosis_cache/
    Ignored:    analysis/meta-chronic-vs-acute_cache/
    Ignored:    analysis/mouse-acute-apap_cache/
    Ignored:    analysis/mouse-acute-bdl_cache/
    Ignored:    analysis/mouse-acute-ccl4_cache/
    Ignored:    analysis/mouse-acute-lps_cache/
    Ignored:    analysis/mouse-acute-ph_cache/
    Ignored:    analysis/mouse-acute-tunicamycin_cache/
    Ignored:    analysis/mouse-chronic-ccl4_cache/
    Ignored:    analysis/plot-acute-apap_cache/
    Ignored:    analysis/plot-acute-bdl_cache/
    Ignored:    analysis/plot-acute-ccl4_cache/
    Ignored:    analysis/plot-acute-ph_cache/
    Ignored:    analysis/plot-chronic-ccl4_cache/
    Ignored:    analysis/plot-chronic-vs-acute_cache/
    Ignored:    analysis/plot-mouse-vs-human_cache/
    Ignored:    analysis/plot-precision-recall_cache/
    Ignored:    analysis/plot-study-overview_cache/
    Ignored:    analysis/plot-teufel-integration_cache/
    Ignored:    analysis/save-tables_cache/
    Ignored:    code/.DS_Store
    Ignored:    code/README.html
    Ignored:    code/meta-mouse-vs-human/.DS_Store
    Ignored:    data.zip
    Ignored:    data/.DS_Store
    Ignored:    data/annotation/
    Ignored:    data/human-diehl-nafld/
    Ignored:    data/human-hampe13-nash/
    Ignored:    data/human-hampe14-misc/
    Ignored:    data/human-hoang-nafld/
    Ignored:    data/human-ramnath-fibrosis/
    Ignored:    data/meta-chronic-vs-acute/
    Ignored:    data/meta-mouse-vs-human/
    Ignored:    data/mouse-acute-apap/
    Ignored:    data/mouse-acute-bdl/
    Ignored:    data/mouse-acute-ccl4/
    Ignored:    data/mouse-acute-lps/
    Ignored:    data/mouse-acute-ph/
    Ignored:    data/mouse-acute-tunicamycin/
    Ignored:    data/mouse-chronic-ccl4/
    Ignored:    external_software/.DS_Store
    Ignored:    external_software/README.html
    Ignored:    external_software/stem/.DS_Store
    Ignored:    figures/
    Ignored:    geo_submission/
    Ignored:    output/.DS_Store
    Ignored:    output/README.html
    Ignored:    output/human-diehl-nafld/
    Ignored:    output/human-hampe13-nash/
    Ignored:    output/human-hampe14-misc/
    Ignored:    output/human-hoang-nafld/
    Ignored:    output/human-ramnath-fibrosis/
    Ignored:    output/meta-chronic-vs-acute/
    Ignored:    output/meta-mouse-vs-human/
    Ignored:    output/mouse-acute-apap/
    Ignored:    output/mouse-acute-bdl/
    Ignored:    output/mouse-acute-ccl4/
    Ignored:    output/mouse-acute-lps/
    Ignored:    output/mouse-acute-ph/
    Ignored:    output/mouse-acute-tunicamycin/
    Ignored:    output/mouse-chronic-ccl4/
    Ignored:    renv/library/
    Ignored:    renv/staging/
    Ignored:    tables/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/meta-mouse-vs-human.Rmd</code>) and HTML (<code>docs/meta-mouse-vs-human.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/saezlab/liver-disease-atlas/c6bfb3b22a5734c0b4669c86bf8e6829e5e79125/docs/meta-mouse-vs-human.html" target="_blank">c6bfb3b</a>
</td>
<td>
christianholland
</td>
<td>
2021-02-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/c541e97f41b9ff7a39aaa24044ec8106b69d6b05/analysis/meta-mouse-vs-human.Rmd" target="_blank">c541e97</a>
</td>
<td>
christianholland
</td>
<td>
2021-02-27
</td>
<td>
added pr for chronicitiy
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/saezlab/liver-disease-atlas/9c62197c1a318b00f52dbf3e9e3613ebd34b9b33/docs/meta-mouse-vs-human.html" target="_blank">9c62197</a>
</td>
<td>
christianholland
</td>
<td>
2021-01-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/saezlab/liver-disease-atlas/45a33e4bf9976b61178a021e1c8f7a32ac72697e/docs/meta-mouse-vs-human.html" target="_blank">45a33e4</a>
</td>
<td>
christianholland
</td>
<td>
2021-01-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/e6ada135831299c07c500b553a2bade8628ff65a/analysis/meta-mouse-vs-human.Rmd" target="_blank">e6ada13</a>
</td>
<td>
christianholland
</td>
<td>
2021-01-07
</td>
<td>
added example gsea plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/saezlab/liver-disease-atlas/73708b19dedba527877512a3a2af452a702962be/docs/meta-mouse-vs-human.html" target="_blank">73708b1</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/e0164baee3fa08d8e244bf02e04ab52451064edf/analysis/meta-mouse-vs-human.Rmd" target="_blank">e0164ba</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-29
</td>
<td>
wflow_publish(c(“analysis/meta-mouse-vs-human.Rmd”, “analysis/plot-precision-recall.Rmd”),
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/saezlab/liver-disease-atlas/beaeae06be81850d16b0c023633b0e640d5b47a4/docs/meta-mouse-vs-human.html" target="_blank">beaeae0</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/f20866cebd9d059f3d4da269f394992148f6f35a/analysis/meta-mouse-vs-human.Rmd" target="_blank">f20866c</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-29
</td>
<td>
added code for teufel integration
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/saezlab/liver-disease-atlas/067c9337e4cb6becda6b2d2ace64a70dc92107a3/docs/meta-mouse-vs-human.html" target="_blank">067c933</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/d4f78fae08eb96da82f28d39a0c6d2962abb7ecd/analysis/meta-mouse-vs-human.Rmd" target="_blank">d4f78fa</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-23
</td>
<td>
wflow_publish("analysis/*.Rmd", delete_cache = T)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/saezlab/liver-disease-atlas/bf6fcc69c8835ef80fc56aa386ae6deda2b44db6/docs/meta-mouse-vs-human.html" target="_blank">bf6fcc6</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/f97eb55864265696bdc3226c4fcfd1e4e93d679a/analysis/meta-mouse-vs-human.Rmd" target="_blank">f97eb55</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-22
</td>
<td>
added deg numbers of mouse models
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/saezlab/liver-disease-atlas/f8d83433873c01bdc55750a44f5e3a14f0a6c338/docs/meta-mouse-vs-human.html" target="_blank">f8d8343</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/ba489192cbcdd22f0be5f02d6ee20365bb7aa62e/analysis/meta-mouse-vs-human.Rmd" target="_blank">ba48919</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-21
</td>
<td>
added go downstream analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/saezlab/liver-disease-atlas/d7ef743c73acc1d9fb82d5d3c390b4aa7c8d5eb9/docs/meta-mouse-vs-human.html" target="_blank">d7ef743</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/e686c2a615345f43572458e1b2ec2ff4801e5ce3/analysis/meta-mouse-vs-human.Rmd" target="_blank">e686c2a</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-20
</td>
<td>
complete analyses
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Here we integrate various patient cohorts of chronic liver diseases with the chronic CCl4 mouse model to identify consistently deregulated genes in mouse and human.</p>
</div>
<div id="libraries-and-sources" class="section level2">
<h2>Libraries and sources</h2>
<p>These libraries and sources are used for this analysis.</p>
<pre class="r"><code>library(tidyverse)
library(tidylog)
library(here)
library(readxl)
library(glue)
library(tidytext)
library(janitor)

library(fgsea)
library(dorothea)
library(progeny)
library(biobroom)

library(circlize)
library(AachenColorPalette)
library(lemon)
library(VennDiagram)
library(ComplexHeatmap)
library(gridExtra)
library(cowplot)
library(ggpubr)
library(UpSetR)
library(ggwordcloud)
library(patchwork)

library(msigdf) # remotes::install_github(&quot;ToledoEM/msigdf@v7.1&quot;)
library(gtools)

options(&quot;tidylog.display&quot; = list(print))
source(here(&quot;code/utils-utils.R&quot;))
source(here(&quot;code/utils-plots.R&quot;))</code></pre>
<p>Definition of global variables that are used throughout this analysis.</p>
<pre class="r"><code># i/o
data_path &lt;- &quot;data/meta-mouse-vs-human&quot;
output_path &lt;- &quot;output/meta-mouse-vs-human&quot;

# graphical parameters
# fontsize
fz &lt;- 9
# color function for heatmaps
col_fun &lt;- colorRamp2(
  c(-4, 0, 4),
  c(aachen_color(&quot;blue&quot;), &quot;white&quot;, aachen_color(&quot;red&quot;))
)</code></pre>
</div>
<div id="merging-data-of-all-mouse-models" class="section level2">
<h2>Merging data of all mouse models</h2>
<div id="merging-contrasts" class="section level3">
<h3>Merging contrasts</h3>
<p>Contrasts from all available patient cohorts are merged into a single object.</p>
<pre class="r"><code>diehl &lt;- readRDS(here(&quot;output/human-diehl-nafld/limma_result.rds&quot;)) %&gt;%
  mutate(
    phenotype = &quot;nafld&quot;,
    source = &quot;diehl&quot;
  )

hoang &lt;- readRDS(here(&quot;output/human-hoang-nafld/limma_result.rds&quot;)) %&gt;%
  filter(contrast_reference == &quot;stage_0&quot;) %&gt;%
  mutate(
    phenotype = &quot;nafld&quot;,
    source = &quot;hoang&quot;
  )
#&gt; filter: no rows removed

hampe13 &lt;- readRDS(here(&quot;output/human-hampe13-nash/limma_result.rds&quot;)) %&gt;%
  filter(contrast_reference == &quot;control&quot;) %&gt;%
  mutate(
    phenotype = &quot;nash&quot;,
    source = &quot;hampe13&quot;
  )
#&gt; filter: no rows removed

hampe14 &lt;- readRDS(here(&quot;output/human-hampe14-misc/limma_result.rds&quot;)) %&gt;%
  filter(contrast_reference == &quot;control&quot;) %&gt;%
  mutate(
    phenotype = &quot;omni&quot;,
    source = &quot;hampe14&quot;
  )
#&gt; filter: no rows removed

ramnath &lt;- readRDS(here(&quot;output/human-ramnath-fibrosis/limma_result.rds&quot;)) %&gt;%
  mutate(
    phenotype = &quot;fibrosis&quot;,
    source = &quot;ramnath&quot;
  )

combined_contrasts &lt;- bind_rows(
  diehl, ramnath, hoang, hampe13,
  hampe14
) %&gt;%
  select(-contrast_reference) %&gt;%
  mutate(contrast = as_factor(contrast)) %&gt;%
  assign_deg() %&gt;%
  # phenotypes related to obesity are removed
  filter(phenotype != &quot;obesity&quot;) %&gt;%
  filter(!str_detect(contrast, &quot;obese&quot;))
#&gt; select: dropped one variable (contrast_reference)
#&gt; filter: no rows removed
#&gt; filter: removed 37,642 rows (13%), 257,688 rows remaining

saveRDS(combined_contrasts, here(output_path, &quot;limma_result.rds&quot;))</code></pre>
</div>
<div id="merging-meta-data" class="section level3">
<h3>Merging meta data</h3>
<p>Meta data from all available patient cohorts are merged into a single object.</p>
<pre class="r"><code>diehl &lt;- readRDS(here(&quot;output/human-diehl-nafld/meta_data.rds&quot;)) %&gt;%
  select(sample, group = class) %&gt;%
  mutate(
    phenotype = &quot;nafld&quot;,
    source = &quot;diehl&quot;
  )
#&gt; select: renamed one variable (group)

hoang &lt;- readRDS(here(&quot;data/human-hoang-nafld/meta_data.rds&quot;)) %&gt;%
  select(sample, group = nafld) %&gt;%
  mutate(
    phenotype = &quot;nafld&quot;,
    source = &quot;hoang&quot;
  )
#&gt; select: renamed one variable (group) and dropped 6 variables

hampe13 &lt;- readRDS(here(&quot;output/human-hampe13-nash/meta_data.rds&quot;)) %&gt;%
  select(sample, group) %&gt;%
  mutate(
    phenotype = &quot;nash&quot;,
    source = &quot;hampe13&quot;
  )
#&gt; select: dropped 6 variables (gender, inflammation, fibrosis, age, bmi, …)

hampe14 &lt;- readRDS(here(&quot;output/human-hampe14-misc/meta_data.rds&quot;)) %&gt;%
  select(sample, group) %&gt;%
  mutate(
    phenotype = &quot;omni&quot;,
    source = &quot;hampe14&quot;
  )
#&gt; select: dropped 3 variables (gender, age, bmi)

ramnath &lt;- readRDS(here(&quot;data/human-ramnath-fibrosis/meta_data.rds&quot;)) %&gt;%
  select(sample, group) %&gt;%
  mutate(
    phenotype = &quot;fibrosis&quot;,
    source = &quot;ramnath&quot;
  )
#&gt; select: dropped 8 variables (age, gender, disease, stage, infect, …)

combined_meta &lt;- bind_rows(
  diehl, ramnath, hoang, hampe13,
  hampe14
) %&gt;%
  # patients related to obesity are removed
  filter(group != &quot;obese&quot;) %&gt;%
  # assign whether patient is sick or healthy (control)
  mutate(class = case_when(
    str_detect(
      group, &quot;early|control|stage_0|mild&quot;
    ) ~ &quot;control&quot;,
    TRUE ~ &quot;disease&quot;
  ))
#&gt; filter: removed 51 rows (12%), 372 rows remaining

saveRDS(combined_meta, here(output_path, &quot;meta_data.rds&quot;))</code></pre>
</div>
</div>
<div id="number-of-patients" class="section level2">
<h2>Number of patients</h2>
<p>Barplot showing number of patients per study.</p>
<pre class="r"><code>df &lt;- readRDS(here(output_path, &quot;meta_data.rds&quot;)) %&gt;%
  count(phenotype, source, class)
#&gt; count: now 10 rows and 4 columns, ungrouped

df %&gt;%
  ggplot(aes(
    x = n, fct_reorder(interaction(source, phenotype, sep = &quot;_&quot;), n),
    group = class, fill = class
  )) +
  geom_col(position = &quot;dodge&quot;) +
  labs(x = &quot;Number of patients&quot;, y = &quot;Study&quot;) +
  my_theme(grid = &quot;x&quot;, fsize = fz)</code></pre>
<p><img src="figure/meta-mouse-vs-human.Rmd/patient-number-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-patient-number-1">
Past versions of patient-number-1.png
</button>
</p>
<div id="fig-patient-number-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/f8d83433873c01bdc55750a44f5e3a14f0a6c338/docs/figure/meta-mouse-vs-human.Rmd/patient-number-1.png" target="_blank">f8d8343</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gene-coverage" class="section level2">
<h2>Gene coverage</h2>
<p>Barplot showing the gene coverage of the patient cohorts.</p>
<pre class="r"><code>contrasts &lt;- readRDS(here(output_path, &quot;limma_result.rds&quot;)) %&gt;%
  distinct(gene, phenotype, source) %&gt;%
  count(phenotype, source)
#&gt; distinct: removed 167,964 rows (65%), 89,724 rows remaining
#&gt; count: now 5 rows and 3 columns, ungrouped

contrasts %&gt;%
  ggplot(aes(
    x = n, fct_reorder(interaction(source, phenotype, sep = &quot;_&quot;), n),
    group = source
  )) +
  geom_col() +
  labs(x = &quot;Gene coverage&quot;, y = &quot;Study&quot;) +
  my_theme(grid = &quot;x&quot;, fsize = fz)</code></pre>
<p><img src="figure/meta-mouse-vs-human.Rmd/gene-coverage-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-gene-coverage-1">
Past versions of gene-coverage-1.png
</button>
</p>
<div id="fig-gene-coverage-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/f8d83433873c01bdc55750a44f5e3a14f0a6c338/docs/figure/meta-mouse-vs-human.Rmd/gene-coverage-1.png" target="_blank">f8d8343</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/d7ef743c73acc1d9fb82d5d3c390b4aa7c8d5eb9/docs/figure/meta-mouse-vs-human.Rmd/gene-coverage-1.png" target="_blank">d7ef743</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="number-of-differential-expressed-genes" class="section level2">
<h2>Number of differential expressed genes</h2>
<p>Barplot showing the number of differentially expressed genes for each contrast.</p>
<pre class="r"><code>contrasts &lt;- readRDS(here(output_path, &quot;limma_result.rds&quot;))

combined_contrasts %&gt;%
  filter(regulation != &quot;ns&quot;) %&gt;%
  count(contrast, source, regulation) %&gt;%
  ggplot(aes(y = interaction(source, contrast), x = n, fill = regulation)) +
  geom_col(position = position_dodge()) +
  labs(y = NULL, y = &quot;Number of degs&quot;) +
  my_theme(grid = &quot;x&quot;, fsize = fz)
#&gt; filter: removed 255,692 rows (99%), 1,996 rows remaining
#&gt; count: now 26 rows and 4 columns, ungrouped</code></pre>
<p><img src="figure/meta-mouse-vs-human.Rmd/deg-number-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-deg-number-1">
Past versions of deg-number-1.png
</button>
</p>
<div id="fig-deg-number-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/f8d83433873c01bdc55750a44f5e3a14f0a6c338/docs/figure/meta-mouse-vs-human.Rmd/deg-number-1.png" target="_blank">f8d8343</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="interstudy-analysis-of-patient-cohorts" class="section level2">
<h2>Interstudy analysis of patient cohorts</h2>
<div id="mutual-similarity-of-differential-expressed-genes" class="section level3">
<h3>Mutual similarity of differential expressed genes</h3>
<p>This analysis computes the similarity of differential expressed genes for all contrasts of the patient cohorts. Similarity is measured with the Jaccard Index.</p>
<pre class="r"><code>contrasts &lt;- readRDS(here(output_path, &quot;limma_result.rds&quot;))

# populate gene sets with a fixed size selected by effect size (t-value)
mat_top &lt;- contrasts %&gt;%
  group_by(contrast, phenotype, source) %&gt;%
  top_n(500, abs(statistic)) %&gt;%
  mutate(key = row_number()) %&gt;%
  ungroup() %&gt;%
  unite(geneset, source, phenotype, contrast, sep = &quot;-&quot;) %&gt;%
  mutate(geneset = as_factor(geneset)) %&gt;%
  select(geneset, gene, key) %&gt;%
  untdy(key, geneset, gene)
#&gt; top_n (grouped): removed 250,188 rows (97%), 7,500 rows remaining
#&gt; ungroup: no grouping variables
#&gt; select: dropped 5 variables (logFC, statistic, pval, fdr, regulation)
#&gt; select: columns reordered (key, geneset, gene)
#&gt; spread: reorganized (geneset, gene) into (diehl-nafld-advanced_vs_mild, ramnath-fibrosis-hcv_adv_vs_early, ramnath-fibrosis-nafld_adv_vs_early, hoang-nafld-stage_1_vs_0, hoang-nafld-stage_2_vs_0, …) [was 7500x3, now 500x16]

# usage of jaccard index for balanced set sizes
j &lt;- set_similarity(mat_top, measure = &quot;jaccard&quot;, tidy = T)
#&gt; gather: reorganized (diehl-nafld-advanced_vs_mild, ramnath-fibrosis-hcv_adv_vs_early, ramnath-fibrosis-nafld_adv_vs_early, hoang-nafld-stage_1_vs_0, hoang-nafld-stage_2_vs_0, …) into (set2, similarity) [was 15x16, now 225x3]
#&gt; drop_na: removed 105 rows (47%), 120 rows remaining
#&gt; filter: removed 15 rows (12%), 105 rows remaining
#&gt; mutate_if: converted &#39;set1&#39; from character to factor (0 new NA)
#&gt;            converted &#39;set2&#39; from character to factor (0 new NA)

saveRDS(j, here(output_path, &quot;gene_set_similarity.rds&quot;))

j %&gt;%
  ggplot(aes(x = set1, y = set2, fill = similarity)) +
  geom_tile(color = &quot;black&quot;) +
  scale_fill_gradient(low = &quot;white&quot;, high = aachen_color(&quot;green&quot;)) +
  labs(x = NULL, y = NULL, fill = &quot;Jaccard\nIndex&quot;) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  my_theme(fsize = fz, grid = &quot;no&quot;)</code></pre>
<p><img src="figure/meta-mouse-vs-human.Rmd/deg-similarity-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-deg-similarity-1">
Past versions of deg-similarity-1.png
</button>
</p>
<div id="fig-deg-similarity-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/d7ef743c73acc1d9fb82d5d3c390b4aa7c8d5eb9/docs/figure/meta-mouse-vs-human.Rmd/deg-similarity-1.png" target="_blank">d7ef743</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="mutual-enrichment-of-differential-expressed-genes" class="section level3">
<h3>Mutual enrichment of differential expressed genes</h3>
<p>This analysis explores whether the top differential expressed genes of specific contrasts of the acute mouse models are consistently regulated across the acute mouse models.</p>
<pre class="r"><code>contrasts &lt;- readRDS(here(output_path, &quot;limma_result.rds&quot;))

# populate gene sets with a fixed size selected by effect size (t-value)
genesets_top &lt;- contrasts %&gt;%
  mutate(direction = case_when(
    sign(statistic) &gt;= 0 ~ &quot;up&quot;,
    sign(statistic) &lt; 0 ~ &quot;down&quot;
  )) %&gt;%
  group_by(source, phenotype, contrast, direction) %&gt;%
  top_n(500, abs(statistic)) %&gt;%
  ungroup() %&gt;%
  unite(geneset, source, phenotype, contrast, sep = &quot;-&quot;) %&gt;%
  unite(geneset, geneset, direction, sep = &quot;|&quot;) %&gt;%
  mutate(geneset = as_factor(geneset)) %&gt;%
  select(geneset, gene)
#&gt; top_n (grouped): removed 242,688 rows (94%), 15,000 rows remaining
#&gt; ungroup: no grouping variables
#&gt; select: dropped 5 variables (logFC, statistic, pval, fdr, regulation)

# construct signature matrix/data frame
signature_df &lt;- contrasts %&gt;%
  unite(signature, source, phenotype, contrast, sep = &quot;-&quot;) %&gt;%
  mutate(signature = as_factor(signature)) %&gt;%
  untdy(&quot;gene&quot;, &quot;signature&quot;, &quot;statistic&quot;)
#&gt; select: dropped 4 variables (logFC, pval, fdr, regulation)
#&gt; spread: reorganized (signature, statistic) into (diehl-nafld-advanced_vs_mild, ramnath-fibrosis-hcv_adv_vs_early, ramnath-fibrosis-nafld_adv_vs_early, hoang-nafld-stage_1_vs_0, hoang-nafld-stage_2_vs_0, …) [was 257688x3, now 24107x16]

# run gsea
set.seed(123)
gsea_res_top &lt;- run_gsea(signature_df, genesets_top, tidy = T) %&gt;%
  separate(geneset, into = c(&quot;geneset&quot;, &quot;direction&quot;), sep = &quot;[|]&quot;) %&gt;%
  mutate(
    signature = as_factor(signature),
    geneset = as_factor(geneset)
  )
#&gt; summarise: now 30 rows and 2 columns, ungrouped
#&gt; rename: renamed one variable (geneset)
#&gt; select: dropped one variable (gene)
#&gt; distinct: removed 14,970 rows (&gt;99%), 30 rows remaining
#&gt; left_join: added no columns
#&gt;            &gt; rows only in x     0
#&gt;            &gt; rows only in y  (  0)
#&gt;            &gt; matched rows     450
#&gt;            &gt;                 =====
#&gt;            &gt; rows total       450

saveRDS(gsea_res_top, here(output_path, &quot;interstudy_enrichment.rds&quot;))

# individual gsea plot
make_gsea_plot(
  signature = signature_df %&gt;%
    rownames_to_column(&quot;gene&quot;) %&gt;%
    select(gene, t = `hoang-nafld-stage_6_vs_0`) %&gt;%
    drop_na(),
  geneset = genesets_top %&gt;%
    filter(geneset == &quot;diehl-nafld-advanced_vs_mild|up&quot;),
  gene_level_stat = t
)
#&gt; select: renamed one variable (t) and dropped 14 variables
#&gt; drop_na: removed 8,914 rows (37%), 15,193 rows remaining
#&gt; rename: renamed one variable (stat)
#&gt; transmute: changed 15,193 values (100%) of &#39;stat&#39; (0 new NA)
#&gt;            new variable &#39;rank&#39; (integer) with 15,193 unique values and 0% NA
#&gt; select: dropped one variable (rank)
#&gt; filter: removed 14,500 rows (97%), 500 rows remaining
#&gt; inner_join: added one column (geneset)
#&gt;             &gt; rows only in x  (14,720)
#&gt;             &gt; rows only in y  (    27)
#&gt;             &gt; matched rows        473
#&gt;             &gt;                 ========
#&gt;             &gt; rows total          473
#&gt; top_n: removed 947 rows (&gt;99%), one row remaining</code></pre>
<p><img src="figure/meta-mouse-vs-human.Rmd/mututal-enrichment-degs-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-mututal-enrichment-degs-1">
Past versions of mututal-enrichment-degs-1.png
</button>
</p>
<div id="fig-mututal-enrichment-degs-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/45a33e4bf9976b61178a021e1c8f7a32ac72697e/docs/figure/meta-mouse-vs-human.Rmd/mututal-enrichment-degs-1.png" target="_blank">45a33e4</a>
</td>
<td>
christianholland
</td>
<td>
2021-01-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/d7ef743c73acc1d9fb82d5d3c390b4aa7c8d5eb9/docs/figure/meta-mouse-vs-human.Rmd/mututal-enrichment-degs-1.png" target="_blank">d7ef743</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>
# summary gsea plot
gsea_res_top %&gt;%
  mutate(label = stars.pval(padj)) %&gt;%
  ggplot(aes(x = signature, y = geneset, fill = ES)) +
  geom_tile() +
  geom_text(aes(label = label)) +
  facet_wrap(~direction) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  scale_fill_gradient2() +
  my_theme(fsize = fz, grid = &quot;no&quot;) +
  labs(x = &quot;Signature&quot;, y = &quot;Gene set&quot;)</code></pre>
<p><img src="figure/meta-mouse-vs-human.Rmd/mututal-enrichment-degs-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-mututal-enrichment-degs-2">
Past versions of mututal-enrichment-degs-2.png
</button>
</p>
<div id="fig-mututal-enrichment-degs-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/45a33e4bf9976b61178a021e1c8f7a32ac72697e/docs/figure/meta-mouse-vs-human.Rmd/mututal-enrichment-degs-2.png" target="_blank">45a33e4</a>
</td>
<td>
christianholland
</td>
<td>
2021-01-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="comparison-of-mouse-and-human-data" class="section level2">
<h2>Comparison of mouse and human data</h2>
<p>With this analysis we check whether the top differential expressed human genes have the same direction of regulation in the chronic mouse model.</p>
<div id="enrichment-of-human-gene-sets-in-mouse-signatures" class="section level3">
<h3>Enrichment of human gene sets in mouse signatures</h3>
<pre class="r"><code>contrasts &lt;- readRDS(here(output_path, &quot;limma_result.rds&quot;))

chronic_mouse &lt;- readRDS(
  here(&quot;output/mouse-chronic-ccl4/limma_result_hs.rds&quot;)
) %&gt;%
  filter(contrast_reference == &quot;pure_ccl4&quot;)
#&gt; filter: removed 120,807 rows (75%), 40,269 rows remaining

# populate gene sets with a fixed size selected by effect size (t-value)
genesets_top &lt;- contrasts %&gt;%
  mutate(direction = case_when(
    sign(statistic) &gt;= 0 ~ &quot;up&quot;,
    sign(statistic) &lt; 0 ~ &quot;down&quot;
  )) %&gt;%
  group_by(source, phenotype, contrast, direction) %&gt;%
  top_n(500, abs(statistic)) %&gt;%
  ungroup() %&gt;%
  unite(geneset, source, phenotype, contrast, sep = &quot;-&quot;) %&gt;%
  unite(geneset, geneset, direction, sep = &quot;|&quot;) %&gt;%
  mutate(geneset = as_factor(geneset)) %&gt;%
  select(geneset, gene)
#&gt; top_n (grouped): removed 242,688 rows (94%), 15,000 rows remaining
#&gt; ungroup: no grouping variables
#&gt; select: dropped 5 variables (logFC, statistic, pval, fdr, regulation)

signature_df &lt;- chronic_mouse %&gt;%
  untdy(&quot;gene&quot;, &quot;contrast&quot;, &quot;statistic&quot;)
#&gt; select: dropped 5 variables (logFC, pval, fdr, regulation, contrast_reference)
#&gt; spread: reorganized (contrast, statistic) into (pure_ccl_2m_vs_0m, pure_ccl_6m_vs_0m, pure_ccl_12m_vs_0m) [was 40269x3, now 13423x4]

# run gsea
set.seed(123)
gsea_res_top &lt;- run_gsea(signature_df, genesets_top, tidy = T) %&gt;%
  separate(geneset, into = c(&quot;geneset&quot;, &quot;direction&quot;), sep = &quot;[|]&quot;) %&gt;%
  separate(geneset,
    into = c(&quot;source&quot;, &quot;phenotype&quot;, &quot;contrast&quot;), sep = &quot;-&quot;,
    remove = F
  ) %&gt;%
  mutate(
    signature = as_factor(signature),
    geneset = as_factor(geneset),
    time = parse_number(as.character(signature))
  )
#&gt; summarise: now 30 rows and 2 columns, ungrouped
#&gt; rename: renamed one variable (geneset)
#&gt; select: dropped one variable (gene)
#&gt; distinct: removed 14,970 rows (&gt;99%), 30 rows remaining
#&gt; left_join: added no columns
#&gt;            &gt; rows only in x    0
#&gt;            &gt; rows only in y  ( 0)
#&gt;            &gt; matched rows     90
#&gt;            &gt;                 ====
#&gt;            &gt; rows total       90

saveRDS(gsea_res_top, here(output_path, &quot;gsea_res.rds&quot;))

# individual gsea plot
make_gsea_plot(
  signature = signature_df %&gt;%
    rownames_to_column(&quot;gene&quot;) %&gt;%
    select(gene, t = `pure_ccl_12m_vs_0m`) %&gt;%
    drop_na(),
  geneset = genesets_top %&gt;%
    filter(geneset == &quot;diehl-nafld-advanced_vs_mild|up&quot;),
  gene_level_stat = t
)
#&gt; select: renamed one variable (t) and dropped 2 variables
#&gt; drop_na: no rows removed
#&gt; rename: renamed one variable (stat)
#&gt; transmute: changed 13,423 values (100%) of &#39;stat&#39; (0 new NA)
#&gt;            new variable &#39;rank&#39; (integer) with 13,423 unique values and 0% NA
#&gt; select: dropped one variable (rank)
#&gt; filter: removed 14,500 rows (97%), 500 rows remaining
#&gt; inner_join: added one column (geneset)
#&gt;             &gt; rows only in x  (12,990)
#&gt;             &gt; rows only in y  (    67)
#&gt;             &gt; matched rows        433
#&gt;             &gt;                 ========
#&gt;             &gt; rows total          433
#&gt; top_n: removed 867 rows (&gt;99%), one row remaining</code></pre>
<p><img src="figure/meta-mouse-vs-human.Rmd/enrichment-hs-gs-mm-sig-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-enrichment-hs-gs-mm-sig-1">
Past versions of enrichment-hs-gs-mm-sig-1.png
</button>
</p>
<div id="fig-enrichment-hs-gs-mm-sig-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/45a33e4bf9976b61178a021e1c8f7a32ac72697e/docs/figure/meta-mouse-vs-human.Rmd/enrichment-hs-gs-mm-sig-1.png" target="_blank">45a33e4</a>
</td>
<td>
christianholland
</td>
<td>
2021-01-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/d7ef743c73acc1d9fb82d5d3c390b4aa7c8d5eb9/docs/figure/meta-mouse-vs-human.Rmd/enrichment-hs-gs-mm-sig-1.png" target="_blank">d7ef743</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>
# summary gsea plot
gsea_res_top %&gt;%
  mutate(label = stars.pval(padj)) %&gt;%
  ggplot(aes(x = signature, y = geneset, fill = ES)) +
  geom_tile() +
  geom_text(aes(label = label), size = fz / (14 / 5), vjust = 1) +
  facet_wrap(~direction, ncol = 2) +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.line = element_blank(),
    axis.ticks = element_blank()
  ) +
  scale_fill_gradient2(
    low = aachen_color(&quot;blue&quot;), mid = &quot;white&quot;,
    high = aachen_color(&quot;red&quot;)
  ) +
  my_theme(grid = &quot;no&quot;, fsize = fz) +
  labs(x = &quot;Signature&quot;, y = &quot;Gene Set&quot;, fill = &quot;ES&quot;) +
  guides(fill = guide_colorbar(title = &quot;ES&quot;))</code></pre>
<p><img src="figure/meta-mouse-vs-human.Rmd/enrichment-hs-gs-mm-sig-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-enrichment-hs-gs-mm-sig-2">
Past versions of enrichment-hs-gs-mm-sig-2.png
</button>
</p>
<div id="fig-enrichment-hs-gs-mm-sig-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/45a33e4bf9976b61178a021e1c8f7a32ac72697e/docs/figure/meta-mouse-vs-human.Rmd/enrichment-hs-gs-mm-sig-2.png" target="_blank">45a33e4</a>
</td>
<td>
christianholland
</td>
<td>
2021-01-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="leading-edge-extraction" class="section level2">
<h2>Leading edge extraction</h2>
<p>The enrichment of human genes sets in mouse signatures reveals that there is a set of genes which is significantly consistently deregulated in mouse and human. To identify these genes we extract the leading edge genes from the enrichment analysis.</p>
<pre class="r"><code>gsea_res &lt;- readRDS(here(output_path, &quot;gsea_res.rds&quot;))

# extract leading edges from significant and correctly directed enrichments
leading_edges &lt;- gsea_res %&gt;%
  filter(padj &lt;= 0.05 &amp;
    (direction == &quot;up&quot; &amp; ES &gt;= 0) | (direction == &quot;down&quot; &amp; ES &lt; 0)) %&gt;%
  unnest(leadingEdge) %&gt;%
  rename(gene = leadingEdge)
#&gt; filter: removed 12 rows (13%), 78 rows remaining
#&gt; rename: renamed one variable (gene)

saveRDS(leading_edges, here(output_path, &quot;individual_le.rds&quot;))

# for each study a union of leading edges is build across all contrast per time
# point
# subsequently we count how often a gene appears per time and direction
# (max 5 times because we have 5 studies in total)
# filter for those leading edges that appear in at least three studies
unified_le &lt;- leading_edges %&gt;%
  distinct(signature, direction, time, source, phenotype, gene) %&gt;%
  count(signature, time, gene, direction, sort = T, name = &quot;n_studies&quot;) %&gt;%
  filter(n_studies &gt;= 3)
#&gt; distinct: removed 3,923 rows (32%), 8,316 rows remaining
#&gt; count: now 6,080 rows and 5 columns, ungrouped
#&gt; filter: removed 5,632 rows (93%), 448 rows remaining

# translate to mgi genes for later use
unified_le_mgi &lt;- unified_le %&gt;%
  translate_gene_ids(from = &quot;symbol_hgnc&quot;, to = &quot;symbol_mgi&quot;) %&gt;%
  distinct(direction, gene, signature) %&gt;%
  # remove predicted genes
  filter(!str_detect(gene, &quot;Gm[0-9]+&quot;))
#&gt; select: dropped 6 variables (ensembl_mgi, ensembl_v_mgi, entrez_mgi, ensembl_hgnc, ensembl_v_hgnc, …)
#&gt; drop_na: removed 1,905 rows (6%), 30,461 rows remaining
#&gt; rename: renamed one variable (symbol_hgnc)
#&gt; left_join: added one column (symbol_mgi)
#&gt;            &gt; rows only in x        0
#&gt;            &gt; rows only in y  (30,218)
#&gt;            &gt; matched rows        473    (includes duplicates)
#&gt;            &gt;                 ========
#&gt;            &gt; rows total          473
#&gt; select: renamed one variable (gene) and dropped one variable
#&gt; distinct: removed 6 rows (1%), 467 rows remaining
#&gt; filter: removed 4 rows (1%), 463 rows remaining

# overlap of unified and consistent leading edge genes per time point
v &lt;- unified_le %&gt;%
  rename(regulation = direction) %&gt;%
  mutate(class = str_c(&quot;Month &quot;, time)) %&gt;%
  group_split(class) %&gt;%
  plot_venn_diagram()
#&gt; rename: renamed one variable (regulation)
#&gt; distinct: removed 157 rows (99%), one row remaining
#&gt; distinct: removed 124 rows (99%), one row remaining
#&gt; distinct: removed 164 rows (99%), one row remaining
#&gt; count: now 2 rows and 2 columns, ungrouped
#&gt; count: now 2 rows and 2 columns, ungrouped
#&gt; count: now 2 rows and 2 columns, ungrouped
#&gt; filter: removed one row (50%), one row remaining
#&gt; filter: removed one row (50%), one row remaining
#&gt; filter: removed one row (50%), one row remaining
#&gt; filter: removed 73 rows (46%), 85 rows remaining
#&gt; filter: removed 48 rows (38%), 77 rows remaining
#&gt; filter: removed 48 rows (38%), 77 rows remaining
#&gt; filter: removed 68 rows (41%), 97 rows remaining
#&gt; filter: removed 73 rows (46%), 85 rows remaining
#&gt; filter: removed 68 rows (41%), 97 rows remaining
#&gt; filter: removed 73 rows (46%), 85 rows remaining
#&gt; filter: removed 48 rows (38%), 77 rows remaining
#&gt; filter: removed 68 rows (41%), 97 rows remaining</code></pre>
<p><img src="figure/meta-mouse-vs-human.Rmd/leading-edge-extraction-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-leading-edge-extraction-1">
Past versions of leading-edge-extraction-1.png
</button>
</p>
<div id="fig-leading-edge-extraction-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/d7ef743c73acc1d9fb82d5d3c390b4aa7c8d5eb9/docs/figure/meta-mouse-vs-human.Rmd/leading-edge-extraction-1.png" target="_blank">d7ef743</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>#&gt; filter: removed one row (50%), one row remaining
#&gt; filter: removed one row (50%), one row remaining
#&gt; filter: removed one row (50%), one row remaining
#&gt; filter: removed 85 rows (54%), 73 rows remaining
#&gt; filter: removed 77 rows (62%), 48 rows remaining
#&gt; filter: removed 77 rows (62%), 48 rows remaining
#&gt; filter: removed 97 rows (59%), 68 rows remaining
#&gt; filter: removed 85 rows (54%), 73 rows remaining
#&gt; filter: removed 97 rows (59%), 68 rows remaining
#&gt; filter: removed 85 rows (54%), 73 rows remaining
#&gt; filter: removed 77 rows (62%), 48 rows remaining
#&gt; filter: removed 97 rows (59%), 68 rows remaining</code></pre>
<p><img src="figure/meta-mouse-vs-human.Rmd/leading-edge-extraction-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-leading-edge-extraction-2">
Past versions of leading-edge-extraction-2.png
</button>
</p>
<div id="fig-leading-edge-extraction-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/d7ef743c73acc1d9fb82d5d3c390b4aa7c8d5eb9/docs/figure/meta-mouse-vs-human.Rmd/leading-edge-extraction-2.png" target="_blank">d7ef743</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>
saveRDS(unified_le, here(output_path, &quot;leading_edges.rds&quot;))
saveRDS(unified_le_mgi, here(output_path, &quot;leading_edges_mgi.rds&quot;))</code></pre>
<div id="heatmap-of-leading-edge-genes" class="section level3">
<h3>Heatmap of leading edge genes</h3>
<pre class="r"><code># load leading edge genes and human and mouse contrasts
le &lt;- readRDS(here(output_path, &quot;leading_edges.rds&quot;))
contrasts &lt;- readRDS(here(output_path, &quot;limma_result.rds&quot;))
chronic_mouse &lt;- readRDS(
  here(&quot;output/mouse-chronic-ccl4/limma_result_hs.rds&quot;)
) %&gt;%
  filter(contrast_reference == &quot;pure_ccl4&quot;)
#&gt; filter: removed 120,807 rows (75%), 40,269 rows remaining


# filter mouse and human genes for leading edge genes
c &lt;- chronic_mouse %&gt;%
  inner_join(le, by = &quot;gene&quot;) %&gt;%
  mutate(class = &quot;chronic&quot;) %&gt;%
  select(gene, contrast, logFC, class)
#&gt; inner_join: added 4 columns (signature, time, direction, n_studies)
#&gt;             &gt; rows only in x  (39,585)
#&gt;             &gt; rows only in y  (     0)
#&gt;             &gt; matched rows      1,344    (includes duplicates)
#&gt;             &gt;                 ========
#&gt;             &gt; rows total        1,344
#&gt; select: dropped 9 variables (statistic, pval, fdr, regulation, contrast_reference, …)

h &lt;- contrasts %&gt;%
  inner_join(le, by = &quot;gene&quot;) %&gt;%
  mutate(class = &quot;human&quot;) %&gt;%
  unite(contrast, source, phenotype, contrast, sep = &quot;-&quot;) %&gt;%
  select(gene, contrast, logFC, class)
#&gt; inner_join: added 4 columns (signature, time, direction, n_studies)
#&gt;             &gt; rows only in x  (254,285)
#&gt;             &gt; rows only in y  (      0)
#&gt;             &gt; matched rows       6,681    (includes duplicates)
#&gt;             &gt;                 =========
#&gt;             &gt; rows total         6,681
#&gt; select: dropped 8 variables (statistic, pval, fdr, regulation, signature, …)

df &lt;- bind_rows(c, h) %&gt;%
  mutate(contrast = as_factor(contrast)) %&gt;%
  distinct()
#&gt; distinct: removed 3,938 rows (49%), 4,087 rows remaining

# assign a rank for each gene based on absolute mean logfc
df_ranked &lt;- df %&gt;%
  group_by(gene) %&gt;%
  summarise(mean_logfc = mean(logFC)) %&gt;%
  transmute(gene, rank = row_number(-abs(mean_logfc))) %&gt;%
  inner_join(df, by = &quot;gene&quot;)
#&gt; summarise: now 228 rows and 2 columns, ungrouped
#&gt; transmute: dropped one variable (mean_logfc)
#&gt;            new variable &#39;rank&#39; (integer) with 228 unique values and 0% NA
#&gt; inner_join: added 3 columns (contrast, logFC, class)
#&gt;             &gt; rows only in x  (    0)
#&gt;             &gt; rows only in y  (    0)
#&gt;             &gt; matched rows     4,087    (includes duplicates)
#&gt;             &gt;                 =======
#&gt;             &gt; rows total       4,087

# get degs from single cell data
# see &quot;code/meta-mouse-vs-human/sc_deg.R&quot; to see how this object was created.
sc_degs &lt;- readRDS(here(data_path, &quot;single_cell_degs.rds&quot;)) %&gt;%
  enframe(&quot;celltype&quot;) %&gt;%
  unnest(value) %&gt;%
  filter(p_val_adj &lt;= 0.05) %&gt;%
  mutate(adjusted_logfc = case_when(
    avg_logFC &gt;= 0 &amp; cluster == &quot;Uninjured&quot; ~ -avg_logFC,
    avg_logFC &lt; 0 &amp; cluster == &quot;Uninjured&quot; ~ -avg_logFC,
    TRUE ~ avg_logFC
  )) %&gt;%
  filter(abs(avg_logFC) &gt;= 0.25)
#&gt; filter: removed 19,662 rows (38%), 31,422 rows remaining
#&gt; filter: removed 26,682 rows (85%), 4,740 rows remaining

# combine consistent genes with cell types
df_ranked_ct &lt;- df_ranked %&gt;%
  left_join(sc_degs, by = &quot;gene&quot;) %&gt;%
  mutate(celltype = fct_explicit_na(celltype, na_level = &quot;Unknown&quot;))
#&gt; left_join: added 8 columns (celltype, p_val, avg_logFC, pct.1, pct.2, …)
#&gt;            &gt; rows only in x   3,187
#&gt;            &gt; rows only in y  (4,616)
#&gt;            &gt; matched rows     2,232    (includes duplicates)
#&gt;            &gt;                 =======
#&gt;            &gt; rows total       5,419

saveRDS(df_ranked_ct, here(output_path, &quot;consistent_genes.rds&quot;))

mat &lt;- df_ranked_ct %&gt;%
  filter(rank &lt;= 100) %&gt;%
  distinct(gene, contrast, logFC) %&gt;%
  untdy(feature = &quot;gene&quot;, key = &quot;contrast&quot;, value = &quot;logFC&quot;) %&gt;%
  as.matrix()
#&gt; filter: removed 2,675 rows (49%), 2,744 rows remaining
#&gt; distinct: removed 954 rows (35%), 1,790 rows remaining
#&gt; select: no changes
#&gt; spread: reorganized (contrast, logFC) into (pure_ccl_2m_vs_0m, pure_ccl_6m_vs_0m, pure_ccl_12m_vs_0m, diehl-nafld-advanced_vs_mild, ramnath-fibrosis-hcv_adv_vs_early, …) [was 1790x3, now 100x19]

ComplexHeatmap::Heatmap(t(as.matrix(mat)),
  col = col_fun,
  cluster_rows = F,
  cluster_columns = T,
  row_names_gp = gpar(fontsize = fz), column_names_gp = gpar(fontsize = fz - 4),
  name = &quot;logFC&quot;,
  row_gap = unit(2.5, &quot;mm&quot;),
  border = T,
  row_split = c(rep(&quot;Mouse&quot;, 3), rep(&quot;Human&quot;, 15))
)</code></pre>
<p><img src="figure/meta-mouse-vs-human.Rmd/le-visualization-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-le-visualization-1">
Past versions of le-visualization-1.png
</button>
</p>
<div id="fig-le-visualization-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/d7ef743c73acc1d9fb82d5d3c390b4aa7c8d5eb9/docs/figure/meta-mouse-vs-human.Rmd/le-visualization-1.png" target="_blank">d7ef743</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="characterization-of-leading-edge-genes" class="section level3">
<h3>Characterization of leading edge genes</h3>
<p>Up and down-regulated leading edge genes are characterized with GO terms, <a href="http://saezlab.github.io/progeny/">PROGENy’s</a> pathways and <a href="http://saezlab.github.io/dorothea/">DoRothEA’s</a> TFs. As statistic over-representation analysis is used.</p>
<pre class="r"><code>signatures &lt;- readRDS(here(output_path, &quot;leading_edges.rds&quot;)) %&gt;%
  distinct(gene, regulation = direction)
#&gt; distinct: removed 220 rows (49%), 228 rows remaining

genesets &lt;- load_genesets(organism = &quot;human&quot;) %&gt;%
  filter(confidence %in% c(NA, &quot;A&quot;, &quot;B&quot;, &quot;C&quot;))
#&gt; filter: removed 2,643,412 rows (80%), 662,851 rows remaining
#&gt; select: renamed one variable (gene) and dropped 2 variables
#&gt; gather: reorganized (Androgen, EGFR, Estrogen, Hypoxia, JAK-STAT, …) into (geneset, weight) [was 1301x15, now 18214x3]
#&gt; filter: removed 16,814 rows (92%), 1,400 rows remaining
#&gt; select: dropped one variable (weight)
#&gt; select: renamed 2 variables (geneset, gene) and dropped one variable
#&gt; filter: removed 472,718 rows (41%), 678,284 rows remaining

ora_res &lt;- signatures %&gt;%
  nest(sig = -c(regulation)) %&gt;%
  dplyr::mutate(ora = sig %&gt;% map(run_ora,
    sets = genesets, min_size = 10,
    options = list(alternative = &quot;greater&quot;),
    background_n = 20000
  )) %&gt;%
  select(-sig) %&gt;%
  unnest(ora)
#&gt; add_count: new variable &#39;n&#39; (integer) with 671 unique values and 0% NA
#&gt; filter: removed 13,277 rows (2%), 665,007 rows remaining
#&gt; select: dropped one variable (n)
#&gt; ungroup: no grouping variables
#&gt; select: dropped 4 variables (set, conf.low, conf.high, method)
#&gt; add_count: new variable &#39;n&#39; (integer) with 671 unique values and 0% NA
#&gt; filter: removed 13,277 rows (2%), 665,007 rows remaining
#&gt; select: dropped one variable (n)
#&gt; ungroup: no grouping variables
#&gt; select: dropped 4 variables (set, conf.low, conf.high, method)
#&gt; select: dropped one variable (sig)

saveRDS(ora_res, here(output_path, &quot;leading_edges_characterization.rds&quot;))</code></pre>
</div>
</div>
<div id="go-downstream-analysis" class="section level2">
<h2>GO downstream analysis</h2>
<p>In this section significant GO terms are summarized in two different ways. i) Which words appear the most among the GO terms and ii) Position of members of manually created GO cluster in a ranked list of significant GO terms (based on p-value)</p>
<div id="text-analysis" class="section level3">
<h3>Text analysis</h3>
<p>GO terms are splitted into words and their frequency is counted.</p>
<pre class="r"><code># list of words that will be ignored
stop_go_words &lt;- readRDS(here(&quot;data/annotation/stop_go_words.rds&quot;))

go_terms &lt;- readRDS(here(output_path, &quot;leading_edges_characterization.rds&quot;)) %&gt;%
  filter(group == &quot;go&quot; &amp; fdr &lt;= 0.05) %&gt;%
  mutate(
    term = str_remove(geneset, &quot;GO_&quot;),
    term = str_replace_all(term, &quot;_&quot;, &quot; &quot;),
    term = str_to_lower(term)
  ) %&gt;%
  select(regulation, term)
#&gt; filter: removed 4,026 rows (90%), 441 rows remaining
#&gt; select: dropped 8 variables (geneset, group, confidence, contingency_table, estimate, …)

go_wordcounts &lt;- go_terms %&gt;%
  # glue words that should be treated as one
  mutate(
    term = str_replace(term, &quot;smooth muscle&quot;, &quot;smoothmuscle&quot;),
    term = str_replace(term, &quot;amino acid&quot;, &quot;aminoacid&quot;),
    term = str_replace(term, &quot;cell cycle&quot;, &quot;cellcycle&quot;),
    term = str_replace(term, &quot;endoplasmic reticulum stress&quot;, &quot;endoplasmicreticulumstress&quot;),
    term = str_replace(term, &quot;endoplasmic reticulum&quot;, &quot;endoplasmicreticulum&quot;)
  ) %&gt;%
  unnest_tokens(word, term) %&gt;%
  anti_join(stop_words, by = &quot;word&quot;) %&gt;%
  # correct for abbreviations
  mutate(word = case_when(
    word == &quot;er&quot; ~ &quot;endoplasmicreticulum&quot;,
    TRUE ~ word
  )) %&gt;%
  # remove words which are pure numbers
  filter(!str_detect(word, &quot;^[0-9]+&quot;)) %&gt;%
  # count word frequency
  count(regulation, word, sort = T) %&gt;%
  # dissect prior glued words
  mutate(word = case_when(
    word == &quot;smoothmuscle&quot; ~ &quot;smooth-muscle&quot;,
    word == &quot;aminoacid&quot; ~ &quot;amino-acid&quot;,
    word == &quot;cellcycle&quot; ~ &quot;cell-cycle&quot;,
    word == &quot;endoplasmicreticulumstress&quot; ~ &quot;endoplasmic-reticulum-stress&quot;,
    word == &quot;endoplasmicreticulum&quot; ~ &quot;endoplasmic-reticulum&quot;,
    TRUE ~ word
  )) %&gt;%
  # remove meaning less words
  anti_join(stop_go_words, by = &quot;word&quot;)
#&gt; anti_join: added no columns
#&gt;            &gt; rows only in x   1,527
#&gt;            &gt; rows only in y  (1,101)
#&gt;            &gt; matched rows    (  283)
#&gt;            &gt;                 =======
#&gt;            &gt; rows total       1,527
#&gt; filter: removed 2 rows (&lt;1%), 1,525 rows remaining
#&gt; count: now 353 rows and 3 columns, ungrouped
#&gt; anti_join: added no columns
#&gt;            &gt; rows only in x   309
#&gt;            &gt; rows only in y  (  8)
#&gt;            &gt; matched rows    ( 44)
#&gt;            &gt;                 =====
#&gt;            &gt; rows total       309

saveRDS(go_wordcounts, here(output_path, &quot;go_wordcounts.rds&quot;))

go_wordcounts %&gt;%
  filter((regulation == &quot;up&quot; &amp; n &gt; 4) | (regulation == &quot;down&quot; &amp; n &gt; 2)) %&gt;%
  ggplot(aes(label = word, size = n)) +
  geom_text_wordcloud() +
  scale_size_area(max_size = fz / (14 / 5)) +
  facet_wrap(~regulation) +
  theme(axis.line = element_blank()) +
  my_theme(grid = &quot;no&quot;, fsize = fz)
#&gt; filter: removed 268 rows (87%), 41 rows remaining</code></pre>
<p><img src="figure/meta-mouse-vs-human.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/f8d83433873c01bdc55750a44f5e3a14f0a6c338/docs/figure/meta-mouse-vs-human.Rmd/unnamed-chunk-1-1.png" target="_blank">f8d8343</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/d7ef743c73acc1d9fb82d5d3c390b4aa7c8d5eb9/docs/figure/meta-mouse-vs-human.Rmd/unnamed-chunk-1-1.png" target="_blank">d7ef743</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-ranking" class="section level3">
<h3>Cluster ranking</h3>
<p>Distribution of manually created GO-cluster.</p>
<pre class="r"><code>go_cluster_mapping &lt;- tribble(
  ~cluster, ~description,
  1, &quot;Migration&quot;,
  2, &quot;Development and Morphogenesis&quot;,
  3, &quot;Metabolism&quot;
)

go_cluster &lt;- read_excel(here(data_path, &quot;manual_go_cluster_anno.xlsx&quot;)) %&gt;%
  distinct(regulation, term, cluster) %&gt;%
  drop_na() %&gt;%
  inner_join(go_cluster_mapping)
#&gt; distinct: no rows removed
#&gt; drop_na: removed 210 rows (52%), 192 rows remaining
#&gt; inner_join: added one column (description)
#&gt;             &gt; rows only in x  (  0)
#&gt;             &gt; rows only in y  (  0)
#&gt;             &gt; matched rows     192
#&gt;             &gt;                 =====
#&gt;             &gt; rows total       192

go_terms_ranking &lt;- readRDS(
  here(output_path, &quot;leading_edges_characterization.rds&quot;)
) %&gt;%
  mutate(
    term = str_remove(geneset, &quot;GO_&quot;),
    term = str_replace_all(term, &quot;_&quot;, &quot; &quot;),
    term = str_to_lower(term)
  ) %&gt;%
  filter(fdr &lt;= 0.05) %&gt;%
  group_by(regulation) %&gt;%
  mutate(rank = row_number(p.value)) %&gt;%
  mutate(max_rank = max(rank)) %&gt;%
  ungroup() %&gt;%
  select(regulation, term, rank, max_rank)
#&gt; filter: removed 3,997 rows (89%), 470 rows remaining
#&gt; ungroup: no grouping variables
#&gt; select: dropped 8 variables (geneset, group, confidence, contingency_table, estimate, …)

go_cluster_ranking &lt;- go_terms_ranking %&gt;%
  inner_join(go_cluster)
#&gt; inner_join: added 2 columns (cluster, description)
#&gt;             &gt; rows only in x  (292)
#&gt;             &gt; rows only in y  ( 14)
#&gt;             &gt; matched rows     178
#&gt;             &gt;                 =====
#&gt;             &gt; rows total       178

saveRDS(go_cluster_ranking, here(output_path, &quot;go_cluster_ranking.rds&quot;))

go_cluster_ranking %&gt;%
  mutate(regulation = as_factor(regulation)) %&gt;%
  plot_go_rank_density() +
  my_theme(grid = &quot;no&quot;, fsize = fz)</code></pre>
<p><img src="figure/meta-mouse-vs-human.Rmd/go-cluster-ranking-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-go-cluster-ranking-1">
Past versions of go-cluster-ranking-1.png
</button>
</p>
<div id="fig-go-cluster-ranking-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/f8d83433873c01bdc55750a44f5e3a14f0a6c338/docs/figure/meta-mouse-vs-human.Rmd/go-cluster-ranking-1.png" target="_blank">f8d8343</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="integration-with-chronicity-categories" class="section level2">
<h2>Integration with chronicity categories</h2>
<div id="pr-of-chronicity-and-etiologies" class="section level3">
<h3>PR of chronicity and etiologies</h3>
<pre class="r"><code># load etiology gene sets
etiology_genesets = readRDS(here(output_path, &quot;etiology_gene_sets.rds&quot;)) %&gt;%
  mutate(join_var = &quot;xy&quot;) %&gt;%
  nest(etiology_genes = -c(etiology, regulation, join_var))

# load exclusive chronic and acute and common genes
chronic = readRDS(here(&quot;output/meta-chronic-vs-acute/ranked_exclusive_chronic_genes.rds&quot;)) %&gt;%
  mutate(class = &quot;chronic&quot;) %&gt;%
  select(gene, rank, class, statistic = chronic_statistic)
#&gt; select: renamed one variable (statistic) and dropped 4 variables

acute = readRDS(here(&quot;output/meta-chronic-vs-acute/ranked_exclusive_acute_genes.rds&quot;)) %&gt;%
  mutate(class = &quot;acute&quot;) %&gt;%
  select(gene, rank, class, statistic = acute_statistic)
#&gt; select: renamed one variable (statistic) and dropped 4 variables

common = readRDS(here(&quot;output/meta-chronic-vs-acute/ranked_common_genes.rds&quot;)) %&gt;%
  mutate(class = &quot;common&quot;) %&gt;%
  select(gene, rank, class, statistic = chronic_statistic)
#&gt; select: renamed one variable (statistic) and dropped 4 variables

chronicity = bind_rows(chronic, acute, common) %&gt;%
  mutate(regulation = if_else(sign(statistic) &gt;= 0, &quot;up&quot;, &quot;down&quot;))

chronicity_hgnc = chronicity %&gt;%
  translate_gene_ids(from = &quot;symbol_mgi&quot;, to = &quot;symbol_hgnc&quot;, na_rm = TRUE) %&gt;%
  distinct() %&gt;%
  # drop genes that appear due to translation now in multiple classes
  group_by(gene) %&gt;%
  filter(n() == 1) %&gt;%
  ungroup()
#&gt; select: dropped 6 variables (ensembl_mgi, ensembl_v_mgi, entrez_mgi, ensembl_hgnc, ensembl_v_hgnc, …)
#&gt; drop_na: removed 1,905 rows (6%), 30,461 rows remaining
#&gt; rename: renamed one variable (symbol_mgi)
#&gt; left_join: added one column (symbol_hgnc)
#&gt;            &gt; rows only in x      168
#&gt;            &gt; rows only in y  (26,671)
#&gt;            &gt; matched rows      3,790    (includes duplicates)
#&gt;            &gt;                 ========
#&gt;            &gt; rows total        3,958
#&gt; select: renamed one variable (gene) and dropped one variable
#&gt; drop_na: removed 168 rows (4%), 3,790 rows remaining
#&gt; distinct: removed 37 rows (1%), 3,753 rows remaining
#&gt; filter (grouped): removed 173 rows (5%), 3,580 rows remaining
#&gt; ungroup: no grouping variables

# number of genes per regulated chronicity category
chronicity_hgnc %&gt;%
  count(class, regulation) %&gt;%
  ggplot(aes(x=class, y=n, fill = regulation)) +
  geom_col(position = &quot;dodge&quot;) +
  geom_hline(yintercept = 120)
#&gt; count: now 6 rows and 3 columns, ungrouped</code></pre>
<p><img src="figure/meta-mouse-vs-human.Rmd/pr-chronicity-etiology-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pr-chronicity-etiology-1">
Past versions of pr-chronicity-etiology-1.png
</button>
</p>
<div id="fig-pr-chronicity-etiology-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/c6bfb3b22a5734c0b4669c86bf8e6829e5e79125/docs/figure/meta-mouse-vs-human.Rmd/pr-chronicity-etiology-1.png" target="_blank">c6bfb3b</a>
</td>
<td>
christianholland
</td>
<td>
2021-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>
setup = chronicity_hgnc %&gt;%
  group_by(class, regulation) %&gt;%
  slice_max(order_by = rank, n = 120, with_ties = FALSE) %&gt;%
  mutate(join_var = &quot;xy&quot;) %&gt;%
  nest(mm_genes = -c(class, regulation, join_var)) %&gt;%
  ungroup() %&gt;%
  inner_join(etiology_genesets, by=c(&quot;regulation&quot;, &quot;join_var&quot;)) %&gt;%
  select(-join_var)
#&gt; slice_max (grouped): removed 2,860 rows (80%), 720 rows remaining
#&gt; ungroup: no grouping variables
#&gt; inner_join: added 2 columns (etiology, etiology_genes)
#&gt;             &gt; rows only in x  ( 0)
#&gt;             &gt; rows only in y  ( 0)
#&gt;             &gt; matched rows     30    (includes duplicates)
#&gt;             &gt;                 ====
#&gt;             &gt; rows total       30
#&gt; select: dropped one variable (join_var)

pr = setup %&gt;%
  mutate(pr = map2(etiology_genes, mm_genes, function(hs_genes, mm_genes) {
    z &lt;- inner_join(
      select(mm_genes, gene), 
      select(hs_genes, gene), by = &quot;gene&quot;)

    tibble(
      recall = nrow(z) / nrow(hs_genes),
      recall_ratio = glue(&quot;{nrow(z)}/{nrow(hs_genes)}&quot;),
      precision = nrow(z) / nrow(mm_genes),
      precision_ratio = glue(&quot;{nrow(z)}/{nrow(mm_genes)}&quot;)
    )}
  )) %&gt;%
  unnest(pr)
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (112)
#&gt;             &gt; rows only in y  (684)
#&gt;             &gt; matched rows       8
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         8
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (119)
#&gt;             &gt; rows only in y  (202)
#&gt;             &gt; matched rows       1
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         1
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (119)
#&gt;             &gt; rows only in y  (113)
#&gt;             &gt; matched rows       1
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         1
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (119)
#&gt;             &gt; rows only in y  ( 50)
#&gt;             &gt; matched rows       1
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         1
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (113)
#&gt;             &gt; rows only in y  (344)
#&gt;             &gt; matched rows       7
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         7
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (  113)
#&gt;             &gt; rows only in y  (1,330)
#&gt;             &gt; matched rows         7
#&gt;             &gt;                 =======
#&gt;             &gt; rows total           7
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (  107)
#&gt;             &gt; rows only in y  (1,121)
#&gt;             &gt; matched rows        13
#&gt;             &gt;                 =======
#&gt;             &gt; rows total          13
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (114)
#&gt;             &gt; rows only in y  (478)
#&gt;             &gt; matched rows       6
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         6
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (117)
#&gt;             &gt; rows only in y  (145)
#&gt;             &gt; matched rows       3
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         3
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (120)
#&gt;             &gt; rows only in y  (109)
#&gt;             &gt; matched rows       0
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         0
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (112)
#&gt;             &gt; rows only in y  (684)
#&gt;             &gt; matched rows       8
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         8
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (119)
#&gt;             &gt; rows only in y  (202)
#&gt;             &gt; matched rows       1
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         1
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (119)
#&gt;             &gt; rows only in y  (113)
#&gt;             &gt; matched rows       1
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         1
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (119)
#&gt;             &gt; rows only in y  ( 50)
#&gt;             &gt; matched rows       1
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         1
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (113)
#&gt;             &gt; rows only in y  (344)
#&gt;             &gt; matched rows       7
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         7
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (  102)
#&gt;             &gt; rows only in y  (1,319)
#&gt;             &gt; matched rows        18
#&gt;             &gt;                 =======
#&gt;             &gt; rows total          18
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (   96)
#&gt;             &gt; rows only in y  (1,110)
#&gt;             &gt; matched rows        24
#&gt;             &gt;                 =======
#&gt;             &gt; rows total          24
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (111)
#&gt;             &gt; rows only in y  (475)
#&gt;             &gt; matched rows       9
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         9
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (119)
#&gt;             &gt; rows only in y  (147)
#&gt;             &gt; matched rows       1
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         1
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (119)
#&gt;             &gt; rows only in y  (108)
#&gt;             &gt; matched rows       1
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         1
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 96)
#&gt;             &gt; rows only in y  (668)
#&gt;             &gt; matched rows      24
#&gt;             &gt;                 =====
#&gt;             &gt; rows total        24
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (114)
#&gt;             &gt; rows only in y  (197)
#&gt;             &gt; matched rows       6
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         6
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (120)
#&gt;             &gt; rows only in y  (114)
#&gt;             &gt; matched rows       0
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         0
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (117)
#&gt;             &gt; rows only in y  ( 48)
#&gt;             &gt; matched rows       3
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         3
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (105)
#&gt;             &gt; rows only in y  (336)
#&gt;             &gt; matched rows      15
#&gt;             &gt;                 =====
#&gt;             &gt; rows total        15
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (  109)
#&gt;             &gt; rows only in y  (1,326)
#&gt;             &gt; matched rows        11
#&gt;             &gt;                 =======
#&gt;             &gt; rows total          11
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (   84)
#&gt;             &gt; rows only in y  (1,098)
#&gt;             &gt; matched rows        36
#&gt;             &gt;                 =======
#&gt;             &gt; rows total          36
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (110)
#&gt;             &gt; rows only in y  (474)
#&gt;             &gt; matched rows      10
#&gt;             &gt;                 =====
#&gt;             &gt; rows total        10
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (116)
#&gt;             &gt; rows only in y  (144)
#&gt;             &gt; matched rows       4
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         4
#&gt; select: dropped 2 variables (rank, statistic)
#&gt; select: no changes
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (115)
#&gt;             &gt; rows only in y  (104)
#&gt;             &gt; matched rows       5
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         5

saveRDS(pr, here(output_path, &quot;precision_recall_chronicity.rds&quot;))

set.seed(123)
pr %&gt;%
  mutate(regulation = factor(regulation, levels = c(&quot;up&quot;, &quot;down&quot;))) %&gt;%
  ggplot(aes(x = recall, y = precision, label = class, 
             color = class)) +
  geom_jitter(width = 0.0015, height = 0) +
  facet_rep_grid(etiology ~ regulation) +
  geom_abline(lty = &quot;dashed&quot;) +
  expand_limits(x = 0, y = 0) +
  labs(x = &quot;Recall&quot;, y = &quot;Precision&quot;) +
  my_theme(fsize = fz) +
  scale_color_viridis_d()</code></pre>
<p><img src="figure/meta-mouse-vs-human.Rmd/pr-chronicity-etiology-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pr-chronicity-etiology-2">
Past versions of pr-chronicity-etiology-2.png
</button>
</p>
<div id="fig-pr-chronicity-etiology-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/c6bfb3b22a5734c0b4669c86bf8e6829e5e79125/docs/figure/meta-mouse-vs-human.Rmd/pr-chronicity-etiology-2.png" target="_blank">c6bfb3b</a>
</td>
<td>
christianholland
</td>
<td>
2021-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="integration-of-published-chronic-mouse-models" class="section level2">
<h2>Integration of published chronic mouse models</h2>
<p>In this section the newly generated and previously published chronic mouse models are analyzed in terms of how well they reflect different human etiologies. The differential expressed genes of the already published mouse models is accessed from <a href="https://doi.org/10.1053/j.gastro.2016.05.051">Teufel et al., 2016</a>, Table S2.</p>
<div id="load-teufel-genes" class="section level3">
<h3>Load Teufel genes</h3>
<p>Here the differentially expressed genes of the mouse models from the Teufel study are loaded.</p>
<pre class="r"><code>teufel_genes &lt;- read_excel(here(data_path, &quot;TableS2_Teufel_clean.xlsx&quot;),
  sheet = &quot;mouse&quot;
) %&gt;%
  rename(gene = ...1) %&gt;%
  pivot_longer(-gene) %&gt;%
  separate(name, into = c(&quot;study&quot;, &quot;key&quot;)) %&gt;%
  pivot_wider(names_from = key, values_from = value) %&gt;%
  assign_deg(fdr_cutoff = 0.05, effect_size_cutoff = log2(1.5), effect_size_id = logfc) %&gt;%
  mutate(study = str_to_upper(study)) %&gt;%
  mutate(study = factor(study, levels = c(
    &quot;HF12&quot;, &quot;HF18&quot;, &quot;STZ12&quot;, &quot;STZ18&quot;, &quot;MCD4&quot;,
    &quot;MCD8&quot;, &quot;PTEN&quot;, &quot;HF30&quot;, &quot;WTD&quot;
  )))
#&gt; rename: renamed one variable (gene)
#&gt; pivot_longer: reorganized (hf12_logfc, hf12_fdr, hf18_logfc, hf18_fdr, stz12_logfc, …) into (name, value) [was 1513x19, now 27234x3]
#&gt; pivot_wider: reorganized (key, value) into (logfc, fdr) [was 27234x4, now 13617x4]

# for late use mgi symbols are translated to hgnc
teufel_genes_hs &lt;- teufel_genes %&gt;%
  translate_gene_ids(from = &quot;symbol_mgi&quot;, to = &quot;symbol_hgnc&quot;) %&gt;%
  drop_na(gene) %&gt;%
  distinct() %&gt;%
  # in case of duplicated genes take the one with the higher logfc
  group_by(study, gene) %&gt;%
  slice_max(n = 1, order_by = abs(logfc)) %&gt;%
  ungroup()
#&gt; select: dropped 6 variables (ensembl_mgi, ensembl_v_mgi, entrez_mgi, ensembl_hgnc, ensembl_v_hgnc, …)
#&gt; drop_na: removed 1,905 rows (6%), 30,461 rows remaining
#&gt; rename: renamed one variable (symbol_mgi)
#&gt; left_join: added one column (symbol_hgnc)
#&gt;            &gt; rows only in x    1,539
#&gt;            &gt; rows only in y  (29,020)
#&gt;            &gt; matched rows     12,969    (includes duplicates)
#&gt;            &gt;                 ========
#&gt;            &gt; rows total       14,508
#&gt; select: renamed one variable (gene) and dropped one variable
#&gt; drop_na: removed 1,539 rows (11%), 12,969 rows remaining
#&gt; distinct: removed 171 rows (1%), 12,798 rows remaining
#&gt; slice_max (grouped): removed 222 rows (2%), 12,576 rows remaining
#&gt; ungroup: no grouping variables

saveRDS(teufel_genes, here(output_path, &quot;teufel_genes.rds&quot;))
saveRDS(teufel_genes_hs, here(output_path, &quot;teufel_genes_hs.rds&quot;))</code></pre>
</div>
<div id="chronic-mouse-models-in-numbers" class="section level3">
<h3>Chronic mouse models in numbers</h3>
<p>Number of differentially expressed genes per chronic mouse model.</p>
<pre class="r"><code>teufel_genes &lt;- readRDS(here(output_path, &quot;teufel_genes.rds&quot;)) %&gt;%
  assign_deg(effect_size_cutoff = log2(1.5), effect_size_id = logfc) %&gt;%
  filter(regulation != &quot;ns&quot;)
#&gt; filter: removed 11,046 rows (81%), 2,571 rows remaining
chronic &lt;- readRDS(here(&quot;output/mouse-chronic-ccl4/limma_result.rds&quot;)) %&gt;%
  filter(contrast_reference == &quot;pure_ccl4&quot;) %&gt;%
  assign_deg(effect_size_cutoff = log2(1.5)) %&gt;%
  filter(regulation != &quot;ns&quot;)
#&gt; filter: removed 138,042 rows (75%), 46,014 rows remaining
#&gt; filter: removed 40,017 rows (87%), 5,997 rows remaining

teufel_count &lt;- teufel_genes %&gt;%
  count(study, regulation)
#&gt; count: now 18 rows and 3 columns, ungrouped

chronic_count &lt;- chronic %&gt;%
  count(contrast, regulation) %&gt;%
  rename(study = contrast)
#&gt; count: now 6 rows and 3 columns, ungrouped
#&gt; rename: renamed one variable (study)

df &lt;- bind_rows(teufel_count, chronic_count)

saveRDS(df, here(output_path, &quot;chronic_mouse_deg_numbers.rds&quot;))

df %&gt;%
  ggplot(aes(y = study, x = n, fill = regulation)) +
  geom_col(position = &quot;dodge&quot;) +
  labs(x = &quot;Number of differential expressed genes&quot;, y = NULL) +
  scale_x_log10(
    breaks = scales::trans_breaks(&quot;log10&quot;, function(x) 10^x),
    labels = scales::trans_format(&quot;log10&quot;, scales::math_format(10^.x))
  ) +
  my_theme(grid = &quot;x&quot;, fsize = fz) +
  annotation_logticks(sides = &quot;b&quot;)</code></pre>
<p><img src="figure/meta-mouse-vs-human.Rmd/vis-chronic-mm-gene-numbers-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-vis-chronic-mm-gene-numbers-1">
Past versions of vis-chronic-mm-gene-numbers-1.png
</button>
</p>
<div id="fig-vis-chronic-mm-gene-numbers-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/bf6fcc69c8835ef80fc56aa386ae6deda2b44db6/docs/figure/meta-mouse-vs-human.Rmd/vis-chronic-mm-gene-numbers-1.png" target="_blank">bf6fcc6</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="similarity-of-chronic-mouse-models-and-patient-cohorts" class="section level3">
<h3>Similarity of chronic mouse models and patient cohorts</h3>
<pre class="r"><code>patient &lt;- readRDS(here(output_path, &quot;limma_result.rds&quot;)) %&gt;%
  select(gene, contrast, logFC, fdr, regulation, source, phenotype, contrast)
#&gt; select: dropped 2 variables (statistic, pval)

teufel &lt;- readRDS(here(output_path, &quot;teufel_genes_hs.rds&quot;)) %&gt;%
  rename(contrast = study, logFC = logfc)
#&gt; rename: renamed 2 variables (contrast, logFC)

chronic &lt;- readRDS(here(&quot;output/mouse-chronic-ccl4/limma_result_hs.rds&quot;)) %&gt;%
  filter(contrast_reference == &quot;pure_ccl4&quot;) %&gt;%
  select(-c(statistic, pval, contrast_reference))
#&gt; filter: removed 120,807 rows (75%), 40,269 rows remaining
#&gt; select: dropped 3 variables (statistic, pval, contrast_reference)

contrasts &lt;- bind_rows(patient, teufel, chronic) %&gt;%
  assign_deg(effect_size_cutoff = log2(1.5))

# populate gene sets with only significantly expressed genes
mat_sig &lt;- contrasts %&gt;%
  filter(regulation != &quot;ns&quot;) %&gt;%
  unite(geneset, source, phenotype, contrast, sep = &quot;-&quot;, na.rm = T) %&gt;%
  add_count(geneset, name = &quot;size&quot;) %&gt;%
  # exclude gene sets with a size &lt; 20
  filter(size &gt;= 20) %&gt;%
  group_by(geneset) %&gt;%
  mutate(key = row_number()) %&gt;%
  ungroup() %&gt;%
  mutate(geneset = as_factor(geneset)) %&gt;%
  select(geneset, gene, key) %&gt;%
  untdy(key, geneset, gene)
#&gt; filter: removed 296,977 rows (96%), 13,556 rows remaining
#&gt; add_count: new variable &#39;size&#39; (integer) with 26 unique values and 0% NA
#&gt; filter: removed one row (&lt;1%), 13,555 rows remaining
#&gt; ungroup: no grouping variables
#&gt; select: dropped 4 variables (logFC, fdr, regulation, size)
#&gt; select: columns reordered (key, geneset, gene)
#&gt; spread: reorganized (geneset, gene) into (diehl-nafld-advanced_vs_mild, ramnath-fibrosis-hcv_adv_vs_early, hoang-nafld-stage_2_vs_0, hoang-nafld-stage_3_vs_0, hoang-nafld-stage_4_vs_0, …) [was 13555x3, now 3640x26]

o &lt;- set_similarity(mat_sig, measure = &quot;overlap_coef&quot;, tidy = T)
#&gt; gather: reorganized (diehl-nafld-advanced_vs_mild, ramnath-fibrosis-hcv_adv_vs_early, hoang-nafld-stage_2_vs_0, hoang-nafld-stage_3_vs_0, hoang-nafld-stage_4_vs_0, …) into (set2, similarity) [was 25x26, now 625x3]
#&gt; drop_na: removed 300 rows (48%), 325 rows remaining
#&gt; filter: removed 25 rows (8%), 300 rows remaining
#&gt; mutate_if: converted &#39;set1&#39; from character to factor (0 new NA)
#&gt;            converted &#39;set2&#39; from character to factor (0 new NA)

saveRDS(o, here(output_path, &quot;cross_species_similarity.rds&quot;))

o %&gt;%
  ggplot(aes(x = set1, y = set2, fill = similarity)) +
  geom_tile() +
  scale_fill_gradient(low = &quot;white&quot;, high = aachen_color(&quot;green&quot;)) +
  labs(x = NULL, y = NULL, fill = &quot;Overlap\ncoefficient&quot;) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  my_theme(grid = &quot;no&quot;, fsize = fz)</code></pre>
<p><img src="figure/meta-mouse-vs-human.Rmd/similarity-chronic-mm-and-hs-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-similarity-chronic-mm-and-hs-1">
Past versions of similarity-chronic-mm-and-hs-1.png
</button>
</p>
<div id="fig-similarity-chronic-mm-and-hs-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/73708b19dedba527877512a3a2af452a702962be/docs/figure/meta-mouse-vs-human.Rmd/similarity-chronic-mm-and-hs-1.png" target="_blank">73708b1</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/beaeae06be81850d16b0c023633b0e640d5b47a4/docs/figure/meta-mouse-vs-human.Rmd/similarity-chronic-mm-and-hs-1.png" target="_blank">beaeae0</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="enrichment-of-chronic-mouse-genes-in-patient-cohorts" class="section level3">
<h3>Enrichment of chronic mouse genes in patient cohorts</h3>
<pre class="r"><code>patient &lt;- readRDS(here(output_path, &quot;limma_result.rds&quot;)) %&gt;%
  select(gene, contrast, logFC, fdr, regulation, source)
#&gt; select: dropped 3 variables (statistic, pval, phenotype)

teufel &lt;- readRDS(here(output_path, &quot;teufel_genes_hs.rds&quot;)) %&gt;%
  rename(contrast = study, logFC = logfc)
#&gt; rename: renamed 2 variables (contrast, logFC)

chronic &lt;- readRDS(here(&quot;output/mouse-chronic-ccl4/limma_result_hs.rds&quot;)) %&gt;%
  filter(contrast_reference == &quot;pure_ccl4&quot;) %&gt;%
  select(-c(statistic, pval, contrast_reference))
#&gt; filter: removed 120,807 rows (75%), 40,269 rows remaining
#&gt; select: dropped 3 variables (statistic, pval, contrast_reference)

# populate gene sets with only significantly expressed genes
genesets_sig &lt;- bind_rows(teufel, chronic) %&gt;%
  # adapt cutoff for the definition of significant genes
  assign_deg(effect_size_cutoff = log2(1.5)) %&gt;%
  filter(regulation != &quot;ns&quot;) %&gt;%
  add_count(contrast, name = &quot;size&quot;) %&gt;%
  # exclude gene sets with a size &lt; 10
  filter(size &gt;= 10) %&gt;%
  unite(geneset, contrast) %&gt;%
  unite(geneset, geneset, regulation, sep = &quot;|&quot;) %&gt;%
  mutate(geneset = as_factor(geneset)) %&gt;%
  select(geneset, gene)
#&gt; filter: removed 45,192 rows (86%), 7,653 rows remaining
#&gt; add_count: new variable &#39;size&#39; (integer) with 12 unique values and 0% NA
#&gt; filter: no rows removed
#&gt; select: dropped 3 variables (logFC, fdr, size)

# construct signature matrix/data frame
signature_df &lt;- patient %&gt;%
  unite(signature, source, contrast) %&gt;%
  mutate(signature = as_factor(signature)) %&gt;%
  untdy(&quot;gene&quot;, &quot;signature&quot;, &quot;logFC&quot;)
#&gt; select: dropped 2 variables (fdr, regulation)
#&gt; spread: reorganized (signature, logFC) into (diehl_advanced_vs_mild, ramnath_hcv_adv_vs_early, ramnath_nafld_adv_vs_early, hoang_stage_1_vs_0, hoang_stage_2_vs_0, …) [was 257688x3, now 24107x16]

# run gsea
set.seed(123)
gsea_res_sig &lt;- run_gsea(signature_df, genesets_sig, tidy = T, nperm = 10000) %&gt;%
  separate(geneset, into = c(&quot;geneset&quot;, &quot;direction&quot;), sep = &quot;[|]&quot;) %&gt;%
  mutate(
    signature = as_factor(signature),
    geneset = as_factor(geneset)
  )
#&gt; summarise: now 24 rows and 2 columns, ungrouped
#&gt; rename: renamed one variable (geneset)
#&gt; select: dropped one variable (gene)
#&gt; distinct: removed 7,629 rows (&gt;99%), 24 rows remaining
#&gt; left_join: added no columns
#&gt;            &gt; rows only in x     0
#&gt;            &gt; rows only in y  (  0)
#&gt;            &gt; matched rows     360
#&gt;            &gt;                 =====
#&gt;            &gt; rows total       360

saveRDS(gsea_res_sig, here(output_path, &quot;cross_species_enrichment.rds&quot;))

a &lt;- gsea_res_sig %&gt;%
  filter(direction == &quot;up&quot;) %&gt;%
  mutate(
    label = stars.pval(padj),
    direction = fct_rev(direction)
  ) %&gt;%
  ggplot(aes(x = signature, y = geneset, fill = ES)) +
  geom_tile() +
  geom_text(aes(label = label)) +
  facet_rep_wrap(~direction, scales = &quot;free&quot;, ncol = 1) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  scale_fill_gradient2() +
  my_theme(grid = &quot;no&quot;, fsize = fz)
#&gt; filter: removed 180 rows (50%), 180 rows remaining

b &lt;- gsea_res_sig %&gt;%
  mutate(direction = fct_rev(direction)) %&gt;%
  filter(direction == &quot;up&quot;) %&gt;%
  ggplot(aes(y = geneset, x = ES)) +
  geom_boxplot() +
  # geom_jitter(aes(color = signature)) +
  facet_rep_wrap(~direction, ncol = 1) +
  geom_vline(xintercept = 0) +
  my_theme(fsize = fz, grid = &quot;x&quot;)
#&gt; filter: removed 180 rows (50%), 180 rows remaining

a + theme(legend.position = &quot;none&quot;) +
  b + theme(
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    axis.line.y = element_blank(),
    axis.ticks.y = element_blank()
  )</code></pre>
<p><img src="figure/meta-mouse-vs-human.Rmd/enrichment-chronic-mm-in-hs-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-enrichment-chronic-mm-in-hs-1">
Past versions of enrichment-chronic-mm-in-hs-1.png
</button>
</p>
<div id="fig-enrichment-chronic-mm-in-hs-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/beaeae06be81850d16b0c023633b0e640d5b47a4/docs/figure/meta-mouse-vs-human.Rmd/enrichment-chronic-mm-in-hs-1.png" target="_blank">beaeae0</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="build-gene-pool-for-human-etiology" class="section level3">
<h3>Build gene pool for human etiology</h3>
<p>For each human etiology (e.g. NAFLD) a union of differentially expressed genes in constructed.</p>
<pre class="r"><code># get etiology for each human contrast
etiologies &lt;- readRDS(here(data_path, &quot;contrast_annotation.rds&quot;)) %&gt;%
  mutate(etiology = str_remove(disease, pattern = &quot; Stage [0-9]&quot;)) %&gt;%
  distinct(source, phenotype, contrast, etiology)
#&gt; distinct: no rows removed

contrasts &lt;- readRDS(here(output_path, &quot;limma_result.rds&quot;))

etiology_union &lt;- contrasts %&gt;%
  inner_join(etiologies, by = c(&quot;source&quot;, &quot;phenotype&quot;, &quot;contrast&quot;)) %&gt;%
  assign_deg(effect_size_cutoff = log2(1.5)) %&gt;%
  filter(regulation != &quot;ns&quot;) %&gt;%
  arrange(etiology, gene, regulation) %&gt;%
  # how often is a gene-regulation pair reported per etiology
  count(gene, regulation, etiology) %&gt;%
  group_by(gene, etiology) %&gt;%
  # extract this gene-regulation-etiology combination that is reported by more
  # contrasts
  # in case of ties take the first one (mostly up-regulation, first factor)
  slice_max(n, with_ties = FALSE) %&gt;%
  ungroup() %&gt;%
  select(-n)
#&gt; inner_join: added one column (etiology)
#&gt;             &gt; rows only in x  (      0)
#&gt;             &gt; rows only in y  (      0)
#&gt;             &gt; matched rows     257,688
#&gt;             &gt;                 =========
#&gt;             &gt; rows total       257,688
#&gt; filter: removed 251,785 rows (98%), 5,903 rows remaining
#&gt; count: now 4,627 rows and 4 columns, ungrouped
#&gt; slice_max (grouped): removed 4 rows (&lt;1%), 4,623 rows remaining
#&gt; ungroup: no grouping variables
#&gt; select: dropped one variable (n)

saveRDS(etiology_union, here(output_path, &quot;etiology_gene_sets.rds&quot;))</code></pre>
<div id="etiology-genes-in-numbers" class="section level4">
<h4>Etiology genes in numbers</h4>
<p>UpsetR plots showing the number of differential expressed genes per etiology and their overlap.</p>
<pre class="r"><code>df &lt;- readRDS(here(output_path, &quot;etiology_gene_sets.rds&quot;))

mat_up &lt;- df %&gt;%
  filter(regulation == &quot;up&quot;) %&gt;%
  select(-regulation) %&gt;%
  mutate(val = 1) %&gt;%
  spread(etiology, val, fill = 0) %&gt;%
  data.frame(row.names = 1)
#&gt; filter: removed 1,411 rows (31%), 3,212 rows remaining
#&gt; select: dropped one variable (regulation)
#&gt; spread: reorganized (etiology, val) into (HCV, NAFLD, NASH, PBC, PSC) [was 3212x3, now 2674x6]

grid.newpage()
upset(mat_up,
  nintersects = NA, mainbar.y.label = &quot;Common genes&quot;,
  sets.x.label = &quot;Total number of genes&quot;
)
grid.text(&quot;Up&quot;, x = 0.65, y = 0.95, gp = gpar(fontsize = fz))</code></pre>
<p><img src="figure/meta-mouse-vs-human.Rmd/vis-etiology-gene-numbers-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-vis-etiology-gene-numbers-1">
Past versions of vis-etiology-gene-numbers-1.png
</button>
</p>
<div id="fig-vis-etiology-gene-numbers-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/f8d83433873c01bdc55750a44f5e3a14f0a6c338/docs/figure/meta-mouse-vs-human.Rmd/vis-etiology-gene-numbers-1.png" target="_blank">f8d8343</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>
mat_down &lt;- df %&gt;%
  filter(regulation == &quot;down&quot;) %&gt;%
  select(-regulation) %&gt;%
  mutate(val = 1) %&gt;%
  spread(etiology, val, fill = 0) %&gt;%
  data.frame(row.names = 1)
#&gt; filter: removed 3,212 rows (69%), 1,411 rows remaining
#&gt; select: dropped one variable (regulation)
#&gt; spread: reorganized (etiology, val) into (HCV, NAFLD, NASH, PBC, PSC) [was 1411x3, now 1229x6]

grid.newpage()
upset(mat_down,
  nintersects = NA, mainbar.y.label = &quot;Common genes&quot;,
  sets.x.label = &quot;Total number of genes&quot;
)
grid.text(&quot;Down&quot;, x = 0.65, y = 0.95, gp = gpar(fontsize = fz))</code></pre>
<p><img src="figure/meta-mouse-vs-human.Rmd/vis-etiology-gene-numbers-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-vis-etiology-gene-numbers-2">
Past versions of vis-etiology-gene-numbers-2.png
</button>
</p>
<div id="fig-vis-etiology-gene-numbers-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/f8d83433873c01bdc55750a44f5e3a14f0a6c338/docs/figure/meta-mouse-vs-human.Rmd/vis-etiology-gene-numbers-2.png" target="_blank">f8d8343</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>

# overlap in percentages
tabyl(c(rowSums(mat_up), rowSums(mat_down)))
#&gt;  c(rowSums(mat_up), rowSums(mat_down))    n     percent
#&gt;                                      1 3314 0.849090443
#&gt;                                      2  476 0.121957469
#&gt;                                      3   95 0.024340251
#&gt;                                      4   18 0.004611837</code></pre>
</div>
</div>
<div id="precision-and-recall-of-mouse-models" class="section level3">
<h3>Precision and Recall of mouse models</h3>
<p>How well mouse models capture the genes that are deregulated in human is computed by precision and recall.</p>
<pre class="r"><code>etiology_pool &lt;- readRDS(here(output_path, &quot;etiology_gene_sets.rds&quot;)) %&gt;%
  filter(regulation != &quot;ns&quot;) %&gt;%
  mutate(key = &quot;key&quot;) %&gt;%
  nest(hs_genes = c(gene))
#&gt; filter: no rows removed

teufel_genes &lt;- readRDS(here(output_path, &quot;teufel_genes_hs.rds&quot;)) %&gt;%
  filter(regulation != &quot;ns&quot;) %&gt;%
  mutate(class = &quot;teufel&quot;) %&gt;%
  distinct(gene, study, regulation, class)
#&gt; filter: removed 10,223 rows (81%), 2,353 rows remaining
#&gt; distinct: no rows removed

chronic &lt;- readRDS(here(&quot;output/mouse-chronic-ccl4/limma_result_hs.rds&quot;)) %&gt;%
  filter(contrast_reference == &quot;pure_ccl4&quot;) %&gt;%
  select(gene, contrast,
    logfc = logFC, fdr, regulation,
    source = contrast_reference
  ) %&gt;%
  assign_deg(
    fdr_cutoff = 0.05, effect_size_cutoff = log2(1.5),
    effect_size_id = logfc
  ) %&gt;%
  filter(regulation != &quot;ns&quot;) %&gt;%
  mutate(class = &quot;chronic&quot;) %&gt;%
  distinct(gene, study = contrast, regulation, class)
#&gt; filter: removed 120,807 rows (75%), 40,269 rows remaining
#&gt; select: renamed 2 variables (logfc, source) and dropped 2 variables
#&gt; filter: removed 34,969 rows (87%), 5,300 rows remaining
#&gt; distinct: no rows removed


# merge gene pools of all chronic mouse models
mm &lt;- bind_rows(teufel_genes, chronic)

setup &lt;- mm %&gt;%
  mutate(key = &quot;key&quot;) %&gt;%
  # rename(direction = regulation) %&gt;%
  nest(mm_genes = c(gene)) %&gt;%
  inner_join(etiology_pool, by = c(&quot;key&quot;, &quot;regulation&quot;)) %&gt;%
  select(-key)
#&gt; inner_join: added 2 columns (etiology, hs_genes)
#&gt;             &gt; rows only in x  (  0)
#&gt;             &gt; rows only in y  (  0)
#&gt;             &gt; matched rows     120    (includes duplicates)
#&gt;             &gt;                 =====
#&gt;             &gt; rows total       120
#&gt; select: dropped one variable (key)

pr &lt;- setup %&gt;%
  mutate(metrics = pmap(., .f = function(mm_genes, hs_genes, ...) {
    z &lt;- inner_join(mm_genes, hs_genes, by = &quot;gene&quot;)

    tibble(
      recall = nrow(z) / nrow(hs_genes),
      recall_ratio = glue(&quot;{nrow(z)}/{nrow(hs_genes)}&quot;),
      precision = nrow(z) / nrow(mm_genes),
      precision_ratio = glue(&quot;{nrow(z)}/{nrow(mm_genes)}&quot;)
    )
  })) %&gt;%
  unnest(metrics)
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 65)
#&gt;             &gt; rows only in y  (682)
#&gt;             &gt; matched rows      10
#&gt;             &gt;                 =====
#&gt;             &gt; rows total        10
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 74)
#&gt;             &gt; rows only in y  (202)
#&gt;             &gt; matched rows       1
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         1
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 75)
#&gt;             &gt; rows only in y  (114)
#&gt;             &gt; matched rows       0
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         0
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (73)
#&gt;             &gt; rows only in y  (49)
#&gt;             &gt; matched rows      2
#&gt;             &gt;                 ====
#&gt;             &gt; rows total        2
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 67)
#&gt;             &gt; rows only in y  (343)
#&gt;             &gt; matched rows       8
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         8
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (   64)
#&gt;             &gt; rows only in y  (1,335)
#&gt;             &gt; matched rows         2
#&gt;             &gt;                 =======
#&gt;             &gt; rows total           2
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (   59)
#&gt;             &gt; rows only in y  (1,127)
#&gt;             &gt; matched rows         7
#&gt;             &gt;                 =======
#&gt;             &gt; rows total           7
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 64)
#&gt;             &gt; rows only in y  (482)
#&gt;             &gt; matched rows       2
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         2
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 65)
#&gt;             &gt; rows only in y  (147)
#&gt;             &gt; matched rows       1
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         1
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 66)
#&gt;             &gt; rows only in y  (109)
#&gt;             &gt; matched rows       0
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         0
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (   13)
#&gt;             &gt; rows only in y  (1,336)
#&gt;             &gt; matched rows         1
#&gt;             &gt;                 =======
#&gt;             &gt; rows total           1
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (   10)
#&gt;             &gt; rows only in y  (1,130)
#&gt;             &gt; matched rows         4
#&gt;             &gt;                 =======
#&gt;             &gt; rows total           4
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 14)
#&gt;             &gt; rows only in y  (484)
#&gt;             &gt; matched rows       0
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         0
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 13)
#&gt;             &gt; rows only in y  (147)
#&gt;             &gt; matched rows       1
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         1
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 14)
#&gt;             &gt; rows only in y  (109)
#&gt;             &gt; matched rows       0
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         0
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 13)
#&gt;             &gt; rows only in y  (689)
#&gt;             &gt; matched rows       3
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         3
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 15)
#&gt;             &gt; rows only in y  (202)
#&gt;             &gt; matched rows       1
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         1
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 16)
#&gt;             &gt; rows only in y  (114)
#&gt;             &gt; matched rows       0
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         0
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (16)
#&gt;             &gt; rows only in y  (51)
#&gt;             &gt; matched rows      0
#&gt;             &gt;                 ====
#&gt;             &gt; rows total        0
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 14)
#&gt;             &gt; rows only in y  (349)
#&gt;             &gt; matched rows       2
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         2
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 12)
#&gt;             &gt; rows only in y  (686)
#&gt;             &gt; matched rows       6
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         6
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 17)
#&gt;             &gt; rows only in y  (202)
#&gt;             &gt; matched rows       1
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         1
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 18)
#&gt;             &gt; rows only in y  (114)
#&gt;             &gt; matched rows       0
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         0
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (16)
#&gt;             &gt; rows only in y  (49)
#&gt;             &gt; matched rows      2
#&gt;             &gt;                 ====
#&gt;             &gt; rows total        2
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 16)
#&gt;             &gt; rows only in y  (349)
#&gt;             &gt; matched rows       2
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         2
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (   31)
#&gt;             &gt; rows only in y  (1,336)
#&gt;             &gt; matched rows         1
#&gt;             &gt;                 =======
#&gt;             &gt; rows total           1
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (   30)
#&gt;             &gt; rows only in y  (1,132)
#&gt;             &gt; matched rows         2
#&gt;             &gt;                 =======
#&gt;             &gt; rows total           2
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 32)
#&gt;             &gt; rows only in y  (484)
#&gt;             &gt; matched rows       0
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         0
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 30)
#&gt;             &gt; rows only in y  (146)
#&gt;             &gt; matched rows       2
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         2
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 31)
#&gt;             &gt; rows only in y  (108)
#&gt;             &gt; matched rows       1
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         1
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (   27)
#&gt;             &gt; rows only in y  (1,334)
#&gt;             &gt; matched rows         3
#&gt;             &gt;                 =======
#&gt;             &gt; rows total           3
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (   23)
#&gt;             &gt; rows only in y  (1,127)
#&gt;             &gt; matched rows         7
#&gt;             &gt;                 =======
#&gt;             &gt; rows total           7
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 29)
#&gt;             &gt; rows only in y  (483)
#&gt;             &gt; matched rows       1
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         1
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 29)
#&gt;             &gt; rows only in y  (147)
#&gt;             &gt; matched rows       1
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         1
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 29)
#&gt;             &gt; rows only in y  (108)
#&gt;             &gt; matched rows       1
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         1
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 24)
#&gt;             &gt; rows only in y  (685)
#&gt;             &gt; matched rows       7
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         7
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 30)
#&gt;             &gt; rows only in y  (202)
#&gt;             &gt; matched rows       1
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         1
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 31)
#&gt;             &gt; rows only in y  (114)
#&gt;             &gt; matched rows       0
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         0
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (30)
#&gt;             &gt; rows only in y  (50)
#&gt;             &gt; matched rows      1
#&gt;             &gt;                 ====
#&gt;             &gt; rows total        1
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 27)
#&gt;             &gt; rows only in y  (347)
#&gt;             &gt; matched rows       4
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         4
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (286)
#&gt;             &gt; rows only in y  (661)
#&gt;             &gt; matched rows      31
#&gt;             &gt;                 =====
#&gt;             &gt; rows total        31
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (307)
#&gt;             &gt; rows only in y  (193)
#&gt;             &gt; matched rows      10
#&gt;             &gt;                 =====
#&gt;             &gt; rows total        10
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (316)
#&gt;             &gt; rows only in y  (113)
#&gt;             &gt; matched rows       1
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         1
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (313)
#&gt;             &gt; rows only in y  ( 47)
#&gt;             &gt; matched rows       4
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         4
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (279)
#&gt;             &gt; rows only in y  (313)
#&gt;             &gt; matched rows      38
#&gt;             &gt;                 =====
#&gt;             &gt; rows total        38
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (  289)
#&gt;             &gt; rows only in y  (1,315)
#&gt;             &gt; matched rows        22
#&gt;             &gt;                 =======
#&gt;             &gt; rows total          22
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (  248)
#&gt;             &gt; rows only in y  (1,071)
#&gt;             &gt; matched rows        63
#&gt;             &gt;                 =======
#&gt;             &gt; rows total          63
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (275)
#&gt;             &gt; rows only in y  (448)
#&gt;             &gt; matched rows      36
#&gt;             &gt;                 =====
#&gt;             &gt; rows total        36
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (301)
#&gt;             &gt; rows only in y  (138)
#&gt;             &gt; matched rows      10
#&gt;             &gt;                 =====
#&gt;             &gt; rows total        10
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (305)
#&gt;             &gt; rows only in y  (103)
#&gt;             &gt; matched rows       6
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         6
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (430)
#&gt;             &gt; rows only in y  (645)
#&gt;             &gt; matched rows      47
#&gt;             &gt;                 =====
#&gt;             &gt; rows total        47
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (465)
#&gt;             &gt; rows only in y  (191)
#&gt;             &gt; matched rows      12
#&gt;             &gt;                 =====
#&gt;             &gt; rows total        12
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (476)
#&gt;             &gt; rows only in y  (113)
#&gt;             &gt; matched rows       1
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         1
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (472)
#&gt;             &gt; rows only in y  ( 46)
#&gt;             &gt; matched rows       5
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         5
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (424)
#&gt;             &gt; rows only in y  (298)
#&gt;             &gt; matched rows      53
#&gt;             &gt;                 =====
#&gt;             &gt; rows total        53
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (  479)
#&gt;             &gt; rows only in y  (1,286)
#&gt;             &gt; matched rows        51
#&gt;             &gt;                 =======
#&gt;             &gt; rows total          51
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (  425)
#&gt;             &gt; rows only in y  (1,029)
#&gt;             &gt; matched rows       105
#&gt;             &gt;                 =======
#&gt;             &gt; rows total         105
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (482)
#&gt;             &gt; rows only in y  (436)
#&gt;             &gt; matched rows      48
#&gt;             &gt;                 =====
#&gt;             &gt; rows total        48
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (513)
#&gt;             &gt; rows only in y  (131)
#&gt;             &gt; matched rows      17
#&gt;             &gt;                 =====
#&gt;             &gt; rows total        17
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (522)
#&gt;             &gt; rows only in y  (101)
#&gt;             &gt; matched rows       8
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         8
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (   49)
#&gt;             &gt; rows only in y  (1,333)
#&gt;             &gt; matched rows         4
#&gt;             &gt;                 =======
#&gt;             &gt; rows total           4
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (   43)
#&gt;             &gt; rows only in y  (1,124)
#&gt;             &gt; matched rows        10
#&gt;             &gt;                 =======
#&gt;             &gt; rows total          10
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 50)
#&gt;             &gt; rows only in y  (481)
#&gt;             &gt; matched rows       3
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         3
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 51)
#&gt;             &gt; rows only in y  (146)
#&gt;             &gt; matched rows       2
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         2
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 52)
#&gt;             &gt; rows only in y  (108)
#&gt;             &gt; matched rows       1
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         1
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 30)
#&gt;             &gt; rows only in y  (688)
#&gt;             &gt; matched rows       4
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         4
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 34)
#&gt;             &gt; rows only in y  (203)
#&gt;             &gt; matched rows       0
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         0
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 32)
#&gt;             &gt; rows only in y  (112)
#&gt;             &gt; matched rows       2
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         2
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (33)
#&gt;             &gt; rows only in y  (50)
#&gt;             &gt; matched rows      1
#&gt;             &gt;                 ====
#&gt;             &gt; rows total        1
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 28)
#&gt;             &gt; rows only in y  (345)
#&gt;             &gt; matched rows       6
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         6
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 77)
#&gt;             &gt; rows only in y  (687)
#&gt;             &gt; matched rows       5
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         5
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 80)
#&gt;             &gt; rows only in y  (201)
#&gt;             &gt; matched rows       2
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         2
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 81)
#&gt;             &gt; rows only in y  (113)
#&gt;             &gt; matched rows       1
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         1
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (81)
#&gt;             &gt; rows only in y  (50)
#&gt;             &gt; matched rows      1
#&gt;             &gt;                 ====
#&gt;             &gt; rows total        1
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 72)
#&gt;             &gt; rows only in y  (341)
#&gt;             &gt; matched rows      10
#&gt;             &gt;                 =====
#&gt;             &gt; rows total        10
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (  118)
#&gt;             &gt; rows only in y  (1,317)
#&gt;             &gt; matched rows        20
#&gt;             &gt;                 =======
#&gt;             &gt; rows total          20
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (  105)
#&gt;             &gt; rows only in y  (1,101)
#&gt;             &gt; matched rows        33
#&gt;             &gt;                 =======
#&gt;             &gt; rows total          33
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (121)
#&gt;             &gt; rows only in y  (467)
#&gt;             &gt; matched rows      17
#&gt;             &gt;                 =====
#&gt;             &gt; rows total        17
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (130)
#&gt;             &gt; rows only in y  (140)
#&gt;             &gt; matched rows       8
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         8
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (135)
#&gt;             &gt; rows only in y  (106)
#&gt;             &gt; matched rows       3
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         3
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (   73)
#&gt;             &gt; rows only in y  (1,329)
#&gt;             &gt; matched rows         8
#&gt;             &gt;                 =======
#&gt;             &gt; rows total           8
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (   54)
#&gt;             &gt; rows only in y  (1,107)
#&gt;             &gt; matched rows        27
#&gt;             &gt;                 =======
#&gt;             &gt; rows total          27
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 69)
#&gt;             &gt; rows only in y  (472)
#&gt;             &gt; matched rows      12
#&gt;             &gt;                 =====
#&gt;             &gt; rows total        12
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 77)
#&gt;             &gt; rows only in y  (144)
#&gt;             &gt; matched rows       4
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         4
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 79)
#&gt;             &gt; rows only in y  (107)
#&gt;             &gt; matched rows       2
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         2
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 43)
#&gt;             &gt; rows only in y  (687)
#&gt;             &gt; matched rows       5
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         5
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 47)
#&gt;             &gt; rows only in y  (202)
#&gt;             &gt; matched rows       1
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         1
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 48)
#&gt;             &gt; rows only in y  (114)
#&gt;             &gt; matched rows       0
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         0
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (46)
#&gt;             &gt; rows only in y  (49)
#&gt;             &gt; matched rows      2
#&gt;             &gt;                 ====
#&gt;             &gt; rows total        2
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  ( 42)
#&gt;             &gt; rows only in y  (345)
#&gt;             &gt; matched rows       6
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         6
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (132)
#&gt;             &gt; rows only in y  (681)
#&gt;             &gt; matched rows      11
#&gt;             &gt;                 =====
#&gt;             &gt; rows total        11
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (137)
#&gt;             &gt; rows only in y  (197)
#&gt;             &gt; matched rows       6
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         6
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (143)
#&gt;             &gt; rows only in y  (114)
#&gt;             &gt; matched rows       0
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         0
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (143)
#&gt;             &gt; rows only in y  ( 51)
#&gt;             &gt; matched rows       0
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         0
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (134)
#&gt;             &gt; rows only in y  (342)
#&gt;             &gt; matched rows       9
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         9
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (  611)
#&gt;             &gt; rows only in y  (1,239)
#&gt;             &gt; matched rows        98
#&gt;             &gt;                 =======
#&gt;             &gt; rows total          98
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (539)
#&gt;             &gt; rows only in y  (964)
#&gt;             &gt; matched rows     170
#&gt;             &gt;                 =====
#&gt;             &gt; rows total       170
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (663)
#&gt;             &gt; rows only in y  (438)
#&gt;             &gt; matched rows      46
#&gt;             &gt;                 =====
#&gt;             &gt; rows total        46
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (686)
#&gt;             &gt; rows only in y  (125)
#&gt;             &gt; matched rows      23
#&gt;             &gt;                 =====
#&gt;             &gt; rows total        23
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (700)
#&gt;             &gt; rows only in y  (100)
#&gt;             &gt; matched rows       9
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         9
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (  525)
#&gt;             &gt; rows only in y  (1,220)
#&gt;             &gt; matched rows       117
#&gt;             &gt;                 =======
#&gt;             &gt; rows total         117
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (506)
#&gt;             &gt; rows only in y  (998)
#&gt;             &gt; matched rows     136
#&gt;             &gt;                 =====
#&gt;             &gt; rows total       136
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (596)
#&gt;             &gt; rows only in y  (438)
#&gt;             &gt; matched rows      46
#&gt;             &gt;                 =====
#&gt;             &gt; rows total        46
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (616)
#&gt;             &gt; rows only in y  (122)
#&gt;             &gt; matched rows      26
#&gt;             &gt;                 =====
#&gt;             &gt; rows total        26
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (633)
#&gt;             &gt; rows only in y  (100)
#&gt;             &gt; matched rows       9
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         9
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (148)
#&gt;             &gt; rows only in y  (674)
#&gt;             &gt; matched rows      18
#&gt;             &gt;                 =====
#&gt;             &gt; rows total        18
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (160)
#&gt;             &gt; rows only in y  (197)
#&gt;             &gt; matched rows       6
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         6
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (166)
#&gt;             &gt; rows only in y  (114)
#&gt;             &gt; matched rows       0
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         0
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (164)
#&gt;             &gt; rows only in y  ( 49)
#&gt;             &gt; matched rows       2
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         2
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (157)
#&gt;             &gt; rows only in y  (342)
#&gt;             &gt; matched rows       9
#&gt;             &gt;                 =====
#&gt;             &gt; rows total         9
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (2,114)
#&gt;             &gt; rows only in y  (  998)
#&gt;             &gt; matched rows       339
#&gt;             &gt;                 =======
#&gt;             &gt; rows total         339
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (2,026)
#&gt;             &gt; rows only in y  (  707)
#&gt;             &gt; matched rows       427
#&gt;             &gt;                 =======
#&gt;             &gt; rows total         427
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (2,294)
#&gt;             &gt; rows only in y  (  325)
#&gt;             &gt; matched rows       159
#&gt;             &gt;                 =======
#&gt;             &gt; rows total         159
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (2,394)
#&gt;             &gt; rows only in y  (   89)
#&gt;             &gt; matched rows        59
#&gt;             &gt;                 =======
#&gt;             &gt; rows total          59
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (2,416)
#&gt;             &gt; rows only in y  (   72)
#&gt;             &gt; matched rows        37
#&gt;             &gt;                 =======
#&gt;             &gt; rows total          37
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (1,077)
#&gt;             &gt; rows only in y  (  582)
#&gt;             &gt; matched rows       110
#&gt;             &gt;                 =======
#&gt;             &gt; rows total         110
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (1,159)
#&gt;             &gt; rows only in y  (  175)
#&gt;             &gt; matched rows        28
#&gt;             &gt;                 =======
#&gt;             &gt; rows total          28
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (1,181)
#&gt;             &gt; rows only in y  (  108)
#&gt;             &gt; matched rows         6
#&gt;             &gt;                 =======
#&gt;             &gt; rows total           6
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (1,176)
#&gt;             &gt; rows only in y  (   40)
#&gt;             &gt; matched rows        11
#&gt;             &gt;                 =======
#&gt;             &gt; rows total          11
#&gt; inner_join: added no columns
#&gt;             &gt; rows only in x  (1,121)
#&gt;             &gt; rows only in y  (  285)
#&gt;             &gt; matched rows        66
#&gt;             &gt;                 =======
#&gt;             &gt; rows total          66

# check how well pr improves when going from 2 to 12 months on average
pr %&gt;% 
  filter(study %in% c(&quot;pure_ccl_12m_vs_0m&quot;, &quot;pure_ccl_2m_vs_0m&quot;)) %&gt;% 
  select(study, regulation, recall, etiology) %&gt;% 
  spread(study, recall) %&gt;% 
  mutate(diff = pure_ccl_12m_vs_0m - pure_ccl_2m_vs_0m) %&gt;% 
  pull(diff) %&gt;% 
  mean()
#&gt; filter: removed 100 rows (83%), 20 rows remaining
#&gt; select: dropped 6 variables (class, mm_genes, hs_genes, recall_ratio, precision, …)
#&gt; spread: reorganized (study, recall) into (pure_ccl_2m_vs_0m, pure_ccl_12m_vs_0m) [was 20x4, now 10x4]
#&gt; [1] 0.182263

saveRDS(pr, here(output_path, &quot;precision_recall.rds&quot;))</code></pre>
<div id="pr-scatter-plot" class="section level4">
<h4>PR scatter plot</h4>
<p>Scatter plot showing precision and recall for each chronic mouse model.</p>
<pre class="r"><code>pr &lt;- readRDS(here(output_path, &quot;precision_recall.rds&quot;))

pr %&gt;%
  ggplot(aes(x = recall, y = precision, label = study, color = study)) +
  geom_point() +
  facet_rep_grid(etiology ~ regulation) +
  geom_abline(lty = &quot;dashed&quot;) +
  expand_limits(x = 0, y = 0) +
  labs(x = &quot;Recall&quot;, y = &quot;Precision&quot;, color = &quot;Mouse model&quot;) +
  my_theme(fsize = fz) +
  scale_color_viridis_d()</code></pre>
<p><img src="figure/meta-mouse-vs-human.Rmd/pr-scatter-plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pr-scatter-plot-1">
Past versions of pr-scatter-plot-1.png
</button>
</p>
<div id="fig-pr-scatter-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/f8d83433873c01bdc55750a44f5e3a14f0a6c338/docs/figure/meta-mouse-vs-human.Rmd/pr-scatter-plot-1.png" target="_blank">f8d8343</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Time spend to execute this analysis: 04:00 minutes.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
#&gt; R version 4.0.2 (2020-06-22)
#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)
#&gt; Running under: macOS Mojave 10.14.5
#&gt; 
#&gt; Matrix products: default
#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
#&gt; 
#&gt; locale:
#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
#&gt; 
#&gt; attached base packages:
#&gt; [1] grid      stats     graphics  grDevices datasets  utils     methods  
#&gt; [8] base     
#&gt; 
#&gt; other attached packages:
#&gt;  [1] gtools_3.8.2             msigdf_7.1               patchwork_1.1.1         
#&gt;  [4] ggwordcloud_0.5.0        UpSetR_1.4.0             ggpubr_0.4.0            
#&gt;  [7] cowplot_1.1.0            gridExtra_2.3            ComplexHeatmap_2.4.3    
#&gt; [10] VennDiagram_1.6.20       futile.logger_1.4.3      lemon_0.4.5             
#&gt; [13] AachenColorPalette_1.1.2 circlize_0.4.11          biobroom_1.20.0         
#&gt; [16] broom_0.7.3              progeny_1.10.0           dorothea_1.0.1          
#&gt; [19] fgsea_1.14.0             janitor_2.0.1            tidytext_0.2.6          
#&gt; [22] glue_1.4.2               readxl_1.3.1             here_1.0.1              
#&gt; [25] tidylog_1.0.2            forcats_0.5.0            stringr_1.4.0           
#&gt; [28] dplyr_1.0.2              purrr_0.3.4              readr_1.4.0             
#&gt; [31] tidyr_1.1.2              tibble_3.0.4             ggplot2_3.3.2           
#&gt; [34] tidyverse_1.3.0          workflowr_1.6.2         
#&gt; 
#&gt; loaded via a namespace (and not attached):
#&gt;  [1] colorspace_2.0-0     ggsignif_0.6.0       rjson_0.2.20        
#&gt;  [4] rio_0.5.16           ellipsis_0.3.1       rprojroot_2.0.2     
#&gt;  [7] snakecase_0.11.0     GlobalOptions_0.1.2  fs_1.5.0            
#&gt; [10] clue_0.3-58          rstudioapi_0.13      farver_2.0.3        
#&gt; [13] SnowballC_0.7.0      ggrepel_0.9.0        fansi_0.4.1         
#&gt; [16] lubridate_1.7.9.2    xml2_1.3.2           codetools_0.2-16    
#&gt; [19] knitr_1.30           jsonlite_1.7.2       bcellViper_1.24.0   
#&gt; [22] cluster_2.1.0        dbplyr_2.0.0         png_0.1-7           
#&gt; [25] compiler_4.0.2       httr_1.4.2           backports_1.2.1     
#&gt; [28] assertthat_0.2.1     Matrix_1.2-18        cli_2.2.0           
#&gt; [31] later_1.1.0.1        formatR_1.7          htmltools_0.5.0     
#&gt; [34] tools_4.0.2          gtable_0.3.0         fastmatch_1.1-0     
#&gt; [37] Rcpp_1.0.5           carData_3.0-4        Biobase_2.48.0      
#&gt; [40] cellranger_1.1.0     vctrs_0.3.6          xfun_0.19           
#&gt; [43] openxlsx_4.2.3       rvest_0.3.6          lifecycle_0.2.0     
#&gt; [46] renv_0.12.3          rstatix_0.6.0        scales_1.1.1        
#&gt; [49] clisymbols_1.2.0     hms_0.5.3            promises_1.1.1      
#&gt; [52] parallel_4.0.2       lambda.r_1.2.4       RColorBrewer_1.1-2  
#&gt; [55] curl_4.3             yaml_2.2.1           stringi_1.5.3       
#&gt; [58] tokenizers_0.2.1     BiocGenerics_0.34.0  zip_2.1.1           
#&gt; [61] BiocParallel_1.22.0  shape_1.4.5          rlang_0.4.9         
#&gt; [64] pkgconfig_2.0.3      evaluate_0.14        lattice_0.20-41     
#&gt; [67] labeling_0.4.2       tidyselect_1.1.0     plyr_1.8.6          
#&gt; [70] magrittr_2.0.1       R6_2.5.0             generics_0.1.0      
#&gt; [73] DBI_1.1.0            foreign_0.8-80       pillar_1.4.7        
#&gt; [76] haven_2.3.1          whisker_0.4          withr_2.3.0         
#&gt; [79] abind_1.4-5          car_3.0-10           janeaustenr_0.1.5   
#&gt; [82] modelr_0.1.8         crayon_1.3.4         futile.options_1.0.1
#&gt; [85] rmarkdown_2.6        GetoptLong_1.0.5     data.table_1.13.4   
#&gt; [88] git2r_0.27.1         reprex_0.3.0         digest_0.6.27       
#&gt; [91] httpuv_1.5.4         munsell_0.5.0        viridisLite_0.3.0</code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
