<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Acute CCl4 mouse model</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Liver disease atlas</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fas fa-home"></span>
     
    Home
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Mouse models
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="mouse-chronic-ccl4.html">CCl4 (chronic)</a>
    </li>
    <li>
      <a href="mouse-acute-apap.html">APAP (acute)</a>
    </li>
    <li>
      <a href="mouse-acute-ccl4.html">CCl4 (acute)</a>
    </li>
    <li>
      <a href="mouse-acute-bdl.html">BDL (acute)</a>
    </li>
    <li>
      <a href="mouse-acute-ph.html">PH (acute)</a>
    </li>
    <li>
      <a href="mouse-acute-lps.html">LPS (acute)</a>
    </li>
    <li>
      <a href="mouse-acute-tunicamycin.html">Tunicamycin (acute)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Patient cohorts
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="human-hoang-nafld.html">NAFLD - Hoang et al.</a>
    </li>
    <li>
      <a href="human-diehl-nafld.html">NAFLD - Diehl et al.</a>
    </li>
    <li>
      <a href="human-ramnath-fibrosis.html">Fibrosis - Ramnath et al.</a>
    </li>
    <li>
      <a href="human-hampe13-nash.html">NASH - Hampe et al. '13</a>
    </li>
    <li>
      <a href="human-hampe14-misc.html">Misc - Hampe et al. '14</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Meta analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="meta-chronic-vs-acute.html">Chronic vs. acute</a>
    </li>
    <li>
      <a href="meta-mouse-vs-human.html">Mouse vs. human</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Figures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="plot-study-overview.html">Study overview (SFig. 0.1)</a>
    </li>
    <li>
      <a href="plot-chronic-ccl4.html">Chronic CCl4 (Fig. 1)</a>
    </li>
    <li>
      <a href="plot-acute-apap.html">APAP (Fig. 2)</a>
    </li>
    <li>
      <a href="plot-acute-ccl4.html">Acute CCl4 (SFig. 2.2)</a>
    </li>
    <li>
      <a href="plot-acute-ph.html">PH (SFig. 2.4)</a>
    </li>
    <li>
      <a href="plot-acute-bdl.html">BDL (SFig. 2.6)</a>
    </li>
    <li>
      <a href="plot-chronic-vs-acute.html">Chronic vs. acute (Fig. 3)</a>
    </li>
    <li>
      <a href="plot-mouse-vs-human.html">Mouse vs. human (Fig. 4)</a>
    </li>
    <li>
      <a href="plot-precision-recall.html">Precision and recall analysis (Fig. 6)</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://doi.org/10.5281/zenodo.4407208">
    <span class="fas fa-database"></span>
     
    Get data
  </a>
</li>
<li>
  <a href="https://github.com/saezlab/liver-disease-atlas">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Acute CCl4 mouse model</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-02-27
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>liver-disease-atlas/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20201218code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20201218)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20201218code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20201218)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsaezlabliverdiseaseatlastree40203d5fcfd0af9fa5b006d6711874376826d215targetblank40203d5a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/saezlab/liver-disease-atlas/tree/40203d5fcfd0af9fa5b006d6711874376826d215" target="_blank">40203d5</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsaezlabliverdiseaseatlastree40203d5fcfd0af9fa5b006d6711874376826d215targetblank40203d5a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/saezlab/liver-disease-atlas/tree/40203d5fcfd0af9fa5b006d6711874376826d215" target="_blank">40203d5</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/human-diehl-nafld_cache/
    Ignored:    analysis/human-hampe13-nash_cache/
    Ignored:    analysis/human-hampe14-misc_cache/
    Ignored:    analysis/human-hoang-nafld_cache/
    Ignored:    analysis/human-ramnath-fibrosis_cache/
    Ignored:    analysis/meta-chronic-vs-acute_cache/
    Ignored:    analysis/meta-mouse-vs-human_cache/
    Ignored:    analysis/mouse-acute-apap_cache/
    Ignored:    analysis/mouse-acute-bdl_cache/
    Ignored:    analysis/mouse-acute-lps_cache/
    Ignored:    analysis/mouse-acute-ph_cache/
    Ignored:    analysis/mouse-acute-tunicamycin_cache/
    Ignored:    analysis/mouse-chronic-ccl4_cache/
    Ignored:    analysis/plot-acute-apap_cache/
    Ignored:    analysis/plot-acute-bdl_cache/
    Ignored:    analysis/plot-acute-ccl4_cache/
    Ignored:    analysis/plot-acute-ph_cache/
    Ignored:    analysis/plot-chronic-ccl4_cache/
    Ignored:    analysis/plot-chronic-vs-acute_cache/
    Ignored:    analysis/plot-mouse-vs-human_cache/
    Ignored:    analysis/plot-precision-recall_cache/
    Ignored:    analysis/plot-study-overview_cache/
    Ignored:    analysis/plot-teufel-integration_cache/
    Ignored:    analysis/save-tables_cache/
    Ignored:    code/.DS_Store
    Ignored:    code/README.html
    Ignored:    code/meta-mouse-vs-human/.DS_Store
    Ignored:    data.zip
    Ignored:    data/.DS_Store
    Ignored:    data/annotation/
    Ignored:    data/human-diehl-nafld/
    Ignored:    data/human-hampe13-nash/
    Ignored:    data/human-hampe14-misc/
    Ignored:    data/human-hoang-nafld/
    Ignored:    data/human-ramnath-fibrosis/
    Ignored:    data/meta-chronic-vs-acute/
    Ignored:    data/meta-mouse-vs-human/
    Ignored:    data/mouse-acute-apap/
    Ignored:    data/mouse-acute-bdl/
    Ignored:    data/mouse-acute-ccl4/
    Ignored:    data/mouse-acute-lps/
    Ignored:    data/mouse-acute-ph/
    Ignored:    data/mouse-acute-tunicamycin/
    Ignored:    data/mouse-chronic-ccl4/
    Ignored:    external_software/.DS_Store
    Ignored:    external_software/README.html
    Ignored:    external_software/stem/.DS_Store
    Ignored:    figures/
    Ignored:    geo_submission/
    Ignored:    output/.DS_Store
    Ignored:    output/README.html
    Ignored:    output/human-diehl-nafld/
    Ignored:    output/human-hampe13-nash/
    Ignored:    output/human-hampe14-misc/
    Ignored:    output/human-hoang-nafld/
    Ignored:    output/human-ramnath-fibrosis/
    Ignored:    output/meta-chronic-vs-acute/
    Ignored:    output/meta-mouse-vs-human/
    Ignored:    output/mouse-acute-apap/
    Ignored:    output/mouse-acute-bdl/
    Ignored:    output/mouse-acute-ccl4/
    Ignored:    output/mouse-acute-lps/
    Ignored:    output/mouse-acute-ph/
    Ignored:    output/mouse-acute-tunicamycin/
    Ignored:    output/mouse-chronic-ccl4/
    Ignored:    renv/library/
    Ignored:    renv/staging/
    Ignored:    tables/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/mouse-acute-ccl4.Rmd</code>) and HTML (<code>docs/mouse-acute-ccl4.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/40203d5fcfd0af9fa5b006d6711874376826d215/analysis/mouse-acute-ccl4.Rmd" target="_blank">40203d5</a>
</td>
<td>
christianholland
</td>
<td>
2021-02-27
</td>
<td>
wflow_publish("analysis/*", delete_cache = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/saezlab/liver-disease-atlas/9c62197c1a318b00f52dbf3e9e3613ebd34b9b33/docs/mouse-acute-ccl4.html" target="_blank">9c62197</a>
</td>
<td>
christianholland
</td>
<td>
2021-01-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/saezlab/liver-disease-atlas/067c9337e4cb6becda6b2d2ace64a70dc92107a3/docs/mouse-acute-ccl4.html" target="_blank">067c933</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/d4f78fae08eb96da82f28d39a0c6d2962abb7ecd/analysis/mouse-acute-ccl4.Rmd" target="_blank">d4f78fa</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-23
</td>
<td>
wflow_publish("analysis/*.Rmd", delete_cache = T)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/saezlab/liver-disease-atlas/8a9b7bf0458f500b5efba0985b3c1ac38ad63f7f/docs/mouse-acute-ccl4.html" target="_blank">8a9b7bf</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/saezlab/liver-disease-atlas/e22a40bafb648c6d8240eab4610edfb69278df74/docs/mouse-acute-ccl4.html" target="_blank">e22a40b</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/saezlab/liver-disease-atlas/bea437a2a4cb44a0186b2c0083a042b63195d509/docs/mouse-acute-ccl4.html" target="_blank">bea437a</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/c78e88389929df94c6790585a37c4cf7798d04f8/analysis/mouse-acute-ccl4.Rmd" target="_blank">c78e883</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-20
</td>
<td>
stem characterization
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/saezlab/liver-disease-atlas/af384509e61f6d626b8f2f1bccf80c7f075f5cbc/docs/mouse-acute-ccl4.html" target="_blank">af38450</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/78a25d5d1882aa788f0a75a6b2f9753d203c3d76/analysis/mouse-acute-ccl4.Rmd" target="_blank">78a25d5</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-19
</td>
<td>
minor changes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/saezlab/liver-disease-atlas/c518c89194da6885aa6623ad0fb9b4a989d0a948/docs/mouse-acute-ccl4.html" target="_blank">c518c89</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/961f67d329ccfd433a6c17248aab984db35ed363/analysis/mouse-acute-ccl4.Rmd" target="_blank">961f67d</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-19
</td>
<td>
added acute ccl4 study
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Here we analysis a mouse model of CCl4 induced acute liver damage. The transcriptomic profiles were measured at 9 different time points ranging from 2 hours to 16 days.</p>
</div>
<div id="libraries-and-sources" class="section level2">
<h2>Libraries and sources</h2>
<p>These libraries and sources are used for this analysis.</p>
<pre class="r"><code>library(mouse4302.db)

library(tidyverse)
library(tidylog)
library(here)

library(oligo)
library(annotate)
library(limma)
library(biobroom)
library(progeny)
library(dorothea)

library(janitor)
library(msigdf) # remotes::install_github(&quot;ToledoEM/msigdf@v7.1&quot;)

library(AachenColorPalette)
library(cowplot)
library(lemon)
library(patchwork)

options(&quot;tidylog.display&quot; = list(print))
source(here(&quot;code/utils-microarray.R&quot;))
source(here(&quot;code/utils-utils.R&quot;))
source(here(&quot;code/utils-plots.R&quot;))</code></pre>
<p>Definition of global variables that are used throughout this analysis.</p>
<pre class="r"><code># i/o
data_path &lt;- &quot;data/mouse-acute-ccl4&quot;
output_path &lt;- &quot;output/mouse-acute-ccl4&quot;

# graphical parameters
# fontsize
fz &lt;- 9</code></pre>
</div>
<div id="data-processing" class="section level2">
<h2>Data processing</h2>
<div id="load-.cel-files-and-quality-control" class="section level3">
<h3>Load .CEL files and quality control</h3>
<p>The array quality is controlled based on the relative log expression values (RLE) and the normalized unscaled standard errors (NUSE).</p>
<pre class="r"><code># load cel files and check quality
platforms &lt;- readRDS(here(&quot;data/annotation/platforms.rds&quot;))
raw_eset &lt;- list.celfiles(here(data_path), listGzipped = T, full.names = T) %&gt;%
  read.celfiles() %&gt;%
  ma_qc() 
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_2hm1_16_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_2hm2_17_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_2hm3_18_(Mouse430_2)_2.CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_2hm4_19_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_2hm5_20_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_8hm1_21_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_8hm2_22_(Mouse430_2)_2.CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_8hm3_23_(Mouse430_2)_2.CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_8hm4_24_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_8hm5_25_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_D16M1_51_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_D16M2_52_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_D16M3_53_(Mouse430_2)_2.CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_D16M4_54_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_D16M5_55_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_D16M6_56_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_d1m1_26_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_d1m2_27_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_d1m3_28_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_d1m4_29_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_d1m5_30_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_d2m1_31_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_d2m2_32_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_d2m3_33_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_d2m4_34_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_d2m5_35_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_d4m1_36_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_d4m2_37_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_d4m3_38_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_d4m4_39_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_d4m5_40_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_d6m1_41_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_d6m2_42_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_d6m3_43_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_d6m4_44_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_d6m5_45_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_D8M1_46_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_D8M2_47_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_D8M3_48_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_D8M4_49_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_D8M5_50_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_KM1_1_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_KM2_2_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_KM3_3_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_KM4_4_(Mouse430_2).CEL
#&gt; Reading in : /Users/cholland/Google Drive/Projects/liver-disease-atlas/data/mouse-acute-ccl4/Patricio_KM5_5_(Mouse430_2).CEL</code></pre>
</div>
<div id="normalization-and-probe-annotation" class="section level3">
<h3>Normalization and probe annotation</h3>
<p>Probe intensities are normalized with the <code>rma()</code> function. Probes are annotated with MGI symbols.</p>
<pre class="r"><code>eset &lt;- rma(raw_eset)
#&gt; Background correcting
#&gt; Normalizing
#&gt; Calculating Expression

# annotate microarray probes with mgi symbols
expr &lt;- ma_annotate(eset, platforms)
pattern &lt;- str_c(c(
  &quot;Patricio_&quot;,
  &quot;_\\(Mouse430_2\\).CEL&quot;,
  &quot;_\\(Mouse430_2\\)_2.CEL&quot;
), collapse = &quot;|&quot;)

colnames(expr) &lt;- str_c(
  &quot;sample&quot;, str_remove_all(colnames(expr), pattern),
  sep = &quot;_&quot;
) %&gt;%
  str_to_lower()

# save normalized expression
saveRDS(expr, here(output_path, &quot;normalized_expression.rds&quot;))</code></pre>
</div>
<div id="build-meta-data" class="section level3">
<h3>Build meta data</h3>
<p>Meta information are parsed from the sample names.</p>
<pre class="r"><code># build meta data
meta &lt;- colnames(expr) %&gt;%
  enframe(name = NULL, value = &quot;sample&quot;) %&gt;%
  separate(sample, into = c(&quot;tmp&quot;, &quot;key&quot;, &quot;run&quot;), remove = F) %&gt;%
  dplyr::select(-tmp) %&gt;%
  separate(key, into = c(&quot;group&quot;, &quot;rep&quot;), sep = &quot;m&quot;, remove = T) %&gt;%
  mutate(time = case_when(
    group == &quot;k&quot; ~ 0,
    str_detect(group, &quot;h&quot;) ~ parse_number(group),
    str_detect(group, &quot;d&quot;) ~ parse_number(group) * 24
  )) %&gt;%
  mutate(time = ordered(time)) %&gt;%
  mutate(group = case_when(
    group == &quot;k&quot; ~ &quot;control&quot;,
    str_detect(group, &quot;h&quot;) ~ str_c(&quot;h&quot;, parse_number(group)),
    TRUE ~ group
  )) %&gt;%
  mutate(group = factor(group, levels = c(
    &quot;control&quot;, &quot;h2&quot;, &quot;h8&quot;, &quot;d1&quot;, &quot;d2&quot;,
    &quot;d4&quot;, &quot;d6&quot;, &quot;d8&quot;, &quot;d16&quot;
  ))) %&gt;%
  arrange(sample)
#&gt; mutate: new variable &#39;time&#39; (double) with 9 unique values and 0% NA
#&gt; mutate: converted &#39;time&#39; from double to ordered factor (0 new NA)
#&gt; mutate: changed 15 values (33%) of &#39;group&#39; (0 new NA)
#&gt; mutate: converted &#39;group&#39; from character to factor (0 new NA)

# save meta data
saveRDS(meta, here(output_path, &quot;meta_data.rds&quot;))</code></pre>
</div>
</div>
<div id="exploratory-analysis" class="section level2">
<h2>Exploratory analysis</h2>
<div id="pca-of-normalized-data" class="section level3">
<h3>PCA of normalized data</h3>
<p>PCA plot of normalized expression data contextualized based on the time point. Only the top 1000 most variable genes are used as features.</p>
<pre class="r"><code>expr &lt;- readRDS(here(output_path, &quot;normalized_expression.rds&quot;))
meta &lt;- readRDS(here(output_path, &quot;meta_data.rds&quot;))

pca_result &lt;- do_pca(expr, meta, top_n_var_genes = 1000)
#&gt; left_join: added 4 columns (group, rep, run, time)
#&gt;            &gt; rows only in x    0
#&gt;            &gt; rows only in y  ( 0)
#&gt;            &gt; matched rows     46
#&gt;            &gt;                 ====
#&gt;            &gt; rows total       46

saveRDS(pca_result, here(output_path, &quot;pca_result.rds&quot;))

plot_pca(pca_result, feature = &quot;time&quot;) +
  my_theme()</code></pre>
<p><img src="figure/mouse-acute-ccl4.Rmd/pca-norm-data-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-norm-data-1">
Past versions of pca-norm-data-1.png
</button>
</p>
<div id="fig-pca-norm-data-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/8a9b7bf0458f500b5efba0985b3c1ac38ad63f7f/docs/figure/mouse-acute-ccl4.Rmd/pca-norm-data-1.png" target="_blank">8a9b7bf</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/e22a40bafb648c6d8240eab4610edfb69278df74/docs/figure/mouse-acute-ccl4.Rmd/pca-norm-data-1.png" target="_blank">e22a40b</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/c518c89194da6885aa6623ad0fb9b4a989d0a948/docs/figure/mouse-acute-ccl4.Rmd/pca-norm-data-1.png" target="_blank">c518c89</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="differential-gene-expression-analysis" class="section level2">
<h2>Differential gene expression analysis</h2>
<div id="running-limma" class="section level3">
<h3>Running limma</h3>
<p>Differential gene expression analysis via limma with the aim to identify the effect of CCl4 intoxication for the different time points.</p>
<pre class="r"><code># load expression and meta data
expr &lt;- readRDS(here(output_path, &quot;normalized_expression.rds&quot;))
meta &lt;- readRDS(here(output_path, &quot;meta_data.rds&quot;))

stopifnot(colnames(expr) == meta$sample)

# build design matrix
design &lt;- model.matrix(~ 0 + group, data = meta)
rownames(design) &lt;- meta$sample
colnames(design) &lt;- levels(meta$group)

# define contrasts
contrasts &lt;- makeContrasts(
  # effect of ccl4 treatment
  ccl_2h_vs_0h = h2 - control,
  ccl_8h_vs_0h = h8 - control,
  ccl_24h_vs_0h = d1 - control,
  ccl_48h_vs_0h = d2 - control,
  ccl_96h_vs_0h = d4 - control,
  ccl_144h_vs_0h = d6 - control,
  ccl_192h_vs_0h = d8 - control,
  ccl_384h_vs_0h = d16 - control,

  # consecutive time point comparison
  consec_2h_vs_0h = h2 - control,
  consec_8h_vs_2h = h8 - h2,
  consec_24h_vs_8h = d1 - h8,
  consec_48h_vs_24h = d2 - d1,
  consec_96h_vs_48h = d4 - d2,
  consec_144h_vs_96h = d6 - d4,
  consec_192h_vs_144h = d8 - d6,
  consec_384h_vs_192h = d16 - d8,
  levels = design
)

limma_result &lt;- run_limma(expr, design, contrasts) %&gt;%
  assign_deg()
#&gt; select: renamed 3 variables (contrast, logFC, pval) and dropped one variable
#&gt; group_by: one grouping variable (contrast)
#&gt; mutate (grouped): new variable &#39;fdr&#39; (double) with 151,248 unique values and 0% NA
#&gt; ungroup: no grouping variables
#&gt; mutate: new variable &#39;regulation&#39; (character) with 3 unique values and 0% NA
#&gt; mutate: converted &#39;regulation&#39; from character to factor (0 new NA)

deg_df &lt;- limma_result %&gt;%
  mutate(contrast = fct_inorder(contrast)) %&gt;%
  mutate(contrast_reference = case_when(
    str_detect(contrast, &quot;ccl&quot;) ~ &quot;ccl4&quot;, 
    str_detect(contrast, &quot;consec&quot;) ~ &quot;consec&quot;
  ))
#&gt; mutate: changed 0 values (0%) of &#39;contrast&#39; (0 new NA)
#&gt; mutate: new variable &#39;contrast_reference&#39; (character) with 2 unique values and 0% NA

saveRDS(deg_df, here(output_path, &quot;limma_result.rds&quot;))</code></pre>
</div>
<div id="volcano-plots" class="section level3">
<h3>Volcano plots</h3>
<p>Volcano plots visualizing the effect of CCl4 on gene expression.</p>
<pre class="r"><code>df &lt;- readRDS(here(output_path, &quot;limma_result.rds&quot;))

df %&gt;%
  filter(contrast_reference == &quot;ccl4&quot;) %&gt;%
  plot_volcano() +
  my_theme(grid = &quot;y&quot;, fsize = fz)
#&gt; filter: removed 164,072 rows (50%), 164,072 rows remaining
#&gt; rename: renamed one variable (p)</code></pre>
<p><img src="figure/mouse-acute-ccl4.Rmd/volcano-plots-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-volcano-plots-1">
Past versions of volcano-plots-1.png
</button>
</p>
<div id="fig-volcano-plots-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/8a9b7bf0458f500b5efba0985b3c1ac38ad63f7f/docs/figure/mouse-acute-ccl4.Rmd/volcano-plots-1.png" target="_blank">8a9b7bf</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/e22a40bafb648c6d8240eab4610edfb69278df74/docs/figure/mouse-acute-ccl4.Rmd/volcano-plots-1.png" target="_blank">e22a40b</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/bea437a2a4cb44a0186b2c0083a042b63195d509/docs/figure/mouse-acute-ccl4.Rmd/volcano-plots-1.png" target="_blank">bea437a</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/c518c89194da6885aa6623ad0fb9b4a989d0a948/docs/figure/mouse-acute-ccl4.Rmd/volcano-plots-1.png" target="_blank">c518c89</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="time-series-clustering" class="section level2">
<h2>Time series clustering</h2>
<p>Gene expression trajectories are clustered using the <a href="http://www.cs.cmu.edu/~jernst/stem/">STEM</a> software. The cluster algorithm is described <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-7-191">here</a>.</p>
<div id="prepare-input" class="section level3">
<h3>Prepare input</h3>
<pre class="r"><code># prepare input for stem analysis
df &lt;- readRDS(here(output_path, &quot;limma_result.rds&quot;))

stem_inputs &lt;- df %&gt;%
  mutate(class = str_c(&quot;Hour &quot;, parse_number(as.character(contrast)))) %&gt;%
  mutate(class = factor(class, levels = unique(.$class))) %&gt;%
  select(gene, class, logFC, contrast_reference)
#&gt; mutate: new variable &#39;class&#39; (character) with 8 unique values and 0% NA
#&gt; mutate: converted &#39;class&#39; from character to factor (0 new NA)
#&gt; select: dropped 5 variables (contrast, statistic, pval, fdr, regulation)

stem_inputs %&gt;%
  filter(contrast_reference == &quot;ccl4&quot;) %&gt;%
  select(-contrast_reference) %&gt;%
  pivot_wider(names_from = class, values_from = logFC) %&gt;%
  write_delim(here(output_path, &quot;stem/input/ccl4.txt&quot;), delim = &quot;\t&quot;)
#&gt; filter: removed 164,072 rows (50%), 164,072 rows remaining
#&gt; select: dropped one variable (contrast_reference)
#&gt; pivot_wider: reorganized (class, logFC) into (Hour 2, Hour 8, Hour 24, Hour 48, Hour 96, …) [was 164072x3, now 20509x9]</code></pre>
</div>
<div id="run-stem" class="section level3">
<h3>Run STEM</h3>
<p>STEM is implemented in Java. The .jar file is called from R. Only significant time series clusters are visualized.</p>
<pre class="r"><code># execute stem
stem_res &lt;- run_stem(file.path(output_path, &quot;stem&quot;), clear_output = T)
#&gt; distinct: no rows removed
#&gt; mutate: new variable &#39;gene&#39; (character) with 20,509 unique values and 0% NA
#&gt; distinct: no rows removed
#&gt; mutate: new variable &#39;key&#39; (character) with one unique value and 0% NA
#&gt; select: dropped one variable (spot)
#&gt; gather: reorganized (x0, hour_2, hour_8, hour_24, hour_48, …) into (time, value) [was 2085x12, now 18765x5]
#&gt; mutate: converted &#39;time&#39; from character to double (0 new NA)
#&gt; mutate: new variable &#39;key&#39; (character) with one unique value and 0% NA
#&gt; select: renamed 4 variables (profile, y_coords, size, p) and dropped 2 variables
#&gt; inner_join: added 3 columns (y_coords, size, p)
#&gt;             &gt; rows only in x  (     0)
#&gt;             &gt; rows only in y  (     0)
#&gt;             &gt; matched rows     18,765
#&gt;             &gt;                 ========
#&gt;             &gt; rows total       18,765
#&gt; inner_join: added one column (symbol)
#&gt;             &gt; rows only in x  (     0)
#&gt;             &gt; rows only in y  (18,424)
#&gt;             &gt; matched rows     18,765
#&gt;             &gt;                 ========
#&gt;             &gt; rows total       18,765
#&gt; transmute: dropped one variable (symbol)
#&gt;            changed 18,621 values (99%) of &#39;gene&#39; (0 new NA)

saveRDS(stem_res, here(output_path, &quot;stem_result.rds&quot;))

stem_res %&gt;%
  filter(p &lt;= 0.05) %&gt;%
  filter(key == &quot;ccl4&quot;) %&gt;%
  distinct() %&gt;%
  plot_stem_profiles(model_profile = F, ncol = 2) +
  labs(x = &quot;Time in Hours&quot;, y = &quot;logFC&quot;) +
  my_theme(grid = &quot;y&quot;, fsize = fz)
#&gt; filter: removed 2,781 rows (15%), 15,984 rows remaining
#&gt; filter: no rows removed
#&gt; distinct: no rows removed
#&gt; group_by: 4 grouping variables (key, profile, p, time)
#&gt; ungroup: no grouping variables</code></pre>
<p><img src="figure/mouse-acute-ccl4.Rmd/run-stem-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-run-stem-1">
Past versions of run-stem-1.png
</button>
</p>
<div id="fig-run-stem-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/8a9b7bf0458f500b5efba0985b3c1ac38ad63f7f/docs/figure/mouse-acute-ccl4.Rmd/run-stem-1.png" target="_blank">8a9b7bf</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/e22a40bafb648c6d8240eab4610edfb69278df74/docs/figure/mouse-acute-ccl4.Rmd/run-stem-1.png" target="_blank">e22a40b</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/bea437a2a4cb44a0186b2c0083a042b63195d509/docs/figure/mouse-acute-ccl4.Rmd/run-stem-1.png" target="_blank">bea437a</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/saezlab/liver-disease-atlas/blob/c518c89194da6885aa6623ad0fb9b4a989d0a948/docs/figure/mouse-acute-ccl4.Rmd/run-stem-1.png" target="_blank">c518c89</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-characterization" class="section level3">
<h3>Cluster characterization</h3>
<p>STEM clusters are characterized by GO terms, <a href="http://saezlab.github.io/progeny/">PROGENy’s</a> pathways and <a href="http://saezlab.github.io/dorothea/">DoRothEA’s</a> TFs. As statistic over-representation analysis is used.</p>
<pre class="r"><code>stem_res = readRDS(here(output_path, &quot;stem_result.rds&quot;))

signatures = stem_res %&gt;%
  filter(p &lt;= 0.05) %&gt;%
  distinct(profile, gene, p_profile = p)
#&gt; filter: removed 2,781 rows (15%), 15,984 rows remaining
#&gt; distinct: removed 14,208 rows (89%), 1,776 rows remaining

genesets = load_genesets() %&gt;%
  filter(confidence %in% c(NA,&quot;A&quot;, &quot;B&quot;, &quot;C&quot;)) 
#&gt; filter: removed 2,340,732 rows (80%), 597,560 rows remaining
#&gt; select: renamed one variable (gene) and dropped 2 variables
#&gt; mutate: new variable &#39;group&#39; (character) with one unique value and 0% NA
#&gt; gather: reorganized (Androgen, EGFR, Estrogen, Hypoxia, JAK-STAT, …) into (geneset, weight) [was 1299x15, now 18186x3]
#&gt; filter: removed 16,785 rows (92%), 1,401 rows remaining
#&gt; select: dropped one variable (weight)
#&gt; mutate: new variable &#39;group&#39; (character) with one unique value and 0% NA
#&gt; select: renamed 2 variables (geneset, gene) and dropped one variable
#&gt; mutate: new variable &#39;group&#39; (character) with one unique value and 0% NA
#&gt; filter: removed 396,818 rows (39%), 612,694 rows remaining

ora_res = signatures %&gt;%
  nest(sig = c(-profile)) %&gt;%
  dplyr::mutate(ora = sig %&gt;% map(run_ora,  sets = genesets, min_size = 10,
                           options = list(alternative = &quot;greater&quot;), 
                           background_n = 20000)) %&gt;%
  select(-sig) %&gt;%
  unnest(ora)
#&gt; add_count: new variable &#39;n&#39; (integer) with 643 unique values and 0% NA
#&gt; filter: removed 13,996 rows (2%), 598,698 rows remaining
#&gt; select: dropped one variable (n)
#&gt; mutate: new variable &#39;contingency_table&#39; (list) with 1,618 unique values and 0% NA
#&gt; mutate: new variable &#39;stat&#39; (list) with 1,520 unique values and 0% NA
#&gt; group_by: one grouping variable (group)
#&gt; mutate (grouped): new variable &#39;fdr&#39; (double) with 679 unique values and 0% NA
#&gt; ungroup: no grouping variables
#&gt; select: dropped 4 variables (set, conf.low, conf.high, method)
#&gt; add_count: new variable &#39;n&#39; (integer) with 643 unique values and 0% NA
#&gt; filter: removed 13,996 rows (2%), 598,698 rows remaining
#&gt; select: dropped one variable (n)
#&gt; mutate: new variable &#39;contingency_table&#39; (list) with 1,681 unique values and 0% NA
#&gt; mutate: new variable &#39;stat&#39; (list) with 1,575 unique values and 0% NA
#&gt; group_by: one grouping variable (group)
#&gt; mutate (grouped): new variable &#39;fdr&#39; (double) with 977 unique values and 0% NA
#&gt; ungroup: no grouping variables
#&gt; select: dropped 4 variables (set, conf.low, conf.high, method)
#&gt; add_count: new variable &#39;n&#39; (integer) with 643 unique values and 0% NA
#&gt; filter: removed 13,996 rows (2%), 598,698 rows remaining
#&gt; select: dropped one variable (n)
#&gt; mutate: new variable &#39;contingency_table&#39; (list) with 1,379 unique values and 0% NA
#&gt; mutate: new variable &#39;stat&#39; (list) with 1,207 unique values and 0% NA
#&gt; group_by: one grouping variable (group)
#&gt; mutate (grouped): new variable &#39;fdr&#39; (double) with 832 unique values and 0% NA
#&gt; ungroup: no grouping variables
#&gt; select: dropped 4 variables (set, conf.low, conf.high, method)
#&gt; add_count: new variable &#39;n&#39; (integer) with 643 unique values and 0% NA
#&gt; filter: removed 13,996 rows (2%), 598,698 rows remaining
#&gt; select: dropped one variable (n)
#&gt; mutate: new variable &#39;contingency_table&#39; (list) with 1,578 unique values and 0% NA
#&gt; mutate: new variable &#39;stat&#39; (list) with 1,403 unique values and 0% NA
#&gt; group_by: one grouping variable (group)
#&gt; mutate (grouped): new variable &#39;fdr&#39; (double) with 412 unique values and 0% NA
#&gt; ungroup: no grouping variables
#&gt; select: dropped 4 variables (set, conf.low, conf.high, method)
#&gt; select: dropped one variable (sig)

saveRDS(ora_res, here(output_path, &quot;stem_characterization.rds&quot;))</code></pre>
<p>Time spend to execute this analysis: 02:41 minutes.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
#&gt; R version 4.0.2 (2020-06-22)
#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)
#&gt; Running under: macOS Mojave 10.14.5
#&gt; 
#&gt; Matrix products: default
#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
#&gt; 
#&gt; locale:
#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
#&gt; 
#&gt; attached base packages:
#&gt; [1] parallel  stats4    stats     graphics  grDevices datasets  utils    
#&gt; [8] methods   base     
#&gt; 
#&gt; other attached packages:
#&gt;  [1] pd.mouse430.2_3.12.0     DBI_1.1.0                RSQLite_2.2.1           
#&gt;  [4] patchwork_1.1.1          lemon_0.4.5              cowplot_1.1.0           
#&gt;  [7] AachenColorPalette_1.1.2 msigdf_7.1               janitor_2.0.1           
#&gt; [10] dorothea_1.0.1           progeny_1.10.0           biobroom_1.20.0         
#&gt; [13] broom_0.7.3              limma_3.44.3             annotate_1.66.0         
#&gt; [16] XML_3.99-0.5             oligo_1.52.1             Biostrings_2.56.0       
#&gt; [19] XVector_0.28.0           oligoClasses_1.50.4      here_1.0.1              
#&gt; [22] tidylog_1.0.2            forcats_0.5.0            stringr_1.4.0           
#&gt; [25] dplyr_1.0.2              purrr_0.3.4              readr_1.4.0             
#&gt; [28] tidyr_1.1.2              tibble_3.0.4             ggplot2_3.3.2           
#&gt; [31] tidyverse_1.3.0          mouse4302.db_3.2.3       org.Mm.eg.db_3.11.4     
#&gt; [34] AnnotationDbi_1.50.3     IRanges_2.22.2           S4Vectors_0.26.1        
#&gt; [37] Biobase_2.48.0           BiocGenerics_0.34.0      workflowr_1.6.2         
#&gt; 
#&gt; loaded via a namespace (and not attached):
#&gt;  [1] colorspace_2.0-0            ellipsis_0.3.1             
#&gt;  [3] rprojroot_2.0.2             snakecase_0.11.0           
#&gt;  [5] GenomicRanges_1.40.0        fs_1.5.0                   
#&gt;  [7] rstudioapi_0.13             farver_2.0.3               
#&gt;  [9] affyio_1.58.0               ggrepel_0.9.0              
#&gt; [11] bit64_4.0.5                 fansi_0.4.1                
#&gt; [13] lubridate_1.7.9.2           xml2_1.3.2                 
#&gt; [15] codetools_0.2-16            splines_4.0.2              
#&gt; [17] knitr_1.30                  jsonlite_1.7.2             
#&gt; [19] bcellViper_1.24.0           dbplyr_2.0.0               
#&gt; [21] BiocManager_1.30.10         compiler_4.0.2             
#&gt; [23] httr_1.4.2                  backports_1.2.1            
#&gt; [25] assertthat_0.2.1            Matrix_1.2-18              
#&gt; [27] cli_2.2.0                   later_1.1.0.1              
#&gt; [29] htmltools_0.5.0             tools_4.0.2                
#&gt; [31] gtable_0.3.0                glue_1.4.2                 
#&gt; [33] GenomeInfoDbData_1.2.3      affxparser_1.60.0          
#&gt; [35] Rcpp_1.0.5                  cellranger_1.1.0           
#&gt; [37] vctrs_0.3.6                 preprocessCore_1.50.0      
#&gt; [39] iterators_1.0.13            xfun_0.19                  
#&gt; [41] rvest_0.3.6                 lifecycle_0.2.0            
#&gt; [43] renv_0.12.3                 zlibbioc_1.34.0            
#&gt; [45] scales_1.1.1                clisymbols_1.2.0           
#&gt; [47] hms_0.5.3                   promises_1.1.1             
#&gt; [49] SummarizedExperiment_1.18.2 yaml_2.2.1                 
#&gt; [51] gridExtra_2.3               memoise_1.1.0              
#&gt; [53] stringi_1.5.3               foreach_1.5.1              
#&gt; [55] GenomeInfoDb_1.24.2         rlang_0.4.9                
#&gt; [57] pkgconfig_2.0.3             bitops_1.0-6               
#&gt; [59] matrixStats_0.57.0          evaluate_0.14              
#&gt; [61] lattice_0.20-41             labeling_0.4.2             
#&gt; [63] bit_4.0.4                   tidyselect_1.1.0           
#&gt; [65] plyr_1.8.6                  magrittr_2.0.1             
#&gt; [67] R6_2.5.0                    generics_0.1.0             
#&gt; [69] DelayedArray_0.14.1         pillar_1.4.7               
#&gt; [71] haven_2.3.1                 whisker_0.4                
#&gt; [73] withr_2.3.0                 RCurl_1.98-1.2             
#&gt; [75] modelr_0.1.8                crayon_1.3.4               
#&gt; [77] rmarkdown_2.6               grid_4.0.2                 
#&gt; [79] readxl_1.3.1                blob_1.2.1                 
#&gt; [81] git2r_0.27.1                reprex_0.3.0               
#&gt; [83] digest_0.6.27               xtable_1.8-4               
#&gt; [85] ff_4.0.4                    httpuv_1.5.4               
#&gt; [87] munsell_0.5.0               viridisLite_0.3.0</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
