<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Christian H. Holland" />

<meta name="date" content="2020-12-18" />

<title>Chronic CCl4 mouse model</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">meta-liver</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Mouse models
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="mouse-chronic-ccl4.html">Chronic CCl4 mouse model</a>
    </li>
    <li>
      <a href="mouse-acute-apap.html">Acute APAP mouse model</a>
    </li>
    <li>
      <a href="mouse-acute-ccl4.html">Acute CCl4 mouse model</a>
    </li>
    <li>
      <a href="mouse-acute-bdl.html">Acute BDL mouse model</a>
    </li>
    <li>
      <a href="mouse-acute-ph.html">Acute PH mouse model</a>
    </li>
    <li>
      <a href="mouse-acute-lps.html">Acute LPS mouse model</a>
    </li>
    <li>
      <a href="mouse-acute-tunicamycin.html">Acute Tunicamycin mouse model</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Meta analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="meta-analysis-chronic-vs-acute.html">Chronic vs. acute</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/saezlab/meta-liver">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Chronic CCl4 mouse model</h1>
<h4 class="author">Christian H. Holland</h4>
<h4 class="date">2020-12-18</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-12-20
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>meta-liver/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20201218code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20201218)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20201218code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20201218)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
cluster-characterization
</li>
<li>
count-distribution
</li>
<li>
lib-size
</li>
<li>
normalization
</li>
<li>
pca-norm-data
</li>
<li>
pca-raw-data
</li>
<li>
prepare-stem-input
</li>
<li>
run-stem
</li>
<li>
running-limma
</li>
<li>
session-info-chunk-inserted-by-workflowr
</li>
<li>
translate-to-hgnc-symbols
</li>
<li>
volcano-plots
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory <code>mouse-chronic-ccl4_cache</code> and re-run the analysis. To have workflowr automatically delete the cache directory prior to building the file, set <code>delete_cache = TRUE</code> when running <code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsaezlabmetalivertreebea437a2a4cb44a0186b2c0083a042b63195d509targetblankbea437aa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/saezlab/meta-liver/tree/bea437a2a4cb44a0186b2c0083a042b63195d509" target="_blank">bea437a</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsaezlabmetalivertreebea437a2a4cb44a0186b2c0083a042b63195d509targetblankbea437aa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/saezlab/meta-liver/tree/bea437a2a4cb44a0186b2c0083a042b63195d509" target="_blank">bea437a</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/meta-analysis-chronic-vs-acute_cache/
    Ignored:    analysis/mouse-acute-apap_cache/
    Ignored:    analysis/mouse-acute-bdl_cache/
    Ignored:    analysis/mouse-acute-ccl4_cache/
    Ignored:    analysis/mouse-acute-lps_cache/
    Ignored:    analysis/mouse-acute-ph_cache/
    Ignored:    analysis/mouse-acute-tunicamycin_cache/
    Ignored:    analysis/mouse-chronic-ccl4_cache/
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/annotation/
    Ignored:    data/mouse-acute-apap/
    Ignored:    data/mouse-acute-bdl/
    Ignored:    data/mouse-acute-ccl4/
    Ignored:    data/mouse-acute-lps/
    Ignored:    data/mouse-acute-ph/
    Ignored:    data/mouse-acute-tunicamycin/
    Ignored:    data/mouse-chronic-ccl4/
    Ignored:    external_software/.DS_Store
    Ignored:    external_software/README.html
    Ignored:    external_software/stem/.DS_Store
    Ignored:    output/.DS_Store
    Ignored:    output/meta-chronic-vs-acute/
    Ignored:    output/mouse-acute-apap/
    Ignored:    output/mouse-acute-bdl/
    Ignored:    output/mouse-acute-ccl4/
    Ignored:    output/mouse-acute-lps/
    Ignored:    output/mouse-acute-ph/
    Ignored:    output/mouse-acute-tunicamycin/
    Ignored:    output/mouse-chronic-ccl4/
    Ignored:    renv/library/
    Ignored:    renv/staging/

Unstaged changes:
    Modified:   analysis/meta-analysis-chronic-vs-acute.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/mouse-chronic-ccl4.Rmd</code>) and HTML (<code>docs/mouse-chronic-ccl4.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/saezlab/meta-liver/bea437a2a4cb44a0186b2c0083a042b63195d509/docs/mouse-chronic-ccl4.html" target="_blank">bea437a</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/saezlab/meta-liver/blob/c78e88389929df94c6790585a37c4cf7798d04f8/analysis/mouse-chronic-ccl4.Rmd" target="_blank">c78e883</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-20
</td>
<td>
stem characterization
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/saezlab/meta-liver/af384509e61f6d626b8f2f1bccf80c7f075f5cbc/docs/mouse-chronic-ccl4.html" target="_blank">af38450</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/saezlab/meta-liver/blob/78a25d5d1882aa788f0a75a6b2f9753d203c3d76/analysis/mouse-chronic-ccl4.Rmd" target="_blank">78a25d5</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-19
</td>
<td>
minor changes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/saezlab/meta-liver/b9d1691163e532f83f3f3cb4af64f45d7bd0b614/docs/mouse-chronic-ccl4.html" target="_blank">b9d1691</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/saezlab/meta-liver/blob/50f8a682ece8c239d15feba059139c3106748856/analysis/mouse-chronic-ccl4.Rmd" target="_blank">50f8a68</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-19
</td>
<td>
minor changes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/saezlab/meta-liver/6f071f7f5eec32cd24a575b4778016e60f77b032/docs/mouse-chronic-ccl4.html" target="_blank">6f071f7</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/saezlab/meta-liver/blob/d516e6fc3243600f69162271164c501531b0dfdb/analysis/mouse-chronic-ccl4.Rmd" target="_blank">d516e6f</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-19
</td>
<td>
fixed typo
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/saezlab/meta-liver/409ba7dcbf996e2503936fd562720d3da5e6985e/docs/mouse-chronic-ccl4.html" target="_blank">409ba7d</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/saezlab/meta-liver/85a3ce4bd899147455aee2b270f0752b222d2712/docs/mouse-chronic-ccl4.html" target="_blank">85a3ce4</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/saezlab/meta-liver/blob/f6b3eed0087381c7d7e31d0668b4657af08e4828/analysis/mouse-chronic-ccl4.Rmd" target="_blank">f6b3eed</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-19
</td>
<td>
wflow_publish(“analysis/mouse-chronic-ccl4.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/saezlab/meta-liver/c12deea6dd850a705e652c04883a266ff9746057/docs/mouse-chronic-ccl4.html" target="_blank">c12deea</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-19
</td>
<td>
Added stem analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/saezlab/meta-liver/blob/d88ada1282a9d4166450d7ccc3171d4e767e5f13/analysis/mouse-chronic-ccl4.Rmd" target="_blank">d88ada1</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-19
</td>
<td>
add stem analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/saezlab/meta-liver/c73e1266a60e2026d63f9225cdbcbe7647929924/docs/mouse-chronic-ccl4.html" target="_blank">c73e126</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/saezlab/meta-liver/blob/e4502dfacd09a0e05ffa691159ebb0481772995a/analysis/mouse-chronic-ccl4.Rmd" target="_blank">e4502df</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-19
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/saezlab/meta-liver/7bb9aabe6b561f25945051f912a51c735be785dc/docs/mouse-chronic-ccl4.html" target="_blank">7bb9aab</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/saezlab/meta-liver/blob/261f1a4389dd35353ef5619e4f184ae778137e69/analysis/mouse-chronic-ccl4.Rmd" target="_blank">261f1a4</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-18
</td>
<td>
Start my new project
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Here we analysis a mouse model of CCl4 induced chronic liver disease. The transcriptomic profiles were measured at time point 0, 2, 6, and 12 month. For the time points 2 and 12 month time-matched oil controls are available.</p>
</div>
<div id="libraries-and-sources" class="section level2">
<h2>Libraries and sources</h2>
<p>These libraries and sources are used for this analysis.</p>
<pre class="r"><code>library(tidyverse)
library(tidylog)
library(here)

library(edgeR)
library(biobroom)
library(progeny)
library(dorothea)

library(janitor)
library(msigdf) # remotes::install_github(&quot;ToledoEM/msigdf@v7.1&quot;)

library(AachenColorPalette)
library(cowplot)
library(lemon)
library(patchwork)

options(&quot;tidylog.display&quot; = list(print))
source(here(&quot;code/utils-rnaseq.R&quot;))
source(here(&quot;code/utils-utils.R&quot;))
source(here(&quot;code/utils-plots.R&quot;))</code></pre>
<p>Definition of global variables that are used throughout this analysis.</p>
<pre class="r"><code># i/o
data_path &lt;- &quot;data/mouse-chronic-ccl4&quot;
output_path &lt;- &quot;output/mouse-chronic-ccl4&quot;

# graphical parameters
# fontsize
fz &lt;- 9</code></pre>
</div>
<div id="preliminary-exploratory-analysis" class="section level2">
<h2>Preliminary exploratory analysis</h2>
<div id="library-size" class="section level3">
<h3>Library size</h3>
<p>Barplot of the library size (total counts) for each of the samples.</p>
<pre class="r"><code>count_matrix &lt;- readRDS(here(data_path, &quot;count_matrix.rds&quot;))

plot_libsize(count_matrix) +
  my_theme(fsize = fz)</code></pre>
<p><img src="figure/mouse-chronic-ccl4.Rmd/lib-size-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-lib-size-1">
Past versions of lib-size-1.png
</button>
</p>
<div id="fig-lib-size-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/meta-liver/blob/409ba7dcbf996e2503936fd562720d3da5e6985e/docs/figure/mouse-chronic-ccl4.Rmd/lib-size-1.png" target="_blank">409ba7d</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="count-distribution" class="section level3">
<h3>Count distribution</h3>
<p>Violin plots of the raw read counts for each of the samples.</p>
<pre class="r"><code>count_matrix &lt;- readRDS(here(data_path, &quot;count_matrix.rds&quot;))
meta &lt;- readRDS(here(data_path, &quot;meta_data.rds&quot;))

count_matrix %&gt;%
  tdy(&quot;gene&quot;, &quot;sample&quot;, &quot;count&quot;, meta) %&gt;%
  arrange(treatment) %&gt;%
  ggplot(aes(
    x = fct_reorder(sample, as.numeric(treatment)), y = log10(count + 1),
    group = sample, fill = treatment
  )) +
  geom_violin() +
  theme(
    axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1),
    legend.position = &quot;top&quot;
  ) +
  labs(x = NULL) +
  my_theme(grid = &quot;no&quot;, fsize = fz)
#&gt; gather: reorganized (I00001, I00002, I00003, I00004, I00005, …) into (sample, count) [was 32544x37, now 1171584x3]
#&gt; left_join: added 3 columns (time, treatment, group)
#&gt;            &gt; rows only in x           0
#&gt;            &gt; rows only in y  (        0)
#&gt;            &gt; matched rows     1,171,584
#&gt;            &gt;                 ===========
#&gt;            &gt; rows total       1,171,584</code></pre>
<p><img src="figure/mouse-chronic-ccl4.Rmd/count-distribution-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-count-distribution-1">
Past versions of count-distribution-1.png
</button>
</p>
<div id="fig-count-distribution-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/meta-liver/blob/409ba7dcbf996e2503936fd562720d3da5e6985e/docs/figure/mouse-chronic-ccl4.Rmd/count-distribution-1.png" target="_blank">409ba7d</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pca-of-raw-data" class="section level3">
<h3>PCA of raw data</h3>
<p>PCA plot of raw read counts contextualized based on the time point and treatment. Before gene with a constant expression across all samples are removed and count values are transformed to log2 scale. Only the top 1000 most variable genes are used as features.</p>
<pre class="r"><code>count_matrix &lt;- readRDS(here(data_path, &quot;count_matrix.rds&quot;))
meta &lt;- readRDS(here(data_path, &quot;meta_data.rds&quot;))

stopifnot(colnames(count_matrix) == meta$sample)

# remove constant expressed genes and transform to log2 scale
preprocessed_count_matrix &lt;- preprocess_count_matrix(count_matrix)
#&gt; Discarding 7711 genes 
#&gt; Keeping 24833 genes


pca_result &lt;- do_pca(preprocessed_count_matrix, meta, top_n_var_genes = 1000)
#&gt; left_join: added 3 columns (time, treatment, group)
#&gt;            &gt; rows only in x    0
#&gt;            &gt; rows only in y  ( 0)
#&gt;            &gt; matched rows     36
#&gt;            &gt;                 ====
#&gt;            &gt; rows total       36

plot_pca(pca_result, feature = &quot;time&quot;) +
  plot_pca(pca_result, feature = &quot;treatment&quot;) &amp;
  my_theme(fsize = fz)</code></pre>
<p><img src="figure/mouse-chronic-ccl4.Rmd/pca-raw-data-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-raw-data-1">
Past versions of pca-raw-data-1.png
</button>
</p>
<div id="fig-pca-raw-data-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/meta-liver/blob/409ba7dcbf996e2503936fd562720d3da5e6985e/docs/figure/mouse-chronic-ccl4.Rmd/pca-raw-data-1.png" target="_blank">409ba7d</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="data-processing" class="section level2">
<h2>Data processing</h2>
<div id="normalization" class="section level3">
<h3>Normalization</h3>
<p>Raw read counts are normalized by first filtering out lowly expressed genes, TMM normalization and finally logCPM transformation.</p>
<pre class="r"><code>count_matrix &lt;- readRDS(here(data_path, &quot;count_matrix.rds&quot;))
meta &lt;- readRDS(here(data_path, &quot;meta_data.rds&quot;))

stopifnot(meta$sample == colnames(count_matrix))

dge_obj &lt;- DGEList(count_matrix, group = meta$group)

# filter low read counts, TMM normalization and logCPM transformation
norm &lt;- voom_normalization(dge_obj)
#&gt; Discarding 17206 genes 
#&gt; Keeping 15338 genes

saveRDS(norm, here(output_path, &quot;normalized_expression.rds&quot;))</code></pre>
</div>
<div id="pca-of-normalized-data" class="section level3">
<h3>PCA of normalized data</h3>
<p>PCA plot of normalized expression data contextualized based on the time point and treatment. Only the top 1000 most variable genes are used as features.</p>
<pre class="r"><code>expr &lt;- readRDS(here(output_path, &quot;normalized_expression.rds&quot;))
meta &lt;- readRDS(here(data_path, &quot;meta_data.rds&quot;))

pca_result &lt;- do_pca(expr, meta, top_n_var_genes = 1000)
#&gt; left_join: added 3 columns (time, treatment, group)
#&gt;            &gt; rows only in x    0
#&gt;            &gt; rows only in y  ( 0)
#&gt;            &gt; matched rows     36
#&gt;            &gt;                 ====
#&gt;            &gt; rows total       36

saveRDS(pca_result, here(output_path, &quot;pca_result.rds&quot;))

plot_pca(pca_result, feature = &quot;time&quot;) +
  plot_pca(pca_result, feature = &quot;treatment&quot;) &amp;
  my_theme(fsize = fz)</code></pre>
<p><img src="figure/mouse-chronic-ccl4.Rmd/pca-norm-data-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-norm-data-1">
Past versions of pca-norm-data-1.png
</button>
</p>
<div id="fig-pca-norm-data-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/meta-liver/blob/409ba7dcbf996e2503936fd562720d3da5e6985e/docs/figure/mouse-chronic-ccl4.Rmd/pca-norm-data-1.png" target="_blank">409ba7d</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="differential-gene-expression-analysis" class="section level2">
<h2>Differential gene expression analysis</h2>
<div id="running-limma" class="section level3">
<h3>Running limma</h3>
<p>Differential gene expression analysis via limma with the aim to identify the effect of CCl4 intoxication while regression out the effect of the oil.</p>
<pre class="r"><code># load expression and meta data
expr &lt;- readRDS(here(output_path, &quot;normalized_expression.rds&quot;))
meta &lt;- readRDS(here(data_path, &quot;meta_data.rds&quot;))

stopifnot(colnames(expr) == meta$sample)

# build design matrix
design &lt;- model.matrix(~ 0 + group, data = meta)
rownames(design) &lt;- meta$sample
colnames(design) &lt;- levels(meta$group)

# define contrasts
contrasts &lt;- makeContrasts(
  # effect of olive oil
  oil_2m_vs_0m = oil.2 - wt,
  oil_12m_vs_0m = oil.12 - wt,
  oil_12m_vs_2m = oil.12 - oil.2,

  # treatment vs control ignoring the effect of oil
  ccl_2m_vs_0m = ccl4.2 - wt,
  ccl_6m_vs_0m = ccl4.6 - wt,
  ccl_12m_vs_0m = ccl4.12 - wt,

  # treatment vs control regressing out the effect of oil
  pure_ccl_2m_vs_0m = (ccl4.2 - wt) - (oil.2 - wt),
  pure_ccl_6m_vs_0m = (ccl4.6 - wt) - ((oil.2 + oil.12) / 2 - wt),
  pure_ccl_12m_vs_0m = (ccl4.12 - wt) - (oil.12 - wt),

  # consecutive time point comparison
  consec_12m_vs_6m = ccl4.12 - ccl4.6,
  consec_12m_vs_2m = ccl4.12 - ccl4.2,
  # consec_48w_vs_8w_2 = (ccl4.48 - oil.48) - (ccl4.8 - oil.8),
  consec_6m_vs_2m = ccl4.6 - ccl4.2,
  levels = design
)

limma_result &lt;- run_limma(expr, design, contrasts) %&gt;%
  assign_deg()
#&gt; Warning: `tbl_df()` is deprecated as of dplyr 1.0.0.
#&gt; Please use `tibble::as_tibble()` instead.
#&gt; This warning is displayed once every 8 hours.
#&gt; Call `lifecycle::last_warnings()` to see where this warning was generated.
#&gt; select: renamed 3 variables (contrast, logFC, pval) and dropped one variable
#&gt; group_by: one grouping variable (contrast)
#&gt; mutate (grouped): new variable &#39;fdr&#39; (double) with 115,674 unique values and 0% NA
#&gt; ungroup: no grouping variables
#&gt; mutate: new variable &#39;regulation&#39; (character) with 3 unique values and 0% NA
#&gt; mutate: converted &#39;regulation&#39; from character to factor (0 new NA)

deg_df &lt;- limma_result %&gt;%
  mutate(contrast = factor(contrast, levels = c(
    &quot;ccl_2m_vs_0m&quot;, &quot;ccl_6m_vs_0m&quot;,
    &quot;ccl_12m_vs_0m&quot;,
    &quot;pure_ccl_2m_vs_0m&quot;,
    &quot;pure_ccl_6m_vs_0m&quot;,
    &quot;pure_ccl_12m_vs_0m&quot;,
    &quot;consec_6m_vs_2m&quot;,
    &quot;consec_12m_vs_2m&quot;,
    &quot;consec_12m_vs_6m&quot;,
    &quot;oil_2m_vs_0m&quot;, &quot;oil_12m_vs_0m&quot;,
    &quot;oil_12m_vs_2m&quot;
  ))) %&gt;%
  mutate(contrast_reference = case_when(
    str_detect(contrast, &quot;oil&quot;) ~ &quot;oil&quot;,
    str_detect(contrast, &quot;^pure_ccl&quot;) ~ &quot;pure_ccl4&quot;,
    str_detect(contrast, &quot;^ccl&quot;) ~ &quot;ccl4&quot;,
    str_detect(contrast, &quot;consec&quot;) ~ &quot;consec&quot;
  ))
#&gt; mutate: changed 0 values (0%) of &#39;contrast&#39; (0 new NA)
#&gt; mutate: new variable &#39;contrast_reference&#39; (character) with 4 unique values and 0% NA

saveRDS(deg_df, here(output_path, &quot;limma_result.rds&quot;))</code></pre>
</div>
<div id="volcano-plots" class="section level3">
<h3>Volcano plots</h3>
<p>Volcano plots visualizing the effect of CCl4 on gene expression.</p>
<pre class="r"><code>df &lt;- readRDS(here(output_path, &quot;limma_result.rds&quot;))

df %&gt;%
  filter(contrast_reference == &quot;pure_ccl4&quot;) %&gt;%
  plot_volcano() +
  my_theme(grid = &quot;y&quot;, fsize = fz)
#&gt; filter: removed 138,042 rows (75%), 46,014 rows remaining
#&gt; rename: renamed one variable (p)</code></pre>
<p><img src="figure/mouse-chronic-ccl4.Rmd/volcano-plots-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-volcano-plots-1">
Past versions of volcano-plots-1.png
</button>
</p>
<div id="fig-volcano-plots-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/meta-liver/blob/409ba7dcbf996e2503936fd562720d3da5e6985e/docs/figure/mouse-chronic-ccl4.Rmd/volcano-plots-1.png" target="_blank">409ba7d</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="time-series-clustering" class="section level2">
<h2>Time series clustering</h2>
<p>Gene expression trajectories are clustered using the <a href="http://www.cs.cmu.edu/~jernst/stem/">STEM</a> software. The cluster algorithm is described <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-7-191">here</a>.</p>
<div id="prepare-input" class="section level3">
<h3>Prepare input</h3>
<pre class="r"><code># prepare input for stem analysis
df &lt;- readRDS(here(output_path, &quot;limma_result.rds&quot;))

stem_inputs &lt;- df %&gt;%
  filter(contrast_reference %in% c(&quot;pure_ccl4&quot;)) %&gt;%
  mutate(class = str_c(&quot;Month &quot;, parse_number(as.character(contrast)))) %&gt;%
  mutate(class = factor(class, levels = c(&quot;Month 2&quot;, &quot;Month 6&quot;, &quot;Month 12&quot;))) %&gt;%
  select(gene, class, logFC, contrast_reference)
#&gt; filter: removed 138,042 rows (75%), 46,014 rows remaining
#&gt; mutate: new variable &#39;class&#39; (character) with 3 unique values and 0% NA
#&gt; mutate: converted &#39;class&#39; from character to factor (0 new NA)
#&gt; select: dropped 5 variables (contrast, statistic, pval, fdr, regulation)

stem_inputs %&gt;%
  select(-contrast_reference) %&gt;%
  pivot_wider(names_from = class, values_from = logFC) %&gt;%
  write_delim(here(output_path, &quot;stem/input/pure_ccl4.txt&quot;), delim = &quot;\t&quot;)
#&gt; select: dropped one variable (contrast_reference)
#&gt; pivot_wider: reorganized (class, logFC) into (Month 2, Month 6, Month 12) [was 46014x3, now 15338x4]</code></pre>
</div>
<div id="run-stem" class="section level3">
<h3>Run STEM</h3>
<p>STEM is implemented in Java. The .jar file is called from R. Only significant time series clusters are visualized.</p>
<pre class="r"><code># execute stem
stem_res &lt;- run_stem(here(output_path, &quot;stem&quot;), clear_output = T)
#&gt; 
#&gt; ── Column specification ────────────────────────────────────────────────────────
#&gt; cols(
#&gt;   gene = col_character(),
#&gt;   `Month 2` = col_double(),
#&gt;   `Month 6` = col_double(),
#&gt;   `Month 12` = col_double()
#&gt; )
#&gt; distinct: no rows removed
#&gt; mutate: new variable &#39;gene&#39; (character) with 15,338 unique values and 0% NA
#&gt; distinct: no rows removed
#&gt; 
#&gt; ── Column specification ────────────────────────────────────────────────────────
#&gt; cols(
#&gt;   gene = col_character(),
#&gt;   SPOT = col_character(),
#&gt;   Profile = col_double(),
#&gt;   `0` = col_double(),
#&gt;   `Month 2` = col_double(),
#&gt;   `Month 6` = col_double(),
#&gt;   `Month 12` = col_double()
#&gt; )
#&gt; mutate: new variable &#39;key&#39; (character) with one unique value and 0% NA
#&gt; select: dropped one variable (spot)
#&gt; gather: reorganized (x0, month_2, month_6, month_12) into (time, value) [was 3193x7, now 12772x5]
#&gt; mutate: converted &#39;time&#39; from character to double (0 new NA)
#&gt; 
#&gt; ── Column specification ────────────────────────────────────────────────────────
#&gt; cols(
#&gt;   `Profile ID` = col_double(),
#&gt;   `Profile Model` = col_character(),
#&gt;   `Cluster (-1 non-significant)` = col_double(),
#&gt;   `# Genes Assigned` = col_double(),
#&gt;   `# Gene Expected` = col_double(),
#&gt;   `p-value` = col_double()
#&gt; )
#&gt; mutate: new variable &#39;key&#39; (character) with one unique value and 0% NA
#&gt; select: renamed 4 variables (profile, y_coords, size, p) and dropped 2 variables
#&gt; Joining, by = c(&quot;profile&quot;, &quot;key&quot;)
#&gt; inner_join: added 3 columns (y_coords, size, p)
#&gt;             &gt; rows only in x  (     0)
#&gt;             &gt; rows only in y  (     0)
#&gt;             &gt; matched rows     12,772
#&gt;             &gt;                 ========
#&gt;             &gt; rows total       12,772
#&gt; inner_join: added one column (symbol)
#&gt;             &gt; rows only in x  (     0)
#&gt;             &gt; rows only in y  (12,145)
#&gt;             &gt; matched rows     12,772
#&gt;             &gt;                 ========
#&gt;             &gt; rows total       12,772
#&gt; transmute: dropped one variable (symbol)
#&gt;            changed 12,644 values (99%) of &#39;gene&#39; (0 new NA)

saveRDS(stem_res, here(output_path, &quot;stem_result.rds&quot;))

stem_res %&gt;%
  filter(p &lt;= 0.05) %&gt;%
  filter(key == &quot;pure_ccl4&quot;) %&gt;%
  distinct() %&gt;%
  plot_stem_profiles(model_profile = F) +
  labs(x = &quot;Time in Month&quot;, y = &quot;logFC&quot;) +
  my_theme(grid = &quot;y&quot;, fsize = fz)
#&gt; filter: removed 3,080 rows (24%), 9,692 rows remaining
#&gt; filter: no rows removed
#&gt; distinct: no rows removed
#&gt; group_by: 4 grouping variables (key, profile, p, time)
#&gt; `summarise()` regrouping output by &#39;key&#39;, &#39;profile&#39;, &#39;p&#39; (override with `.groups` argument)
#&gt; ungroup: no grouping variables</code></pre>
<p><img src="figure/mouse-chronic-ccl4.Rmd/run-stem-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-run-stem-1">
Past versions of run-stem-1.png
</button>
</p>
<div id="fig-run-stem-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/saezlab/meta-liver/blob/409ba7dcbf996e2503936fd562720d3da5e6985e/docs/figure/mouse-chronic-ccl4.Rmd/run-stem-1.png" target="_blank">409ba7d</a>
</td>
<td>
christianholland
</td>
<td>
2020-12-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-characterization" class="section level3">
<h3>Cluster characterization</h3>
<p>STEM clusters are characterized by GO terms, <a href="http://saezlab.github.io/progeny/">PROGENy’s</a> pathways and <a href="http://saezlab.github.io/dorothea/">DoRothEA’s</a> TFs. As statistic over-representation analysis is used.</p>
<pre class="r"><code>stem_res = readRDS(here(output_path, &quot;stem_result.rds&quot;))

signatures = stem_res %&gt;%
  filter(p &lt;= 0.05) %&gt;%
  distinct(profile, gene, p_profile = p)
#&gt; filter: removed 3,080 rows (24%), 9,692 rows remaining
#&gt; distinct: removed 7,269 rows (75%), 2,423 rows remaining

genesets = load_genesets() %&gt;%
  filter(confidence %in% c(NA,&quot;A&quot;, &quot;B&quot;, &quot;C&quot;)) 
#&gt; filter: removed 2,340,732 rows (80%), 597,560 rows remaining
#&gt; select: renamed one variable (gene) and dropped 2 variables
#&gt; mutate: new variable &#39;group&#39; (character) with one unique value and 0% NA
#&gt; gather: reorganized (Androgen, EGFR, Estrogen, Hypoxia, JAK-STAT, …) into (geneset, weight) [was 1299x15, now 18186x3]
#&gt; filter: removed 16,785 rows (92%), 1,401 rows remaining
#&gt; select: dropped one variable (weight)
#&gt; mutate: new variable &#39;group&#39; (character) with one unique value and 0% NA
#&gt; select: renamed 2 variables (geneset, gene) and dropped one variable
#&gt; mutate: new variable &#39;group&#39; (character) with one unique value and 0% NA
#&gt; filter: removed 396,818 rows (39%), 612,694 rows remaining

ora_res = signatures %&gt;%
  nest(sig = c(-profile)) %&gt;%
  dplyr::mutate(ora = sig %&gt;% map(run_ora,  sets = genesets, min_size = 10,
                           options = list(alternative = &quot;greater&quot;), 
                           background_n = 20000)) %&gt;%
  select(-sig) %&gt;%
  unnest(ora)
#&gt; add_count: new variable &#39;n&#39; (integer) with 643 unique values and 0% NA
#&gt; filter: removed 13,996 rows (2%), 598,698 rows remaining
#&gt; select: dropped one variable (n)
#&gt; mutate: new variable &#39;contingency_table&#39; (list) with 1,302 unique values and 0% NA
#&gt; mutate: new variable &#39;stat&#39; (list) with 1,090 unique values and 0% NA
#&gt; group_by: one grouping variable (group)
#&gt; mutate (grouped): new variable &#39;fdr&#39; (double) with 262 unique values and 0% NA
#&gt; ungroup: no grouping variables
#&gt; select: dropped 4 variables (set, conf.low, conf.high, method)
#&gt; add_count: new variable &#39;n&#39; (integer) with 643 unique values and 0% NA
#&gt; filter: removed 13,996 rows (2%), 598,698 rows remaining
#&gt; select: dropped one variable (n)
#&gt; mutate: new variable &#39;contingency_table&#39; (list) with 1,691 unique values and 0% NA
#&gt; mutate: new variable &#39;stat&#39; (list) with 1,572 unique values and 0% NA
#&gt; group_by: one grouping variable (group)
#&gt; mutate (grouped): new variable &#39;fdr&#39; (double) with 1,023 unique values and 0% NA
#&gt; ungroup: no grouping variables
#&gt; select: dropped 4 variables (set, conf.low, conf.high, method)
#&gt; add_count: new variable &#39;n&#39; (integer) with 643 unique values and 0% NA
#&gt; filter: removed 13,996 rows (2%), 598,698 rows remaining
#&gt; select: dropped one variable (n)
#&gt; mutate: new variable &#39;contingency_table&#39; (list) with 1,603 unique values and 0% NA
#&gt; mutate: new variable &#39;stat&#39; (list) with 1,506 unique values and 0% NA
#&gt; group_by: one grouping variable (group)
#&gt; mutate (grouped): new variable &#39;fdr&#39; (double) with 659 unique values and 0% NA
#&gt; ungroup: no grouping variables
#&gt; select: dropped 4 variables (set, conf.low, conf.high, method)
#&gt; add_count: new variable &#39;n&#39; (integer) with 643 unique values and 0% NA
#&gt; filter: removed 13,996 rows (2%), 598,698 rows remaining
#&gt; select: dropped one variable (n)
#&gt; mutate: new variable &#39;contingency_table&#39; (list) with 1,387 unique values and 0% NA
#&gt; mutate: new variable &#39;stat&#39; (list) with 1,209 unique values and 0% NA
#&gt; group_by: one grouping variable (group)
#&gt; mutate (grouped): new variable &#39;fdr&#39; (double) with 303 unique values and 0% NA
#&gt; ungroup: no grouping variables
#&gt; select: dropped 4 variables (set, conf.low, conf.high, method)
#&gt; add_count: new variable &#39;n&#39; (integer) with 643 unique values and 0% NA
#&gt; filter: removed 13,996 rows (2%), 598,698 rows remaining
#&gt; select: dropped one variable (n)
#&gt; mutate: new variable &#39;contingency_table&#39; (list) with 1,402 unique values and 0% NA
#&gt; mutate: new variable &#39;stat&#39; (list) with 1,253 unique values and 0% NA
#&gt; group_by: one grouping variable (group)
#&gt; mutate (grouped): new variable &#39;fdr&#39; (double) with 348 unique values and 0% NA
#&gt; ungroup: no grouping variables
#&gt; select: dropped 4 variables (set, conf.low, conf.high, method)
#&gt; add_count: new variable &#39;n&#39; (integer) with 643 unique values and 0% NA
#&gt; filter: removed 13,996 rows (2%), 598,698 rows remaining
#&gt; select: dropped one variable (n)
#&gt; mutate: new variable &#39;contingency_table&#39; (list) with 1,109 unique values and 0% NA
#&gt; mutate: new variable &#39;stat&#39; (list) with 852 unique values and 0% NA
#&gt; group_by: one grouping variable (group)
#&gt; mutate (grouped): new variable &#39;fdr&#39; (double) with 333 unique values and 0% NA
#&gt; ungroup: no grouping variables
#&gt; select: dropped 4 variables (set, conf.low, conf.high, method)
#&gt; add_count: new variable &#39;n&#39; (integer) with 643 unique values and 0% NA
#&gt; filter: removed 13,996 rows (2%), 598,698 rows remaining
#&gt; select: dropped one variable (n)
#&gt; mutate: new variable &#39;contingency_table&#39; (list) with 1,221 unique values and 0% NA
#&gt; mutate: new variable &#39;stat&#39; (list) with 1,018 unique values and 0% NA
#&gt; group_by: one grouping variable (group)
#&gt; mutate (grouped): new variable &#39;fdr&#39; (double) with 358 unique values and 0% NA
#&gt; ungroup: no grouping variables
#&gt; select: dropped 4 variables (set, conf.low, conf.high, method)
#&gt; select: dropped one variable (sig)

saveRDS(ora_res, here(output_path, &quot;stem_characterization.rds&quot;))</code></pre>
</div>
</div>
<div id="translation-to-hgnc-symbols" class="section level2">
<h2>Translation to HGNC symbols</h2>
<p>For later comparisons to human data the mouse gene symbols are mapped to their human orthologs.</p>
<pre class="r"><code>df &lt;- readRDS(here(output_path, &quot;limma_result.rds&quot;))

mapped_df &lt;- df %&gt;%
  translate_gene_ids(from = &quot;symbol_mgi&quot;, to = &quot;symbol_hgnc&quot;) %&gt;%
  drop_na() %&gt;%
  # for duplicated genes, keep the one with the highest absolute logFC
  group_by(contrast_reference, contrast, gene) %&gt;%
  slice_max(order_by = abs(logFC), n = 1, with_ties = F) %&gt;%
  ungroup()
#&gt; select: dropped 6 variables (ensembl_mgi, ensembl_v_mgi, entrez_mgi, ensembl_hgnc, ensembl_v_hgnc, …)
#&gt; drop_na: removed 1,905 rows (6%), 30,461 rows remaining
#&gt; rename: renamed one variable (symbol_mgi)
#&gt; left_join: added one column (symbol_hgnc)
#&gt;            &gt; rows only in x    22,992
#&gt;            &gt; rows only in y  ( 16,094)
#&gt;            &gt; matched rows     172,404    (includes duplicates)
#&gt;            &gt;                 =========
#&gt;            &gt; rows total       195,396
#&gt; select: renamed one variable (gene) and dropped one variable
#&gt; drop_na: removed 22,992 rows (12%), 172,404 rows remaining
#&gt; group_by: 3 grouping variables (contrast_reference, contrast, gene)
#&gt; slice_max (grouped): removed 11,328 rows (7%), 161,076 rows remaining
#&gt; ungroup: no grouping variables

saveRDS(mapped_df, here(output_path, &quot;limma_result_hs.rds&quot;))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
#&gt; R version 4.0.2 (2020-06-22)
#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)
#&gt; Running under: macOS Mojave 10.14.5
#&gt; 
#&gt; Matrix products: default
#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
#&gt; 
#&gt; locale:
#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
#&gt; 
#&gt; attached base packages:
#&gt; [1] stats     graphics  grDevices datasets  utils     methods   base     
#&gt; 
#&gt; other attached packages:
#&gt;  [1] patchwork_1.1.1          lemon_0.4.5              cowplot_1.1.0           
#&gt;  [4] AachenColorPalette_1.1.2 msigdf_7.1               janitor_2.0.1           
#&gt;  [7] dorothea_1.0.1           progeny_1.10.0           biobroom_1.20.0         
#&gt; [10] broom_0.7.3              edgeR_3.30.3             limma_3.44.3            
#&gt; [13] here_1.0.1               tidylog_1.0.2            forcats_0.5.0           
#&gt; [16] stringr_1.4.0            dplyr_1.0.2              purrr_0.3.4             
#&gt; [19] readr_1.4.0              tidyr_1.1.2              tibble_3.0.4            
#&gt; [22] ggplot2_3.3.2            tidyverse_1.3.0          workflowr_1.6.2         
#&gt; 
#&gt; loaded via a namespace (and not attached):
#&gt;  [1] fs_1.5.0            lubridate_1.7.9.2   httr_1.4.2         
#&gt;  [4] rprojroot_2.0.2     tools_4.0.2         backports_1.2.1    
#&gt;  [7] R6_2.5.0            DBI_1.1.0           BiocGenerics_0.34.0
#&gt; [10] colorspace_2.0-0    withr_2.3.0         tidyselect_1.1.0   
#&gt; [13] gridExtra_2.3       compiler_4.0.2      git2r_0.27.1       
#&gt; [16] cli_2.2.0           rvest_0.3.6         Biobase_2.48.0     
#&gt; [19] xml2_1.3.2          labeling_0.4.2      scales_1.1.1       
#&gt; [22] digest_0.6.27       rmarkdown_2.6       pkgconfig_2.0.3    
#&gt; [25] htmltools_0.5.0     bcellViper_1.24.0   dbplyr_2.0.0       
#&gt; [28] rlang_0.4.9         readxl_1.3.1        rstudioapi_0.13    
#&gt; [31] generics_0.1.0      farver_2.0.3        jsonlite_1.7.2     
#&gt; [34] magrittr_2.0.1      Rcpp_1.0.5          munsell_0.5.0      
#&gt; [37] fansi_0.4.1         lifecycle_0.2.0     stringi_1.5.3      
#&gt; [40] whisker_0.4         yaml_2.2.1          snakecase_0.11.0   
#&gt; [43] plyr_1.8.6          grid_4.0.2          parallel_4.0.2     
#&gt; [46] promises_1.1.1      ggrepel_0.9.0       crayon_1.3.4       
#&gt; [49] lattice_0.20-41     haven_2.3.1         hms_0.5.3          
#&gt; [52] locfit_1.5-9.4      knitr_1.30          pillar_1.4.7       
#&gt; [55] codetools_0.2-16    clisymbols_1.2.0    reprex_0.3.0       
#&gt; [58] glue_1.4.2          evaluate_0.14       renv_0.12.3        
#&gt; [61] modelr_0.1.8        vctrs_0.3.6         httpuv_1.5.4       
#&gt; [64] cellranger_1.1.0    gtable_0.3.0        assertthat_0.2.1   
#&gt; [67] xfun_0.19           later_1.1.0.1       viridisLite_0.3.0  
#&gt; [70] ellipsis_0.3.1</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
