---
title: "Home"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

# Transcriptomic cross-species analysis of chronic liver disease reveals a consistent regulation pattern between humans and mice

The results and the analysis scripts presented on this website ensures the reproducibility of all bioinformatics related findings presented in [_"Transcriptomic cross-species analysis of chronic liver disease reveals a consistent regulation pattern between humans and mice"_](). Please see below to get more information about the individual analysis and some tips on how to reproduce the result.

Abstract
--------

### Background and aims 
Mouse models are frequently used to study chronic liver diseases (CLD). To allow a better assessment of translational relevance, we quantified the similarity of prevalent mouse models to human CLD based on transcriptome data.

### Methods 
RNA-sequencing and gene array data from 372 patients with CLD (NAFLD, NASH, HCV, PBC, and PSC) were compared to acute and chronic mouse models with 227 mice and additionally 9 published gene sets of chronic mouse models. Genes consistently altered in humans and mice were mapped to specific liver cell types based on single-cell RNA-sequencing and validated by immunostaining. 

### Results 
Similarity of the top differentially expressed genes between humans and mice varied among the individual mouse models and depended on the period of damage induction. Some models, e.g. 12 months induction of damage by CCl4, reached high similarity to humans with 0.4 recall and 0.33 precision, respectively. Consistently upregulated genes between the chronic CCl4 mouse model and human data were enriched in inflammatory and developmental processes, and mapped to cholangiocytes, macrophages, endothelial and mesenchymal cells, while downregulated genes were enriched in metabolic functions and mapped to hepatocytes. Immunostaining confirmed selected consistent genes and their cell-type specificity. Upregulated genes in both acute and chronic mouse models showed a higher recall and precision with respect to human CLDNAFLD and NASH than exclusively acute or chronic genes. 

### Conclusion 
Our analyses led to the identification of similarly regulated genes in human and mouse liver disease. Although major species differences exist, mouse models may recall at least 30% of the genes significantly altered in human CLDliver diseases. The relevance of individual genes in CLD can be assessed at https://saezlab.shinyapps.io/liverdiseaseatlas/.

Analysis
--------

The tab `Mouse models` contains Rmarkdown scripts to analyze and characterize the transcriptomic profiles of acute and chronic liver disease mouse models. These analyses comprised

* Normalization
* PCA analysis
* Differential gene expression analysis
* Time series clustering and characterization (if applicable)

The tab `Patient cohorts` contains Rmarkdown scripts to analyze and characterize the transcriptomic profiles of patient cohorts suffering from various chronic liver disease etiologies. These analyses comprised:

* Normalization
* PCA analysis
* Differential gene expression analysis

The tab `Meta analysis` contains Rmarkdown scripts to integrate acute and chronic mouse models with patient cohorts.

* [Chronic vs. acute](meta-chronic-vs-acute.html)
  * Identification of exclusively and commonly- regulated genes of chronic and acute disease in mice.
  
* [Mouse vs. human](meta-mouse-vs-human.html)
  * Identification of consistently regulated genes in the chronic CCl~4~[mouse-chronic-ccl4.html] mouse model and patients.
  * Quantification of the similarity of the in total 12 chronic mouse models with the different human patient cohorts based on precision and recall.
  
The tab `Figures` contains Rmarkdown scripts to generate the figures used in the manuscript.


How can I reproduce the analysis?
--------
We have used the [workflowr](https://jdblischak.github.io/workflowr/) package to organize the analysis scipts within this project so please familiarize yourself with its concept.

1. Clone the repository from https://github.com/saezlab/liver-disease-atlas which automatically provides you with all analysis scripts and the correct directory structure.
2. You need to install all packages that are required for the analyses. The package [renv](https://rstudio.github.io/renv/articles/renv.html) allows you to easily install the packages with the correct versions:

```{r eval=FALSE}
install.packages("renv")
renv::restore()
```

3. The GitHub repository contains only the analysis code and same small objects. All raw data is deposited at [Zenodo](https://doi.org/10.5281/zenodo.4407208). [![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.4407208.svg)](https://doi.org/10.5281/zenodo.4407208)
Download the zipped `data` folder unzipp it and replace the existing `data` folder at the root level of the R-project.

4. Run all analyses by running
```{r eval=FALSE}
install.packages("workflowr")
workflowr::wflow_build(republish = TRUE)
```
All intermediate and final results will be saved in the `output` folder.

How to cite?
--------
>Holland CH, Ramirez Flores RO, Myllys M, Hassan R, Edlund K, Hofmann U, Marchan R, Cadenas C, Reinders J, Hoehme S, Seddek A, Dooley S, Keitel V, Godoy P, Begher-Tibbe B, Trautwein C, Rupp C, Mueller S, Longerich T, Hengstler JG^#^, Saez-Rodriguez J^#^, Ghallab A^#^. "Transcriptomic cross-species analysis of chronic liver disease reveals a consistent regulation pattern between humans and mice." _In preparation_. 2021.

^#^_Shared senior authorship_
